{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter Configs\n",
    "\n",
    "Filter configs tell Ella how permissive to be when filtering out variants for the UI.\n",
    "\n",
    "The examples below are taken from the Ella demo data in `/ella/src/vardb/testdata/filterconfigs.json`. They are included here for the sake of demonstration.\n",
    "\n",
    "Mostly what they do is to filter on frequency in a population. If a SNP is very common it is unlikely to be pathogenic, and along with that to make exceptions for anything that gets a ACGM criteria of N or higher. \n",
    "\n",
    "## Permissive Filter Configs\n",
    "\n",
    "In order to view **absolutely everything** you'll need to pass in an empty list to the filter config.\n",
    "\n",
    "For small gene panels studies this is probably the easiest way to ensure you are seeing all data. For whole exome /genome studies you can pick and choose from the various filtering criteria shown.\n",
    "\n",
    "These should also include usergroups that exist in your database, or else inserting them will cause an error.\n",
    "\n",
    "```json\n",
    "[{\n",
    "    \"name\": \"Default\",\n",
    "    \"usergroups\": [\n",
    "        {\n",
    "            \"name\": \"testgroup01\",\n",
    "            \"order\": 0\n",
    "        },\n",
    "        {\n",
    "            \"name\": \"testgroup03\",\n",
    "            \"order\": 0\n",
    "        }\n",
    "    ],\n",
    "    \"requirements\": [],\n",
    "    \"filterconfig\": {\n",
    "        \"filters\": []\n",
    "    }\n",
    "}]\n",
    "```\n",
    "\n",
    "## Restrictive Filter Configs\n",
    "\n",
    "### Default Config\n",
    "\n",
    "```json\n",
    "[    {\n",
    "        \"name\": \"Default\",\n",
    "        \"usergroups\": [\n",
    "            {\n",
    "                \"name\": \"testgroup01\",\n",
    "                \"order\": 0\n",
    "            },\n",
    "            {\n",
    "                \"name\": \"testgroup03\",\n",
    "                \"order\": 0\n",
    "            }\n",
    "        ],\n",
    "        \"requirements\": [\n",
    "            {\n",
    "                \"function\": \"analysis\",\n",
    "                \"params\": {\n",
    "                    \"name\": \"^(?!HG002|NA12878).*\"\n",
    "                }\n",
    "            }\n",
    "        ],\n",
    "        \"filterconfig\": {\n",
    "            \"filters\": [\n",
    "                {\n",
    "                    \"name\": \"frequency\",\n",
    "                    \"config\": {\n",
    "                        \"groups\": {\n",
    "                            \"external\": {\n",
    "                                \"GNOMAD_GENOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ],\n",
    "                                \"GNOMAD_EXOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ]\n",
    "                            },\n",
    "                            \"internal\": {\n",
    "                                \"inDB\": [\"OUSWES\"]\n",
    "                            }\n",
    "                        },\n",
    "                        \"num_thresholds\": {\n",
    "                            \"GNOMAD_GENOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            },\n",
    "                            \"GNOMAD_EXOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            }\n",
    "                        },\n",
    "                        \"thresholds\": {\n",
    "                            \"AD\": {\n",
    "                                \"external\": 0.005,\n",
    "                                \"internal\": 0.05\n",
    "                            },\n",
    "                            \"default\": {\n",
    "                                \"external\": 0.01,\n",
    "                                \"internal\": 0.05\n",
    "                            }\n",
    "                        }\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"region\",\n",
    "                    \"config\": {\n",
    "                        \"splice_region\": [-20, 6],\n",
    "                        \"utr_region\": [0, 0]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"consequence\",\n",
    "                            \"config\": {\n",
    "                                \"consequences\": [\n",
    "                                    \"transcript_ablation\",\n",
    "                                    \"splice_donor_variant\",\n",
    "                                    \"splice_acceptor_variant\",\n",
    "                                    \"stop_gained\",\n",
    "                                    \"frameshift_variant\",\n",
    "                                    \"start_lost\",\n",
    "                                    \"initiator_codon_variant\",\n",
    "                                    \"stop_lost\",\n",
    "                                    \"inframe_insertion\",\n",
    "                                    \"inframe_deletion\",\n",
    "                                    \"missense_variant\",\n",
    "                                    \"protein_altering_variant\",\n",
    "                                    \"transcript_amplification\",\n",
    "                                    \"incomplete_terminal_codon_variant\",\n",
    "                                    \"synonymous_variant\",\n",
    "                                    \"start_retained_variant\",\n",
    "                                    \"stop_retained_variant\",\n",
    "                                    \"coding_sequence_variant\"\n",
    "                                ]\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                }\n",
    "            ]\n",
    "        }\n",
    "    }]\n",
    "```\n",
    "\n",
    "### Single Default Config\n",
    "    \n",
    "\n",
    "```json\n",
    "    [{\n",
    "        \"name\": \"SingleDefault\",\n",
    "        \"usergroups\": [\n",
    "            {\n",
    "                \"name\": \"testgroup02\",\n",
    "                \"order\": 100\n",
    "            },\n",
    "            {\n",
    "                \"name\": \"testgroup03\",\n",
    "                \"order\": 1100\n",
    "            }\n",
    "        ],\n",
    "        \"requirements\": [\n",
    "            {\n",
    "                \"function\": \"analysis\",\n",
    "                \"params\": {\n",
    "                    \"name\": \"^(?!brca).*\"\n",
    "                }\n",
    "            }\n",
    "        ],\n",
    "        \"filterconfig\": {\n",
    "            \"filters\": [\n",
    "                {\n",
    "                    \"name\": \"frequency\",\n",
    "                    \"config\": {\n",
    "                        \"groups\": {\n",
    "                            \"external\": {\n",
    "                                \"GNOMAD_GENOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ],\n",
    "                                \"GNOMAD_EXOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ]\n",
    "                            },\n",
    "                            \"internal\": {\n",
    "                                \"inDB\": [\"OUSWES\"]\n",
    "                            }\n",
    "                        },\n",
    "                        \"num_thresholds\": {\n",
    "                            \"GNOMAD_GENOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            },\n",
    "                            \"GNOMAD_EXOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            }\n",
    "                        },\n",
    "                        \"thresholds\": {\n",
    "                            \"AD\": {\n",
    "                                \"external\": 0.005,\n",
    "                                \"internal\": 0.05\n",
    "                            },\n",
    "                            \"default\": {\n",
    "                                \"external\": 0.01,\n",
    "                                \"internal\": 0.05\n",
    "                            }\n",
    "                        }\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"region\",\n",
    "                    \"config\": {\n",
    "                        \"splice_region\": [-20, 6],\n",
    "                        \"utr_region\": [0, 0]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"consequence\",\n",
    "                            \"config\": {\n",
    "                                \"genepanel_only\": true,\n",
    "                                \"consequences\": [\n",
    "                                    \"transcript_ablation\",\n",
    "                                    \"splice_donor_variant\",\n",
    "                                    \"splice_acceptor_variant\",\n",
    "                                    \"stop_gained\",\n",
    "                                    \"frameshift_variant\",\n",
    "                                    \"start_lost\",\n",
    "                                    \"initiator_codon_variant\",\n",
    "                                    \"stop_lost\",\n",
    "                                    \"inframe_insertion\",\n",
    "                                    \"inframe_deletion\",\n",
    "                                    \"missense_variant\",\n",
    "                                    \"protein_altering_variant\",\n",
    "                                    \"transcript_amplification\",\n",
    "                                    \"incomplete_terminal_codon_variant\",\n",
    "                                    \"synonymous_variant\",\n",
    "                                    \"start_retained_variant\",\n",
    "                                    \"stop_retained_variant\",\n",
    "                                    \"coding_sequence_variant\"\n",
    "                                ]\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"consequence\",\n",
    "                    \"config\": {\n",
    "                        \"consequences\": [\n",
    "                            \"synonymous_variant\",\n",
    "                            \"stop_retained_variant\",\n",
    "                            \"start_retained_variant\"\n",
    "                        ]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"consequence\",\n",
    "                            \"config\": {\n",
    "                                \"consequences\": [\n",
    "                                    \"transcript_ablation\",\n",
    "                                    \"splice_donor_variant\",\n",
    "                                    \"splice_acceptor_variant\",\n",
    "                                    \"stop_gained\",\n",
    "                                    \"frameshift_variant\",\n",
    "                                    \"start_lost\",\n",
    "                                    \"initiator_codon_variant\",\n",
    "                                    \"stop_lost\",\n",
    "                                    \"inframe_insertion\",\n",
    "                                    \"inframe_deletion\",\n",
    "                                    \"missense_variant\",\n",
    "                                    \"protein_altering_variant\",\n",
    "                                    \"transcript_amplification\",\n",
    "                                    \"splice_region_variant\",\n",
    "                                    \"incomplete_terminal_codon_variant\",\n",
    "                                    \"coding_sequence_variant\",\n",
    "                                    \"mature_miRNA_variant\",\n",
    "                                    \"5_prime_UTR_variant\",\n",
    "                                    \"3_prime_UTR_variant\",\n",
    "                                    \"intron_variant\",\n",
    "                                    \"NMD_transcript_variant\"\n",
    "                                ]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"ppy\",\n",
    "                    \"config\": {\n",
    "                        \"ppy_tract_region\": [-20, -3]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"quality\",\n",
    "                    \"config\": {\n",
    "                        \"allele_ratio\": 0.25\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                }\n",
    "            ]\n",
    "        }\n",
    "    }]\n",
    "```\n",
    "\n",
    "### Trio Default Config\n",
    "\n",
    "```json\n",
    "    [{\n",
    "        \"name\": \"TrioDefault\",\n",
    "        \"usergroups\": [\n",
    "            {\n",
    "                \"name\": \"testgroup02\",\n",
    "                \"order\": 0\n",
    "            },\n",
    "            {\n",
    "                \"name\": \"testgroup03\",\n",
    "                \"order\": 1000\n",
    "            }\n",
    "        ],\n",
    "        \"requirements\": [\n",
    "            {\n",
    "                \"function\": \"analysis\",\n",
    "                \"params\": {\n",
    "                    \"is_trio\": true,\n",
    "                    \"name\": \"^(?!brca).*\"\n",
    "                }\n",
    "            }\n",
    "        ],\n",
    "        \"filterconfig\": {\n",
    "            \"filters\": [\n",
    "                {\n",
    "                    \"name\": \"frequency\",\n",
    "                    \"config\": {\n",
    "                        \"groups\": {\n",
    "                            \"external\": {\n",
    "                                \"GNOMAD_GENOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ],\n",
    "                                \"GNOMAD_EXOMES\": [\n",
    "                                    \"G\",\n",
    "                                    \"AFR\",\n",
    "                                    \"AMR\",\n",
    "                                    \"EAS\",\n",
    "                                    \"FIN\",\n",
    "                                    \"NFE\",\n",
    "                                    \"OTH\",\n",
    "                                    \"SAS\"\n",
    "                                ]\n",
    "                            },\n",
    "                            \"internal\": {\n",
    "                                \"inDB\": [\"OUSWES\"]\n",
    "                            }\n",
    "                        },\n",
    "                        \"num_thresholds\": {\n",
    "                            \"GNOMAD_GENOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            },\n",
    "                            \"GNOMAD_EXOMES\": {\n",
    "                                \"G\": 5000,\n",
    "                                \"AFR\": 5000,\n",
    "                                \"AMR\": 5000,\n",
    "                                \"EAS\": 5000,\n",
    "                                \"FIN\": 5000,\n",
    "                                \"NFE\": 5000,\n",
    "                                \"OTH\": 5000,\n",
    "                                \"SAS\": 5000\n",
    "                            }\n",
    "                        },\n",
    "                        \"thresholds\": {\n",
    "                            \"AD\": {\n",
    "                                \"external\": 0.005,\n",
    "                                \"internal\": 0.05\n",
    "                            },\n",
    "                            \"default\": {\n",
    "                                \"external\": 0.01,\n",
    "                                \"internal\": 0.05\n",
    "                            }\n",
    "                        }\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"region\",\n",
    "                    \"config\": {\n",
    "                        \"splice_region\": [-20, 6],\n",
    "                        \"utr_region\": [0, 0]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"consequence\",\n",
    "                            \"config\": {\n",
    "                                \"genepanel_only\": true,\n",
    "                                \"consequences\": [\n",
    "                                    \"transcript_ablation\",\n",
    "                                    \"splice_donor_variant\",\n",
    "                                    \"splice_acceptor_variant\",\n",
    "                                    \"stop_gained\",\n",
    "                                    \"frameshift_variant\",\n",
    "                                    \"start_lost\",\n",
    "                                    \"initiator_codon_variant\",\n",
    "                                    \"stop_lost\",\n",
    "                                    \"inframe_insertion\",\n",
    "                                    \"inframe_deletion\",\n",
    "                                    \"missense_variant\",\n",
    "                                    \"protein_altering_variant\",\n",
    "                                    \"transcript_amplification\",\n",
    "                                    \"incomplete_terminal_codon_variant\",\n",
    "                                    \"synonymous_variant\",\n",
    "                                    \"stop_retained_variant\",\n",
    "                                    \"coding_sequence_variant\"\n",
    "                                ]\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"consequence\",\n",
    "                    \"config\": {\n",
    "                        \"consequences\": [\n",
    "                            \"synonymous_variant\",\n",
    "                            \"stop_retained_variant\",\n",
    "                            \"start_retained_variant\"\n",
    "                        ]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"consequence\",\n",
    "                            \"config\": {\n",
    "                                \"consequences\": [\n",
    "                                    \"transcript_ablation\",\n",
    "                                    \"splice_donor_variant\",\n",
    "                                    \"splice_acceptor_variant\",\n",
    "                                    \"stop_gained\",\n",
    "                                    \"frameshift_variant\",\n",
    "                                    \"start_lost\",\n",
    "                                    \"initiator_codon_variant\",\n",
    "                                    \"stop_lost\",\n",
    "                                    \"inframe_insertion\",\n",
    "                                    \"inframe_deletion\",\n",
    "                                    \"missense_variant\",\n",
    "                                    \"protein_altering_variant\",\n",
    "                                    \"transcript_amplification\",\n",
    "                                    \"splice_region_variant\",\n",
    "                                    \"incomplete_terminal_codon_variant\",\n",
    "                                    \"coding_sequence_variant\",\n",
    "                                    \"mature_miRNA_variant\",\n",
    "                                    \"5_prime_UTR_variant\",\n",
    "                                    \"3_prime_UTR_variant\",\n",
    "                                    \"intron_variant\",\n",
    "                                    \"NMD_transcript_variant\"\n",
    "                                ]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"ppy\",\n",
    "                    \"config\": {\n",
    "                        \"ppy_tract_region\": [-20, -3]\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"quality\",\n",
    "                    \"config\": {\n",
    "                        \"allele_ratio\": 0.25\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"classification\",\n",
    "                            \"config\": {\n",
    "                                \"classes\": [\"3\", \"4\", \"5\"]\n",
    "                            }\n",
    "                        },\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                },\n",
    "                {\n",
    "                    \"name\": \"segregation\",\n",
    "                    \"config\": {\n",
    "                        \"no_coverage_parents\": { \"enable\": true },\n",
    "                        \"denovo\": {\n",
    "                            \"enable\": true,\n",
    "                            \"gq_threshold\": {\n",
    "                                \"proband\": 20,\n",
    "                                \"mother\": 20,\n",
    "                                \"father\": 20\n",
    "                            }\n",
    "                        },\n",
    "                        \"parental_mosaicism\": { \"enable\": true },\n",
    "                        \"compound_heterozygous\": { \"enable\": true },\n",
    "                        \"recessive_homozygous\": { \"enable\": true }\n",
    "                    },\n",
    "                    \"exceptions\": [\n",
    "                        {\n",
    "                            \"name\": \"external\",\n",
    "                            \"config\": {\n",
    "                                \"clinvar\": {\n",
    "                                    \"num_stars\": [\">=\", 2],\n",
    "                                    \"combinations\": [[\"pathogenic\", \">\", \"benign\"]]\n",
    "                                }\n",
    "                            }\n",
    "                        }\n",
    "                    ]\n",
    "                }\n",
    "            ]\n",
    "        }\n",
    "    }\n",
    "]\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
