--- ella/src/vardb/util/db.py	2020-09-16 10:56:40.000000000 +0300
+++ ella/src/vardb/util/db.py	2020-11-22 10:34:21.000000000 +0300
@@ -22,6 +22,13 @@
 
         if not engine_kwargs:
             engine_kwargs = dict()
+            engine_kwargs['echo'] = True
+            engine_kwargs['echo_pool'] = 'debug'
+        else:
+            # TODO This should really be done based on env vars
+            if 'echo' not in engine_kwargs:
+                engine_kwargs['echo'] = True
+                engine_kwargs['echo_pool'] = 'debug'
 
         self.engine = create_engine(self.host, client_encoding="utf8", **engine_kwargs)
 
