{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sqlalchemy as sa\n",
    "from sqlalchemy import create_engine\n",
    "from sqlalchemy import inspect\n",
    "import json\n",
    "from pprint import pprint\n",
    "import pandas as pd\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'postgresql://postgres:password123@postgresql/postgres'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.environ.get('DB_URL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = sa.create_engine(os.environ.get('DB_URL'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext sql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%sql postgresql://postgres:password123@postgresql/postgres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['annotation',\n",
      " 'gene',\n",
      " 'allele',\n",
      " 'transcript',\n",
      " 'phenotype',\n",
      " 'usergroup',\n",
      " 'genepanel_transcript',\n",
      " 'usergroupgenepanel',\n",
      " 'user',\n",
      " 'customannotation',\n",
      " 'usersession',\n",
      " 'useroldpassword',\n",
      " 'attachment',\n",
      " 'genotype',\n",
      " 'alleleinterpretation',\n",
      " 'reference',\n",
      " 'interpretationstatehistory',\n",
      " 'resourcelog',\n",
      " 'alleleassessmentattachment',\n",
      " 'alleleassessmentreferenceassessment',\n",
      " 'alembic_version',\n",
      " 'analysisinterpretation',\n",
      " 'annotationjob',\n",
      " 'genepanel',\n",
      " 'genepanel_phenotype',\n",
      " 'sample',\n",
      " 'analysis',\n",
      " 'broadcast',\n",
      " 'genotypesampledata',\n",
      " 'clilog',\n",
      " 'usergroupfilterconfig',\n",
      " 'jsonschema',\n",
      " 'filterconfig',\n",
      " 'alleleinterpretationsnapshot',\n",
      " 'analysisinterpretationsnapshot',\n",
      " 'interpretationlog',\n",
      " 'alleleassessment',\n",
      " 'allelereport',\n",
      " 'referenceassessment',\n",
      " 'geneassessment',\n",
      " 'annotationshadowfrequency',\n",
      " 'annotationshadowtranscript']\n"
     ]
    }
   ],
   "source": [
    "# Get column information\n",
    "def get_column_info(tables):\n",
    "    for table in tables:\n",
    "        print(table)\n",
    "        columns = inspector.get_columns(table)\n",
    "        pprint(columns)\n",
    "\n",
    "db_uri = os.environ.get('DB_URL')\n",
    "engine = create_engine(db_uri)\n",
    "\n",
    "inspector = inspect(engine)\n",
    "\n",
    "# Get table information\n",
    "tables = inspector.get_table_names()\n",
    "pprint(tables)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Report Generation\n",
    "\n",
    "Most of the time a report is generated for a particular sample. \n",
    "\n",
    "Sample Id -> Analysis Id -> List(Interpretation Ids) -> List(Allele Id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "7 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>identifier</th>\n",
       "        <th>analysis_id</th>\n",
       "        <th>sample_type</th>\n",
       "        <th>date_deposited</th>\n",
       "        <th>affected</th>\n",
       "        <th>family_id</th>\n",
       "        <th>father_id</th>\n",
       "        <th>mother_id</th>\n",
       "        <th>sibling_id</th>\n",
       "        <th>proband</th>\n",
       "        <th>sex</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>1</td>\n",
       "        <td>brca_allfiltered</td>\n",
       "        <td>1</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:29:42.471284+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>2</td>\n",
       "        <td>brca_sample_1</td>\n",
       "        <td>2</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:29:46.227075+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>4</td>\n",
       "        <td>brca_sample_3</td>\n",
       "        <td>5</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:30:02.294520+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>5</td>\n",
       "        <td>brca_sample_1</td>\n",
       "        <td>6</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:30:05.715108+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>6</td>\n",
       "        <td>brca_sample_master</td>\n",
       "        <td>7</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:30:09.451317+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>7</td>\n",
       "        <td>brca_sample_2</td>\n",
       "        <td>8</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:30:12.751708+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>8</td>\n",
       "        <td>brca_sample_master</td>\n",
       "        <td>9</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 07:30:16.355499+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(1, 'brca_allfiltered', 1, 'HTS', datetime.datetime(2020, 10, 19, 7, 29, 42, 471284, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (2, 'brca_sample_1', 2, 'HTS', datetime.datetime(2020, 10, 19, 7, 29, 46, 227075, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (4, 'brca_sample_3', 5, 'HTS', datetime.datetime(2020, 10, 19, 7, 30, 2, 294520, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (5, 'brca_sample_1', 6, 'HTS', datetime.datetime(2020, 10, 19, 7, 30, 5, 715108, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (6, 'brca_sample_master', 7, 'HTS', datetime.datetime(2020, 10, 19, 7, 30, 9, 451317, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (7, 'brca_sample_2', 8, 'HTS', datetime.datetime(2020, 10, 19, 7, 30, 12, 751708, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (8, 'brca_sample_master', 9, 'HTS', datetime.datetime(2020, 10, 19, 7, 30, 16, 355499, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None)]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%sql\n",
    "\n",
    "SELECT * FROM sample;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis brca_sample_1.HBOCUTV_v01\n",
    "\n",
    "Since any sample can duplicate records with more than 1 analysis, make sure you are getting the correct analysis and the correct. On the UI there needs to be a filter by sample_name, sample_type, date_deposited, genepanel name, and/or genepanel version.\n",
    "\n",
    "Generally the Sample Name is going to be read in from a LIMs system somewhere anyways."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure to use single quotes, '', not \"\"\n",
    "\n",
    "report_data = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>identifier</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_deposited</th>\n",
       "      <th>affected</th>\n",
       "      <th>family_id</th>\n",
       "      <th>father_id</th>\n",
       "      <th>mother_id</th>\n",
       "      <th>sibling_id</th>\n",
       "      <th>proband</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>brca_sample_1</td>\n",
       "      <td>2</td>\n",
       "      <td>HTS</td>\n",
       "      <td>2020-10-19 07:29:46.227075+00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id     identifier  analysis_id sample_type  \\\n",
       "0   2  brca_sample_1            2         HTS   \n",
       "\n",
       "                    date_deposited  affected family_id father_id mother_id  \\\n",
       "0 2020-10-19 07:29:46.227075+00:00      True      None      None      None   \n",
       "\n",
       "  sibling_id  proband   sex  \n",
       "0       None     True  None  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample = %sql select * from sample where analysis_id = 2 AND identifier = 'brca_sample_1';\n",
    "sample_df = sample.DataFrame()\n",
    "sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['sample'] = sample_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>warnings</th>\n",
       "      <th>report</th>\n",
       "      <th>date_deposited</th>\n",
       "      <th>properties</th>\n",
       "      <th>date_requested</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>brca_sample_1.HBOCUTV_v01</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.204898+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id                       name genepanel_name genepanel_version warnings  \\\n",
       "0   2  brca_sample_1.HBOCUTV_v01        HBOCUTV               v01     None   \n",
       "\n",
       "  report                   date_deposited properties date_requested  \n",
       "0   None 2020-10-19 07:29:46.204898+00:00       None           None  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analysis = %sql SELECT * FROM analysis where name = 'brca_sample_1.HBOCUTV_v01';\n",
    "analysis_df = analysis.DataFrame()\n",
    "analysis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['analysis'] = analysis_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>user_state</th>\n",
       "      <th>state</th>\n",
       "      <th>status</th>\n",
       "      <th>date_last_update</th>\n",
       "      <th>date_created</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>user_id</th>\n",
       "      <th>finalized</th>\n",
       "      <th>workflow_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>{'allele': {'3': {'sections': {}, 'allele_id':...</td>\n",
       "      <td>{'allele': {'3': {'report': {'included': True}...</td>\n",
       "      <td>Ongoing</td>\n",
       "      <td>2020-10-19 08:01:33.706453+00:00</td>\n",
       "      <td>2020-10-19 07:29:46.217740+00:00</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>Interpretation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id genepanel_name genepanel_version  \\\n",
       "0   2        HBOCUTV               v01   \n",
       "\n",
       "                                          user_state  \\\n",
       "0  {'allele': {'3': {'sections': {}, 'allele_id':...   \n",
       "\n",
       "                                               state   status  \\\n",
       "0  {'allele': {'3': {'report': {'included': True}...  Ongoing   \n",
       "\n",
       "                  date_last_update                     date_created  \\\n",
       "0 2020-10-19 08:01:33.706453+00:00 2020-10-19 07:29:46.217740+00:00   \n",
       "\n",
       "   analysis_id  user_id finalized workflow_status  \n",
       "0            2        1      None  Interpretation  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# At some point I will want to filter on analyses that are complete\n",
    "\n",
    "analysis_interpretation = %sql select * from analysisinterpretation where analysis_id = 2;\n",
    "analysis_interprtetation_df = analysis_interpretation.DataFrame()\n",
    "analysis_interprtetation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['analysis_interprtetation'] = analysis_interprtetation_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "6 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>classification</th>\n",
       "      <th>evaluation</th>\n",
       "      <th>user_id</th>\n",
       "      <th>date_created</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>previous_assessment_id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>annotation_id</th>\n",
       "      <th>custom_annotation_id</th>\n",
       "      <th>usergroup_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:43:59.585285+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:44:15.013829+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:44:30.335239+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>U</td>\n",
       "      <td>{'acmg': {'included': [], 'suggested': [{'op':...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:44:43.334923+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:44:58.945348+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>U</td>\n",
       "      <td>{'acmg': {'included': [], 'suggested': [{'op':...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 07:45:11.324654+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>8</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id classification                                         evaluation  \\\n",
       "0   1              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "1   2              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "2   3              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "3   4              U  {'acmg': {'included': [], 'suggested': [{'op':...   \n",
       "4   5              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "5   6              U  {'acmg': {'included': [], 'suggested': [{'op':...   \n",
       "\n",
       "   user_id                     date_created date_superceeded  \\\n",
       "0        1 2020-10-19 07:43:59.585285+00:00             None   \n",
       "1        1 2020-10-19 07:44:15.013829+00:00             None   \n",
       "2        1 2020-10-19 07:44:30.335239+00:00             None   \n",
       "3        1 2020-10-19 07:44:43.334923+00:00             None   \n",
       "4        1 2020-10-19 07:44:58.945348+00:00             None   \n",
       "5        1 2020-10-19 07:45:11.324654+00:00             None   \n",
       "\n",
       "  previous_assessment_id  allele_id genepanel_name genepanel_version  \\\n",
       "0                   None          3        HBOCUTV               v01   \n",
       "1                   None          4        HBOCUTV               v01   \n",
       "2                   None          5        HBOCUTV               v01   \n",
       "3                   None          6        HBOCUTV               v01   \n",
       "4                   None          7        HBOCUTV               v01   \n",
       "5                   None          8        HBOCUTV               v01   \n",
       "\n",
       "   analysis_id  annotation_id custom_annotation_id  usergroup_id  \n",
       "0            2              3                 None             1  \n",
       "1            2              4                 None             1  \n",
       "2            2              5                 None             1  \n",
       "3            2              6                 None             1  \n",
       "4            2              7                 None             1  \n",
       "5            2              8                 None             1  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_assessment = %sql select * from alleleassessment where analysis_id = 2;\n",
    "allele_assessment_df = allele_assessment.DataFrame()\n",
    "\n",
    "report_data['allele_assessment'] = allele_assessment_df.to_dict('records')\n",
    "\n",
    "allele_assessment_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>annotations</th>\n",
       "      <th>previous_annotation_id</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>date_created</th>\n",
       "      <th>schema_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 4...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327638+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 1...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327657+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'acc_num':...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327665+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 2...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327672+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 3...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327678+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM?', 'codon': ...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 07:29:46.327685+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  allele_id                                        annotations  \\\n",
       "0   3          3  {'external': {'HGMD': {'tag': 'DM', 'codon': 4...   \n",
       "1   4          4  {'external': {'HGMD': {'tag': 'DM', 'codon': 1...   \n",
       "2   5          5  {'external': {'HGMD': {'tag': 'DM', 'acc_num':...   \n",
       "3   6          6  {'external': {'HGMD': {'tag': 'DM', 'codon': 2...   \n",
       "4   7          7  {'external': {'HGMD': {'tag': 'DM', 'codon': 3...   \n",
       "5   8          8  {'external': {'HGMD': {'tag': 'DM?', 'codon': ...   \n",
       "\n",
       "  previous_annotation_id date_superceeded                     date_created  \\\n",
       "0                   None             None 2020-10-19 07:29:46.327638+00:00   \n",
       "1                   None             None 2020-10-19 07:29:46.327657+00:00   \n",
       "2                   None             None 2020-10-19 07:29:46.327665+00:00   \n",
       "3                   None             None 2020-10-19 07:29:46.327672+00:00   \n",
       "4                   None             None 2020-10-19 07:29:46.327678+00:00   \n",
       "5                   None             None 2020-10-19 07:29:46.327685+00:00   \n",
       "\n",
       "   schema_version  \n",
       "0               1  \n",
       "1               1  \n",
       "2               1  \n",
       "3               1  \n",
       "4               1  \n",
       "5               1  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now let's grab the allele ids\n",
    "allele_ids = allele_assessment_df['allele_id'].tolist() \n",
    "query = sa.text(\"\"\" \n",
    "                SELECT *\n",
    "                FROM annotation\n",
    "                WHERE allele_id IN :values; \n",
    "\"\"\")\n",
    "query = query.bindparams(values=tuple(allele_ids))\n",
    "allele_annotation_df = pd.read_sql(query, engine)\n",
    "allele_annotation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['allele_annotation'] = allele_annotation_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample Analysis DataSet\n",
    "\n",
    "The dataset from a sample analysis should look like:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['sample', 'analysis', 'analysis_interprtetation', 'allele_assessment', 'allele_annotation'])\n"
     ]
    }
   ],
   "source": [
    "pprint(report_data.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'sample': [{'id': 2,\n",
       "   'identifier': 'brca_sample_1',\n",
       "   'analysis_id': 2,\n",
       "   'sample_type': 'HTS',\n",
       "   'date_deposited': Timestamp('2020-10-19 07:29:46.227075+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'affected': True,\n",
       "   'family_id': None,\n",
       "   'father_id': None,\n",
       "   'mother_id': None,\n",
       "   'sibling_id': None,\n",
       "   'proband': True,\n",
       "   'sex': None}],\n",
       " 'analysis': [{'id': 2,\n",
       "   'name': 'brca_sample_1.HBOCUTV_v01',\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'warnings': None,\n",
       "   'report': None,\n",
       "   'date_deposited': Timestamp('2020-10-19 07:29:46.204898+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'properties': None,\n",
       "   'date_requested': None}],\n",
       " 'analysis_interprtetation': [{'id': 2,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'user_state': {'allele': {'3': {'sections': {},\n",
       "      'allele_id': 3,\n",
       "      'showExcludedReferences': False},\n",
       "     '4': {'sections': {}, 'allele_id': 4, 'showExcludedReferences': False},\n",
       "     '5': {'sections': {}, 'allele_id': 5, 'showExcludedReferences': False},\n",
       "     '6': {'sections': {}, 'allele_id': 6, 'showExcludedReferences': False},\n",
       "     '7': {'sections': {}, 'allele_id': 7, 'showExcludedReferences': False},\n",
       "     '8': {'sections': {}, 'allele_id': 8, 'showExcludedReferences': False}}},\n",
       "   'state': {'allele': {'3': {'report': {'included': True},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 3,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 1},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 3, 'reuseCheckedId': 1},\n",
       "      'referenceassessments': []},\n",
       "     '4': {'report': {'included': True},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 4,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 2},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 4, 'reuseCheckedId': 2},\n",
       "      'referenceassessments': []},\n",
       "     '5': {'report': {'included': True},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 5,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 3},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 5, 'reuseCheckedId': 3},\n",
       "      'referenceassessments': []},\n",
       "     '6': {'report': {'included': False},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 6,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 4},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 6, 'reuseCheckedId': 4},\n",
       "      'referenceassessments': []},\n",
       "     '7': {'report': {'included': True},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 7,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 5},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 7, 'reuseCheckedId': 5},\n",
       "      'referenceassessments': []},\n",
       "     '8': {'report': {'included': False},\n",
       "      'analysis': {'comment': '',\n",
       "       'notrelevant': None,\n",
       "       'verification': 'verified'},\n",
       "      'workflow': {'reviewed': True},\n",
       "      'allele_id': 8,\n",
       "      'allelereport': {'evaluation': {'comment': ''}, 'copiedFromId': 6},\n",
       "      'alleleassessment': {'reuse': True, 'allele_id': 8, 'reuseCheckedId': 6},\n",
       "      'referenceassessments': []}},\n",
       "    'filterconfigId': 1,\n",
       "    'manuallyAddedAlleles': []},\n",
       "   'status': 'Ongoing',\n",
       "   'date_last_update': Timestamp('2020-10-19 08:01:33.706453+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.217740+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'analysis_id': 2,\n",
       "   'user_id': 1,\n",
       "   'finalized': None,\n",
       "   'workflow_status': 'Interpretation'}],\n",
       " 'allele_assessment': [{'id': 1,\n",
       "   'classification': '4',\n",
       "   'evaluation': {'acmg': {'included': [{'op': None,\n",
       "       'code': 'PVS1',\n",
       "       'uuid': '7ad2d394-73bb-4df6-a878-2948ca839f36',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''},\n",
       "      {'op': None,\n",
       "       'code': 'PPxPM2',\n",
       "       'uuid': '49d9a1c5-3a8d-4c5f-b106-65317f021d3e',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''}],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_null_variant',\n",
       "       'match': ['stop_gained'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_no_freq',\n",
       "       'match': ['null_freq'],\n",
       "       'source': 'frequencies.commonness'},\n",
       "      {'op': None, 'code': 'PVS1', 'match': None, 'source': None},\n",
       "      {'op': None, 'code': 'PPxPM2', 'match': None, 'source': None}],\n",
       "     'suggested_classification': 4},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:43:59.585285+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 3,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 3,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1},\n",
       "  {'id': 2,\n",
       "   'classification': '4',\n",
       "   'evaluation': {'acmg': {'included': [{'op': None,\n",
       "       'code': 'PVS1',\n",
       "       'uuid': '3dfbb5e1-9cf5-44b7-9c9d-baf91ca8341d',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''},\n",
       "      {'op': None,\n",
       "       'code': 'PPxPM2',\n",
       "       'uuid': '9f74e9e3-33ea-4f24-9be8-ea4120135a36',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''}],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_null_variant',\n",
       "       'match': ['frameshift_variant'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_no_freq',\n",
       "       'match': ['null_freq'],\n",
       "       'source': 'frequencies.commonness'},\n",
       "      {'op': None, 'code': 'PVS1', 'match': None, 'source': None},\n",
       "      {'op': None, 'code': 'PPxPM2', 'match': None, 'source': None}],\n",
       "     'suggested_classification': 4},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:44:15.013829+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 4,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 4,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1},\n",
       "  {'id': 3,\n",
       "   'classification': '4',\n",
       "   'evaluation': {'acmg': {'included': [{'op': None,\n",
       "       'code': 'PVS1',\n",
       "       'uuid': '9517fec5-6196-4ae8-bb0a-1fd28c1cade0',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''},\n",
       "      {'op': None,\n",
       "       'code': 'PPxPM2',\n",
       "       'uuid': '28ca699c-cddd-4289-9cf0-25a16881da79',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''}],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_null_variant',\n",
       "       'match': ['splice_donor_variant'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_no_freq',\n",
       "       'match': ['null_freq'],\n",
       "       'source': 'frequencies.commonness'},\n",
       "      {'op': None, 'code': 'PVS1', 'match': None, 'source': None},\n",
       "      {'op': None, 'code': 'PPxPM2', 'match': None, 'source': None}],\n",
       "     'suggested_classification': 4},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:44:30.335239+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 5,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 5,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1},\n",
       "  {'id': 4,\n",
       "   'classification': 'U',\n",
       "   'evaluation': {'acmg': {'included': [],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_missense',\n",
       "       'match': ['missense_variant'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'}],\n",
       "     'suggested_classification': None},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:44:43.334923+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 6,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 6,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1},\n",
       "  {'id': 5,\n",
       "   'classification': '4',\n",
       "   'evaluation': {'acmg': {'included': [{'op': None,\n",
       "       'code': 'PVS1',\n",
       "       'uuid': '6621c49f-1554-47fe-b6ad-bce4d444791d',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''},\n",
       "      {'op': None,\n",
       "       'code': 'PPxPM2',\n",
       "       'uuid': 'd420c709-1f7f-4f49-8b68-8e918bbd2095',\n",
       "       'match': None,\n",
       "       'source': 'suggested',\n",
       "       'comment': ''}],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_null_variant',\n",
       "       'match': ['stop_gained'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_no_freq',\n",
       "       'match': ['null_freq'],\n",
       "       'source': 'frequencies.commonness'},\n",
       "      {'op': None, 'code': 'PVS1', 'match': None, 'source': None},\n",
       "      {'op': None, 'code': 'PPxPM2', 'match': None, 'source': None}],\n",
       "     'suggested_classification': 4},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:44:58.945348+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 7,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 7,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1},\n",
       "  {'id': 6,\n",
       "   'classification': 'U',\n",
       "   'evaluation': {'acmg': {'included': [],\n",
       "     'suggested': [{'op': '$in',\n",
       "       'code': 'REQ_GP_last_exon_not_important',\n",
       "       'match': ['LENI'],\n",
       "       'source': 'genepanel.last_exon_important'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_GP_LOF_missense',\n",
       "       'match': ['ANY'],\n",
       "       'source': 'genepanel.disease_mode'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_less_common',\n",
       "       'match': ['less_common'],\n",
       "       'source': 'frequencies.commonness'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_no_aa_change',\n",
       "       'match': ['synonymous_variant'],\n",
       "       'source': 'transcript.consequences'},\n",
       "      {'op': '$in',\n",
       "       'code': 'REQ_not_in_last_exon',\n",
       "       'match': ['no'],\n",
       "       'source': 'transcript.in_last_exon'}],\n",
       "     'suggested_classification': None},\n",
       "    'external': {'comment': ''},\n",
       "    'frequency': {'comment': ''},\n",
       "    'reference': {'comment': ''},\n",
       "    'prediction': {'comment': ''},\n",
       "    'classification': {'comment': ''}},\n",
       "   'user_id': 1,\n",
       "   'date_created': Timestamp('2020-10-19 07:45:11.324654+0000', tz='psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)'),\n",
       "   'date_superceeded': None,\n",
       "   'previous_assessment_id': None,\n",
       "   'allele_id': 8,\n",
       "   'genepanel_name': 'HBOCUTV',\n",
       "   'genepanel_version': 'v01',\n",
       "   'analysis_id': 2,\n",
       "   'annotation_id': 8,\n",
       "   'custom_annotation_id': None,\n",
       "   'usergroup_id': 1}],\n",
       " 'allele_annotation': [{'id': 3,\n",
       "   'allele_id': 3,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM',\n",
       "      'codon': 4,\n",
       "      'acc_num': 'CM082514',\n",
       "      'disease': 'Breast and/or ovarian cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'SCV000296805',\n",
       "        'submitter': 'GeneKor MSA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '01/11/2017',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000323955',\n",
       "        'submitter': 'ENIGMA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '18/10/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000257718',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '51063',\n",
       "        'last_evaluated': '18/10/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000239090',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'not provided',\n",
       "        'variant_id': '51063',\n",
       "        'last_evaluated': '01/11/2017',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000326495',\n",
       "        'submitter': 'CIMBA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '02/10/2015',\n",
       "        'clinical_significance_descr': 'Pathogenic'}],\n",
       "      'variant_id': 51063,\n",
       "      'variant_description': 'reviewed by expert panel'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD'],\n",
       "      'pubmed_id': 17453335,\n",
       "      'source_info': {'HGMD': 'Primary literature report. No comments.'}}],\n",
       "    'frequencies': {},\n",
       "    'transcripts': [{'dbsnp': ['rs397507571'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000345108',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False},\n",
       "     {'exon': '2/27',\n",
       "      'HGVSc': 'c.10G>T',\n",
       "      'HGVSp': 'p.Gly4Ter',\n",
       "      'dbsnp': ['rs397507571'],\n",
       "      'codons': 'Gga/Tga',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000369497.3',\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.10G>T',\n",
       "      'amino_acids': 'G/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '2/10',\n",
       "      'HGVSc': 'c.-356G>T',\n",
       "      'dbsnp': ['rs397507571'],\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-356G>T',\n",
       "      'consequences': ['5_prime_UTR_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': -356},\n",
       "     {'dbsnp': ['rs397507571'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'exon': '2/28',\n",
       "      'HGVSc': 'c.10G>T',\n",
       "      'HGVSp': 'p.Gly4Ter',\n",
       "      'dbsnp': ['rs397507571'],\n",
       "      'codons': 'Gga/Tga',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000439902.1',\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.10G>T',\n",
       "      'amino_acids': 'G/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '2/27',\n",
       "      'HGVSc': 'c.10G>T',\n",
       "      'HGVSp': 'p.Gly4Ter',\n",
       "      'dbsnp': ['rs397507571'],\n",
       "      'codons': 'Gga/Tga',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'NP_000050.2',\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.10G>T',\n",
       "      'amino_acids': 'G/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'dbsnp': ['rs397507571'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'NM_001136571.1',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327638+0000', tz='UTC'),\n",
       "   'schema_version': 1},\n",
       "  {'id': 4,\n",
       "   'allele_id': 4,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM',\n",
       "      'codon': 16,\n",
       "      'acc_num': 'CD961847',\n",
       "      'disease': 'Breast cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'RCV000131871',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Hereditary cancer-predisposing syndrome',\n",
       "        'variant_id': '51814',\n",
       "        'last_evaluated': '12/12/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000279262',\n",
       "        'submitter': 'GeneDx',\n",
       "        'traitnames': 'Not Provided',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '21/01/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000282398',\n",
       "        'submitter': 'ENIGMA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '22/04/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000113063',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '51814',\n",
       "        'last_evaluated': '22/04/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000694829',\n",
       "        'submitter': 'Integrated Genetics/Laboratory Corporation of America',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '06/01/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000327129',\n",
       "        'submitter': 'CIMBA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '02/10/2015',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000186926',\n",
       "        'submitter': 'Ambry Genetics',\n",
       "        'traitnames': 'Hereditary cancer-predisposing syndrome',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '23/09/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000146071',\n",
       "        'submitter': 'BIC (BRCA2)',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '29/05/2002',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000219019',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'not provided',\n",
       "        'variant_id': '51814',\n",
       "        'last_evaluated': '21/01/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000683670',\n",
       "        'submitter': 'Color Genomics, Inc.',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '12/12/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000591655',\n",
       "        'submitter': 'Department of Pathology and Laboratory Medicine,Sinai Health System',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '17/02/2015',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000587530',\n",
       "        'submitter': \"Research Molecular Genetics Laboratory,Women's College Hospital, University of Toronto\",\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '31/01/2014',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000496563',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Hereditary breast and ovarian cancer syndrome',\n",
       "        'variant_id': '51814',\n",
       "        'last_evaluated': '06/01/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000044604',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Familial cancer of breast',\n",
       "        'variant_id': '51814',\n",
       "        'last_evaluated': '01/07/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000296839',\n",
       "        'submitter': 'GeneKor MSA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '01/07/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'}],\n",
       "      'variant_id': 51814,\n",
       "      'variant_description': 'reviewed by expert panel'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD', 'CLINVAR'],\n",
       "      'pubmed_id': 8589730,\n",
       "      'source_info': {'HGMD': 'Primary literature report. aka 279delAC.'}},\n",
       "     {'sources': ['HGMD', 'CLINVAR'],\n",
       "      'pubmed_id': 18607349,\n",
       "      'source_info': {'HGMD': 'Functional characterisation. No comments.'}},\n",
       "     {'sources': ['HGMD'],\n",
       "      'pubmed_id': 28324225,\n",
       "      'source_info': {'HGMD': 'Additional phenotype. Breast and/or ovarian cancer.'}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 18092194, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 9245992, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 10070953, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 22430266, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 20927582, 'source_info': {}}],\n",
       "    'frequencies': {},\n",
       "    'transcripts': [{'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000345108',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False},\n",
       "     {'exon': '2/27',\n",
       "      'HGVSc': 'c.51_52delAC',\n",
       "      'HGVSp': 'p.Arg18LeufsTer12',\n",
       "      'codons': 'ACa/a',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000369497.3',\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.51_52delAC',\n",
       "      'amino_acids': 'T/X',\n",
       "      'consequences': ['frameshift_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '2/10',\n",
       "      'HGVSc': 'c.-315_-314delAC',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-315_-314delAC',\n",
       "      'consequences': ['5_prime_UTR_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': -314},\n",
       "     {'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'exon': '2/28',\n",
       "      'HGVSc': 'c.51_52delAC',\n",
       "      'HGVSp': 'p.Arg18LeufsTer12',\n",
       "      'codons': 'ACa/a',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000439902.1',\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.51_52delAC',\n",
       "      'amino_acids': 'T/X',\n",
       "      'consequences': ['frameshift_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '2/27',\n",
       "      'HGVSc': 'c.51_52delAC',\n",
       "      'HGVSp': 'p.Arg18LeufsTer12',\n",
       "      'codons': 'ACa/a',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'NP_000050.2',\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.51_52delAC',\n",
       "      'amino_acids': 'T/X',\n",
       "      'consequences': ['frameshift_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'NM_001136571.1',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327657+0000', tz='UTC'),\n",
       "   'schema_version': 1},\n",
       "  {'id': 5,\n",
       "   'allele_id': 5,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM',\n",
       "      'acc_num': 'CS087186',\n",
       "      'disease': 'Breast cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'RCV000113080',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '52162',\n",
       "        'last_evaluated': '24/02/2003',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000146095',\n",
       "        'submitter': 'BIC (BRCA2)',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '24/02/2003',\n",
       "        'clinical_significance_descr': 'Pathogenic'}],\n",
       "      'variant_id': 52162,\n",
       "      'variant_description': 'no assertion criteria provided'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD'],\n",
       "      'pubmed_id': 18092194,\n",
       "      'source_info': {'HGMD': 'Primary literature report. Descr. in Suppl. Table 1 (online).'}},\n",
       "     {'sources': ['HGMD'],\n",
       "      'pubmed_id': 25525159,\n",
       "      'source_info': {'HGMD': 'Additional report. predicted to induce a large splicing change - Table S4.'}}],\n",
       "    'frequencies': {},\n",
       "    'transcripts': [{'dbsnp': ['rs81002885'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000345108',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False},\n",
       "     {'HGVSc': 'c.67+2T>A',\n",
       "      'dbsnp': ['rs81002885'],\n",
       "      'intron': '2/26',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.67+2T>A',\n",
       "      'consequences': ['splice_donor_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 2,\n",
       "      'coding_region_distance': None},\n",
       "     {'HGVSc': 'c.-303+6T>A',\n",
       "      'dbsnp': ['rs81002885'],\n",
       "      'intron': '2/9',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-303+6T>A',\n",
       "      'consequences': ['intron_variant', 'splice_region_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 6,\n",
       "      'coding_region_distance': None},\n",
       "     {'dbsnp': ['rs81002885'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'HGVSc': 'c.67+2T>A',\n",
       "      'dbsnp': ['rs81002885'],\n",
       "      'intron': '2/27',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.67+2T>A',\n",
       "      'consequences': ['splice_donor_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 2,\n",
       "      'coding_region_distance': None},\n",
       "     {'HGVSc': 'c.67+2T>A',\n",
       "      'dbsnp': ['rs81002885'],\n",
       "      'intron': '2/26',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.67+2T>A',\n",
       "      'consequences': ['splice_donor_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 2,\n",
       "      'coding_region_distance': None},\n",
       "     {'dbsnp': ['rs81002885'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'NM_001136571.1',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327665+0000', tz='UTC'),\n",
       "   'schema_version': 1},\n",
       "  {'id': 6,\n",
       "   'allele_id': 6,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM',\n",
       "      'codon': 24,\n",
       "      'acc_num': 'CM111973',\n",
       "      'disease': 'Breast and/or ovarian cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'SCV000678953',\n",
       "        'submitter': 'ClinVar Staff, National Center for Biotechnology Information (NCBI)',\n",
       "        'traitnames': 'Familial cancer of breast',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '',\n",
       "        'clinical_significance_descr': 'not provided'},\n",
       "       {'rcv': 'RCV000577171',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Familial cancer of breast',\n",
       "        'variant_id': '52303',\n",
       "        'last_evaluated': '',\n",
       "        'clinical_significance_descr': 'not provided'}],\n",
       "      'variant_id': 52303,\n",
       "      'variant_description': 'no assertion criteria provided'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD', 'CLINVAR'],\n",
       "      'pubmed_id': 21356067,\n",
       "      'source_info': {'HGMD': 'Primary literature report. No comments.'}},\n",
       "     {'sources': ['HGMD'],\n",
       "      'pubmed_id': 25525159,\n",
       "      'source_info': {'HGMD': 'Additional report. predicted to induce a large splicing change - Table S4.'}}],\n",
       "    'frequencies': {'ExAC': {'hom': {'G': 0,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 0},\n",
       "      'num': {'G': 119160,\n",
       "       'AFR': 9500,\n",
       "       'AMR': 11530,\n",
       "       'EAS': 8550,\n",
       "       'FIN': 6600,\n",
       "       'NFE': 65594,\n",
       "       'OTH': 894,\n",
       "       'SAS': 16492},\n",
       "      'freq': {'G': 1.6784155756965425e-05,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 0.0,\n",
       "       'EAS': 0.0,\n",
       "       'FIN': 0.00030303030303030303,\n",
       "       'NFE': 0.0,\n",
       "       'OTH': 0.0,\n",
       "       'SAS': 0.0},\n",
       "      'count': {'G': 2,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 2,\n",
       "       'Het': 2,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 0}},\n",
       "     'GNOMAD_EXOMES': {'hom': {'G': 0,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 0,\n",
       "       'Male': 0,\n",
       "       'Female': 0},\n",
       "      'num': {'G': 245534,\n",
       "       'AFR': 15200,\n",
       "       'AMR': 33558,\n",
       "       'ASJ': 9828,\n",
       "       'EAS': 17220,\n",
       "       'FIN': 22286,\n",
       "       'NFE': 111296,\n",
       "       'OTH': 5474,\n",
       "       'SAS': 30672,\n",
       "       'Male': 134558,\n",
       "       'Female': 110976},\n",
       "      'freq': {'G': 2.0363778539835625e-05,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 0.0,\n",
       "       'ASJ': 0.0,\n",
       "       'EAS': 0.0,\n",
       "       'FIN': 0.0002243560979987436,\n",
       "       'NFE': 0.0,\n",
       "       'OTH': 0.0,\n",
       "       'SAS': 0.0,\n",
       "       'Male': 7.431739472941036e-06,\n",
       "       'Female': 3.604382929642445e-05},\n",
       "      'count': {'G': 5,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 5,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 0,\n",
       "       'Male': 1,\n",
       "       'Female': 4},\n",
       "      'filter': {'G': ['PASS']}}},\n",
       "    'transcripts': [{'exon': '3/27',\n",
       "      'HGVSc': 'c.72A>T',\n",
       "      'HGVSp': 'p.Leu24Phe',\n",
       "      'dbsnp': ['rs397507909'],\n",
       "      'codons': 'ttA/ttT',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000369497.3',\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.72A>T',\n",
       "      'amino_acids': 'L/F',\n",
       "      'consequences': ['missense_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/10',\n",
       "      'HGVSc': 'c.-298A>T',\n",
       "      'dbsnp': ['rs397507909'],\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-298A>T',\n",
       "      'consequences': ['5_prime_UTR_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': -298},\n",
       "     {'dbsnp': ['rs397507909'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'exon': '3/28',\n",
       "      'HGVSc': 'c.72A>T',\n",
       "      'HGVSp': 'p.Leu24Phe',\n",
       "      'dbsnp': ['rs397507909'],\n",
       "      'codons': 'ttA/ttT',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000439902.1',\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.72A>T',\n",
       "      'amino_acids': 'L/F',\n",
       "      'consequences': ['missense_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/27',\n",
       "      'HGVSc': 'c.72A>T',\n",
       "      'HGVSp': 'p.Leu24Phe',\n",
       "      'dbsnp': ['rs397507909'],\n",
       "      'codons': 'ttA/ttT',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'NP_000050.2',\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.72A>T',\n",
       "      'amino_acids': 'L/F',\n",
       "      'consequences': ['missense_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327672+0000', tz='UTC'),\n",
       "   'schema_version': 1},\n",
       "  {'id': 7,\n",
       "   'allele_id': 7,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM',\n",
       "      'codon': 33,\n",
       "      'acc_num': 'CM067653',\n",
       "      'disease': 'Breast and/or ovarian cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'SCV000324794',\n",
       "        'submitter': 'ENIGMA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '18/10/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'RCV000256555',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '52900',\n",
       "        'last_evaluated': '18/10/2016',\n",
       "        'clinical_significance_descr': 'Pathogenic'},\n",
       "       {'rcv': 'SCV000328170',\n",
       "        'submitter': 'CIMBA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '02/10/2015',\n",
       "        'clinical_significance_descr': 'Pathogenic'}],\n",
       "      'variant_id': 52900,\n",
       "      'variant_description': 'reviewed by expert panel'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD'],\n",
       "      'pubmed_id': 16760289,\n",
       "      'source_info': {'HGMD': 'Primary literature report. No comments.'}},\n",
       "     {'sources': ['HGMD'],\n",
       "      'pubmed_id': 16847550,\n",
       "      'source_info': {'HGMD': 'Additional report. No comments.'}},\n",
       "     {'sources': ['HGMD'],\n",
       "      'pubmed_id': 25525159,\n",
       "      'source_info': {'HGMD': 'Additional report. predicted to induce a large splicing change - Table S4.'}}],\n",
       "    'frequencies': {},\n",
       "    'transcripts': [{'exon': '3/27',\n",
       "      'HGVSc': 'c.97G>T',\n",
       "      'HGVSp': 'p.Glu33Ter',\n",
       "      'dbsnp': ['rs397508065'],\n",
       "      'codons': 'Gaa/Taa',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000369497.3',\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.97G>T',\n",
       "      'amino_acids': 'E/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/10',\n",
       "      'HGVSc': 'c.-273G>T',\n",
       "      'dbsnp': ['rs397508065'],\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-273G>T',\n",
       "      'consequences': ['5_prime_UTR_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': -273},\n",
       "     {'dbsnp': ['rs397508065'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'exon': '3/28',\n",
       "      'HGVSc': 'c.97G>T',\n",
       "      'HGVSp': 'p.Glu33Ter',\n",
       "      'dbsnp': ['rs397508065'],\n",
       "      'codons': 'Gaa/Taa',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENSP00000439902.1',\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.97G>T',\n",
       "      'amino_acids': 'E/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/27',\n",
       "      'HGVSc': 'c.97G>T',\n",
       "      'HGVSp': 'p.Glu33Ter',\n",
       "      'dbsnp': ['rs397508065'],\n",
       "      'codons': 'Gaa/Taa',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'NP_000050.2',\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.97G>T',\n",
       "      'amino_acids': 'E/*',\n",
       "      'consequences': ['stop_gained'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327678+0000', tz='UTC'),\n",
       "   'schema_version': 1},\n",
       "  {'id': 8,\n",
       "   'allele_id': 8,\n",
       "   'annotations': {'external': {'HGMD': {'tag': 'DM?',\n",
       "      'codon': 66,\n",
       "      'acc_num': 'CM173583',\n",
       "      'disease': 'Breast and/or ovarian cancer'},\n",
       "     'CLINVAR': {'items': [{'rcv': 'RCV000377649',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Fanconi anemia',\n",
       "        'variant_id': '51234',\n",
       "        'last_evaluated': '14/06/2016',\n",
       "        'clinical_significance_descr': 'Likely benign'},\n",
       "       {'rcv': 'RCV000162901',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Hereditary cancer-predisposing syndrome',\n",
       "        'variant_id': '51234',\n",
       "        'last_evaluated': '02/04/2015',\n",
       "        'clinical_significance_descr': 'Likely benign'},\n",
       "       {'rcv': 'SCV000383604',\n",
       "        'submitter': 'Illumina',\n",
       "        'traitnames': 'Hereditary Breast and Ovarian Cancer',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '14/06/2016',\n",
       "        'clinical_significance_descr': 'Likely benign'},\n",
       "       {'rcv': 'SCV000578005',\n",
       "        'submitter': 'ENIGMA',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '29/06/2017',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'RCV000499809',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'not specified',\n",
       "        'variant_id': '51234',\n",
       "        'last_evaluated': '15/11/2012',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'SCV000213388',\n",
       "        'submitter': 'Ambry Genetics',\n",
       "        'traitnames': 'Hereditary cancer-predisposing syndrome',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '04/07/2014',\n",
       "        'clinical_significance_descr': 'Likely benign'},\n",
       "       {'rcv': 'RCV000113265',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '51234',\n",
       "        'last_evaluated': '29/06/2017',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'SCV000071946',\n",
       "        'submitter': 'Invitae',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '03/01/2018',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'SCV000146362',\n",
       "        'submitter': 'BIC (BRCA2)',\n",
       "        'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '18/09/2010',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'SCV000591663',\n",
       "        'submitter': 'Department of Pathology and Laboratory Medicine,Sinai Health System',\n",
       "        'traitnames': 'not specified',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '15/11/2012',\n",
       "        'clinical_significance_descr': 'Benign'},\n",
       "       {'rcv': 'SCV000383605',\n",
       "        'submitter': 'Illumina',\n",
       "        'traitnames': 'Fanconi Anemia',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '14/06/2016',\n",
       "        'clinical_significance_descr': 'Likely benign'},\n",
       "       {'rcv': 'RCV000167850',\n",
       "        'submitter': '',\n",
       "        'traitnames': 'Hereditary breast and ovarian cancer syndrome',\n",
       "        'variant_id': '51234',\n",
       "        'last_evaluated': '03/01/2018',\n",
       "        'clinical_significance_descr': 'Benign/Likely benign'},\n",
       "       {'rcv': 'SCV000683467',\n",
       "        'submitter': 'Color Genomics, Inc.',\n",
       "        'traitnames': '',\n",
       "        'variant_id': '',\n",
       "        'last_evaluated': '02/04/2015',\n",
       "        'clinical_significance_descr': 'Likely benign'}],\n",
       "      'variant_id': 51234,\n",
       "      'variant_description': 'reviewed by expert panel'}},\n",
       "    'prediction': {},\n",
       "    'references': [{'sources': ['HGMD'],\n",
       "      'pubmed_id': 28324225,\n",
       "      'source_info': {'HGMD': 'Primary literature report. Classified as VUS3 (IARC).'}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 18844490, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 28492532, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 19471317, 'source_info': {}},\n",
       "     {'sources': ['CLINVAR'], 'pubmed_id': 10923033, 'source_info': {}}],\n",
       "    'frequencies': {'ExAC': {'hom': {'G': 3,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 3},\n",
       "      'num': {'G': 121222,\n",
       "       'AFR': 10322,\n",
       "       'AMR': 11568,\n",
       "       'EAS': 8652,\n",
       "       'FIN': 6614,\n",
       "       'NFE': 66654,\n",
       "       'OTH': 906,\n",
       "       'SAS': 16506},\n",
       "      'freq': {'G': 0.0010724125983732325,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 8.644536652835408e-05,\n",
       "       'EAS': 0.0,\n",
       "       'FIN': 0.0,\n",
       "       'NFE': 0.00019503705704083775,\n",
       "       'OTH': 0.0,\n",
       "       'SAS': 0.007027747485762753},\n",
       "      'count': {'G': 130,\n",
       "       'AFR': 0,\n",
       "       'AMR': 1,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'Het': 124,\n",
       "       'NFE': 13,\n",
       "       'OTH': 0,\n",
       "       'SAS': 116}},\n",
       "     'inDB': {'hom': {'OUST1': 0},\n",
       "      'num': {'OUST1': 588},\n",
       "      'freq': {'OUST1': 0.0017006802721088435},\n",
       "      'count': {'OUST1': 1},\n",
       "      'filter': {'OUST1': ['PASS']},\n",
       "      'indications': {'OUST1': {'TrioPU': '1'}}},\n",
       "     '1000g': {'freq': {'G': 0.002,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 0.0,\n",
       "       'EAS': 0.0,\n",
       "       'EUR': 0.001,\n",
       "       'SAS': 0.0092}},\n",
       "     'esp6500': {'freq': {'AA': 0.0, 'EA': 0.0003}},\n",
       "     'GNOMAD_EXOMES': {'hom': {'G': 5,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'SAS': 5,\n",
       "       'Male': 2,\n",
       "       'Female': 3},\n",
       "      'num': {'G': 246204,\n",
       "       'AFR': 15296,\n",
       "       'AMR': 33580,\n",
       "       'ASJ': 9848,\n",
       "       'EAS': 17246,\n",
       "       'FIN': 22296,\n",
       "       'NFE': 111674,\n",
       "       'OTH': 5484,\n",
       "       'SAS': 30780,\n",
       "       'Male': 134874,\n",
       "       'Female': 111330},\n",
       "      'freq': {'G': 0.001108836574547936,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 8.933889219773674e-05,\n",
       "       'ASJ': 0.0,\n",
       "       'EAS': 0.0,\n",
       "       'FIN': 0.0,\n",
       "       'NFE': 0.00026863907444884215,\n",
       "       'OTH': 0.0005470459518599562,\n",
       "       'SAS': 0.007699805068226121,\n",
       "       'Male': 0.0014828654892714683,\n",
       "       'Female': 0.0006557082547381659},\n",
       "      'count': {'G': 273,\n",
       "       'AFR': 0,\n",
       "       'AMR': 3,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 30,\n",
       "       'OTH': 3,\n",
       "       'SAS': 237,\n",
       "       'Male': 200,\n",
       "       'Female': 73},\n",
       "      'filter': {'G': ['PASS']}},\n",
       "     'GNOMAD_GENOMES': {'hom': {'G': 0,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 0,\n",
       "       'OTH': 0,\n",
       "       'Male': 0,\n",
       "       'Female': 0},\n",
       "      'num': {'G': 30980,\n",
       "       'AFR': 8736,\n",
       "       'AMR': 838,\n",
       "       'ASJ': 302,\n",
       "       'EAS': 1622,\n",
       "       'FIN': 3494,\n",
       "       'NFE': 15008,\n",
       "       'OTH': 980,\n",
       "       'Male': 17122,\n",
       "       'Female': 13858},\n",
       "      'freq': {'G': 0.0001291155584247902,\n",
       "       'AFR': 0.0,\n",
       "       'AMR': 0.0,\n",
       "       'ASJ': 0.0,\n",
       "       'EAS': 0.0,\n",
       "       'FIN': 0.0,\n",
       "       'NFE': 0.00026652452025586353,\n",
       "       'OTH': 0.0,\n",
       "       'Male': 0.00011680878402055835,\n",
       "       'Female': 0.0001443209698369173},\n",
       "      'count': {'G': 4,\n",
       "       'AFR': 0,\n",
       "       'AMR': 0,\n",
       "       'ASJ': 0,\n",
       "       'EAS': 0,\n",
       "       'FIN': 0,\n",
       "       'NFE': 4,\n",
       "       'OTH': 0,\n",
       "       'Male': 2,\n",
       "       'Female': 2},\n",
       "      'filter': {'G': ['PASS']}}},\n",
       "    'transcripts': [{'exon': '3/27',\n",
       "      'HGVSc': 'c.198A>G',\n",
       "      'HGVSp': 'c.198A>G(p.=)',\n",
       "      'dbsnp': ['rs28897700'],\n",
       "      'codons': 'caA/caG',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENST00000380152.3',\n",
       "      'transcript': 'ENST00000380152',\n",
       "      'HGVSc_short': 'c.198A>G',\n",
       "      'amino_acids': 'Q',\n",
       "      'consequences': ['synonymous_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/10',\n",
       "      'HGVSc': 'c.-172A>G',\n",
       "      'dbsnp': ['rs28897700'],\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'transcript': 'ENST00000530893',\n",
       "      'HGVSc_short': 'c.-172A>G',\n",
       "      'consequences': ['5_prime_UTR_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': False,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': -172},\n",
       "     {'dbsnp': ['rs28897700'],\n",
       "      'strand': -1,\n",
       "      'symbol': 'ZAR1L',\n",
       "      'hgnc_id': 37116,\n",
       "      'transcript': 'ENST00000533490',\n",
       "      'consequences': ['upstream_gene_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True},\n",
       "     {'exon': '3/28',\n",
       "      'HGVSc': 'c.198A>G',\n",
       "      'HGVSp': 'c.198A>G(p.=)',\n",
       "      'dbsnp': ['rs28897700'],\n",
       "      'codons': 'caA/caG',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'ENST00000544455.1',\n",
       "      'transcript': 'ENST00000544455',\n",
       "      'HGVSc_short': 'c.198A>G',\n",
       "      'amino_acids': 'Q',\n",
       "      'consequences': ['synonymous_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0},\n",
       "     {'exon': '3/27',\n",
       "      'HGVSc': 'c.198A>G',\n",
       "      'HGVSp': 'c.198A>G(p.=)',\n",
       "      'dbsnp': ['rs28897700'],\n",
       "      'codons': 'caA/caG',\n",
       "      'strand': 1,\n",
       "      'symbol': 'BRCA2',\n",
       "      'hgnc_id': 1101,\n",
       "      'protein': 'NM_000059.3',\n",
       "      'transcript': 'NM_000059.3',\n",
       "      'HGVSc_short': 'c.198A>G',\n",
       "      'amino_acids': 'Q',\n",
       "      'consequences': ['synonymous_variant'],\n",
       "      'in_last_exon': 'no',\n",
       "      'is_canonical': True,\n",
       "      'exon_distance': 0,\n",
       "      'coding_region_distance': 0}]},\n",
       "   'previous_annotation_id': None,\n",
       "   'date_superceeded': None,\n",
       "   'date_created': Timestamp('2020-10-19 07:29:46.327685+0000', tz='UTC'),\n",
       "   'schema_version': 1}]}"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "report_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
