{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example HBB VCF and Annotation\n",
    "\n",
    "Let's say we want to custom design a gene panel to integrate with Ella. We'll use HBB in this case, and generate some dummy VCFs that are homozygous for the reference, heterozygous, and homozygous for the alternate.\n",
    "\n",
    "[OMIM](http://www.omim.org/entry/141900)\n",
    "\n",
    "**Phenotype HBB c.20A>T**\n",
    "\n",
    "[Clinvar - NM_000518.5(HBB):c.20A>T (p.Glu7Val) AND HEMOGLOBIN S](https://www.ncbi.nlm.nih.gov/clinvar/RCV000016573.11/)\n",
    "\n",
    "[RS334](https://www.ncbi.nlm.nih.gov/snp/rs334)\n",
    "\n",
    "**Location** - Chr11: 5248232 (on Assembly GRCh37)\n",
    "\n",
    "NM_000518.5\n",
    "\n",
    "Alleles\n",
    "T>A / T>C / T>G\n",
    "\n",
    "**Genomic Placements**\n",
    "\n",
    "GRCh37.p13 chr 11\tNC_000011.9:g.5248232T>A\n",
    "\n",
    "GRCh37.p13 chr 11\tNC_000011.9:g.5248232T>C\n",
    "\n",
    "GRCh37.p13 chr 11\tNC_000011.9:g.5248232T>G\n",
    "\n",
    "\n",
    "**HBB c.19G>A (p.Glu6Lys)**\n",
    "\n",
    "[Clinvar HBB c.19G>A (p.Glu6Lys)](https://www.ncbi.nlm.nih.gov/clinvar/variation/15126/)\n",
    "\n",
    "[RS33930165](https://www.ncbi.nlm.nih.gov/snp/rs33930165)\n",
    "\n",
    "**Location** - Chr11: 5248233 (GRCh37)\n",
    "\n",
    "NM_000518.4\n",
    "\n",
    "Alleles\n",
    "C>G / C>T"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Dummy VCF data for SNPs associated to HBB\n",
    "\n",
    "This is a total hack to generate some VCF data that we can then load into Ella, and is only done for demonstration purposes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_data = {\n",
    "    \"sample_a_hom_ref\": [\n",
    "    {\n",
    "        # First one\n",
    "        '#CHROM': 11,\n",
    " 'POS': 5248232,\n",
    " 'ID': 'CM082515',\n",
    " 'REF': \"T\",\n",
    " 'ALT': \"A\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_a_hom_ref': '0/0:107,80:187:99:2048,0,2917'\n",
    "    },    \n",
    "    {\n",
    "    '#CHROM': 11,\n",
    "     'POS': 5248233,\n",
    " 'ID': 'CM082516',\n",
    " 'REF': \"C\",\n",
    " 'ALT': \"G\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_a_hom_ref': '0/0:107,80:187:99:2048,0,2917'\n",
    "    }\n",
    "],\n",
    "    \"hbb_sample_b_het\": [\n",
    "    {\n",
    " '#CHROM': 11,\n",
    " 'POS': 5248232,\n",
    " 'ID': 'CM082515',\n",
    " 'REF': \"T\",\n",
    " 'ALT': \"A\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_b_het': '0/1:107,80:187:99:2048,0,2917'\n",
    "    },    \n",
    "    {\n",
    "    '#CHROM': 11,\n",
    "     'POS': 5248233,\n",
    " 'ID': 'CM082516',\n",
    " 'REF': \"C\",\n",
    " 'ALT': \"G\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_b_het': '0/1:107,80:187:99:2048,0,2917'\n",
    "    }\n",
    "],\n",
    "   \"hbb_sample_c_hom_alt\" : [\n",
    "    {\n",
    " '#CHROM': 11,\n",
    " 'POS': 5248232,\n",
    " 'ID': 'CM082515',\n",
    " 'REF': \"T\",\n",
    " 'ALT': \"A\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_c_hom_alt': '1/1:107,80:187:99:2048,0,2917'\n",
    "    },    \n",
    "    {\n",
    "    '#CHROM': 11,\n",
    "     'POS': 5248233,\n",
    " 'ID': 'CM082516',\n",
    " 'REF': \"C\",\n",
    " 'ALT': \"G\",\n",
    " 'QUAL': 5000.0,\n",
    " 'FILTER': 'PASS',\n",
    " 'INFO': '.',\n",
    " 'FORMAT': 'GT:AD:DP:GQ:PL',\n",
    " 'hbb_sample_c_hom_alt': '1/1:107,80:187:99:2048,0,2917'\n",
    "    }\n",
    "] \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write out VCF Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import os\n",
    "import pandas as pd\n",
    "import requests\n",
    "from pprint import pprint\n",
    "import numpy as np\n",
    "from datetime import datetime\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These are the paths that are local to the jupyterhub notebook. \n",
    "\n",
    "# There is a /data volume that is mounted to all the containers\n",
    "# If you're using a different volume / file system change this\n",
    "BASE_PATH=\"/data\"\n",
    "\n",
    "ANALYSIS_INCOMING_PATH=os.path.join(BASE_PATH, \"analysis/incoming\")\n",
    "ANALYSIS_COMPLETE_PATH=os.path.join(BASE_PATH, \"analysis/complete\")\n",
    "\n",
    "os.makedirs(ANALYSIS_INCOMING_PATH, exist_ok=True)\n",
    "os.makedirs(ANALYSIS_COMPLETE_PATH, exist_ok=True)\n",
    "\n",
    "genepanel_name=\"test_HBB\"\n",
    "genepanel_version=\"v01\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "vcf_info_line = \"\"\"##fileformat=VCFv4.1\n",
    "##contig=<ID=13>\n",
    "##FILTER=<ID=PASS,Description=\"All filters passed\">\n",
    "\"\"\"\n",
    "\n",
    "vcf_columns = [\"#CHROM\", \"POS\", \"ID\", \"REF\", \"ALT\", \"QUAL\", \"FILTER\", \"INFO\", \"FORMAT\"]\n",
    "\n",
    "def write_pandas_csv_with_info_line(file_name, info_line, df):\n",
    "    with open(file_name, 'w') as fp:\n",
    "        fp.write(info_line)\n",
    "        df.to_csv(fp, index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for sample in vcf_data.keys():\n",
    "    columns = copy.deepcopy(vcf_columns)\n",
    "    columns.append(sample)\n",
    "    file_name = os.path.join(ANALYSIS_INCOMING_PATH, \"{}.vcf\".format(sample))\n",
    "    \n",
    "    vcf_records = vcf_data[sample]\n",
    "    df = pd.DataFrame(columns = columns, data=vcf_records)\n",
    "\n",
    "    with open(file_name, 'w') as fp:\n",
    "        fp.write(vcf_info_line)\n",
    "        df.to_csv(fp, index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write out the Annotation Commands\n",
    "\n",
    "This assumes that you are using the docker compose dev stack that is in this github repo. \n",
    "\n",
    "A quick note about the annotatation. Ella expects that VCF files will have the format `{analysis_name}.{genepanel_name}_{genepanel_version}`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "# SAMPLE: sample_a_hom_ref\n",
      "\n",
      "docker-compose exec  ella-anno bash  -c \"/anno/bin/annotate \\\n",
      "        --vcf /data/analysis/incoming/sample_a_hom_ref.vcf \\\n",
      "        -o /data/analysis/complete/sample_a_hom_ref.test_HBB_v01\"\n",
      "        \n",
      "docker-compose exec  ella-anno bash  -c \" mv \\\n",
      "       /data/analysis/complete/sample_a_hom_ref.test_HBB_v01/VCFANNO/output.vcf \\\n",
      "       /data/analysis/complete/sample_a_hom_ref.test_HBB_v01/sample_a_hom_ref.test_HBB_v01.vcf\"\n",
      "       \n",
      "docker-compose exec ella-web bash -c \"ella-cli deposit analysis \\\n",
      "       /data/analysis/complete/sample_a_hom_ref.test_HBB_v01/sample_a_hom_ref.test_HBB_v01.vcf\"\n",
      "\n",
      "    \n",
      "\n",
      "# SAMPLE: hbb_sample_b_het\n",
      "\n",
      "docker-compose exec  ella-anno bash  -c \"/anno/bin/annotate \\\n",
      "        --vcf /data/analysis/incoming/hbb_sample_b_het.vcf \\\n",
      "        -o /data/analysis/complete/hbb_sample_b_het.test_HBB_v01\"\n",
      "        \n",
      "docker-compose exec  ella-anno bash  -c \" mv \\\n",
      "       /data/analysis/complete/hbb_sample_b_het.test_HBB_v01/VCFANNO/output.vcf \\\n",
      "       /data/analysis/complete/hbb_sample_b_het.test_HBB_v01/hbb_sample_b_het.test_HBB_v01.vcf\"\n",
      "       \n",
      "docker-compose exec ella-web bash -c \"ella-cli deposit analysis \\\n",
      "       /data/analysis/complete/hbb_sample_b_het.test_HBB_v01/hbb_sample_b_het.test_HBB_v01.vcf\"\n",
      "\n",
      "    \n",
      "\n",
      "# SAMPLE: hbb_sample_c_hom_alt\n",
      "\n",
      "docker-compose exec  ella-anno bash  -c \"/anno/bin/annotate \\\n",
      "        --vcf /data/analysis/incoming/hbb_sample_c_hom_alt.vcf \\\n",
      "        -o /data/analysis/complete/hbb_sample_c_hom_alt.test_HBB_v01\"\n",
      "        \n",
      "docker-compose exec  ella-anno bash  -c \" mv \\\n",
      "       /data/analysis/complete/hbb_sample_c_hom_alt.test_HBB_v01/VCFANNO/output.vcf \\\n",
      "       /data/analysis/complete/hbb_sample_c_hom_alt.test_HBB_v01/hbb_sample_c_hom_alt.test_HBB_v01.vcf\"\n",
      "       \n",
      "docker-compose exec ella-web bash -c \"ella-cli deposit analysis \\\n",
      "       /data/analysis/complete/hbb_sample_c_hom_alt.test_HBB_v01/hbb_sample_c_hom_alt.test_HBB_v01.vcf\"\n",
      "\n",
      "    \n"
     ]
    }
   ],
   "source": [
    "for sample in vcf_data.keys():\n",
    "    name = \"{}.{}_{}\".format(sample, genepanel_name, genepanel_version)\n",
    "    \n",
    "    command = \"\"\"\n",
    "# SAMPLE: {sample}\n",
    "\n",
    "docker-compose exec  ella-anno bash  -c \"/anno/bin/annotate \\\\\n",
    "        --vcf {incoming}/{sample}.vcf \\\\\n",
    "        -o {complete}/{name}\"\n",
    "        \n",
    "docker-compose exec  ella-anno bash  -c \" mv \\\\\n",
    "       {complete}/{name}/VCFANNO/output.vcf \\\\\n",
    "       {complete}/{name}/{name}.vcf\"\n",
    "       \n",
    "docker-compose exec ella-web bash -c \"ella-cli deposit analysis \\\\\n",
    "       {complete}/{name}/{name}.vcf\"\n",
    "\n",
    "    \"\"\".format(name=name, sample=sample, incoming=ANALYSIS_INCOMING_PATH, complete=ANALYSIS_COMPLETE_PATH)\n",
    "    print(command)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
