{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis brca_sample_1.HBOCUTV_v01 in Ella - Query the Database Directly\n",
    "\n",
    "This shows how to query the Ella database for information on the analysis for custom reporting.\n",
    "\n",
    "This assumes that you ran the dev stack with `make dev; make load` in order to add in a sample analysis. If you haven't, or would like to use another sample or analysis, you will need to query with different IDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sqlalchemy as sa\n",
    "from sqlalchemy import create_engine\n",
    "from sqlalchemy import inspect\n",
    "import json\n",
    "from pprint import pprint\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SQL Alchemy Database Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'postgresql://postgres:password123@postgresql/postgres'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.environ.get('DB_URL')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "engine = sa.create_engine(os.environ.get('DB_URL'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext sql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%sql postgresql://postgres:password123@postgresql/postgres"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['alembic_version',\n",
      " 'analysis',\n",
      " 'alleleassessmentattachment',\n",
      " 'alleleinterpretationsnapshot',\n",
      " 'analysisinterpretation',\n",
      " 'alleleassessmentreferenceassessment',\n",
      " 'analysisinterpretationsnapshot',\n",
      " 'allele',\n",
      " 'annotationshadowfrequency',\n",
      " 'annotation',\n",
      " 'alleleinterpretation',\n",
      " 'allelereport',\n",
      " 'annotationjob',\n",
      " 'broadcast',\n",
      " 'clilog',\n",
      " 'attachment',\n",
      " 'customannotation',\n",
      " 'gene',\n",
      " 'geneassessment',\n",
      " 'genepanel_phenotype',\n",
      " 'phenotype',\n",
      " 'genotype',\n",
      " 'genotypesampledata',\n",
      " 'interpretationlog',\n",
      " 'interpretationstatehistory',\n",
      " 'reference',\n",
      " 'jsonschema',\n",
      " 'genepanel_transcript',\n",
      " 'genepanel',\n",
      " 'resourcelog',\n",
      " 'sample',\n",
      " 'transcript',\n",
      " 'usersession',\n",
      " 'useroldpassword',\n",
      " 'usergroupfilterconfig',\n",
      " 'usergroupgenepanel',\n",
      " 'filterconfig',\n",
      " 'alleleassessment',\n",
      " 'user',\n",
      " 'usergroup',\n",
      " 'referenceassessment',\n",
      " 'annotationshadowtranscript']\n"
     ]
    }
   ],
   "source": [
    "# Get column information\n",
    "def get_column_info(tables):\n",
    "    for table in tables:\n",
    "        print(table)\n",
    "        columns = inspector.get_columns(table)\n",
    "        pprint(columns)\n",
    "\n",
    "db_uri = os.environ.get('DB_URL')\n",
    "engine = create_engine(db_uri)\n",
    "\n",
    "inspector = inspect(engine)\n",
    "\n",
    "# Get table information\n",
    "tables = inspector.get_table_names()\n",
    "pprint(tables)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample brca_sample_1\n",
    "\n",
    "Most of the time a report is generated for a particular sample. \n",
    "\n",
    "Sample Id -> Analysis Id -> List(Interpretation Ids) -> List(Allele Id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "6 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table>\n",
       "    <tr>\n",
       "        <th>id</th>\n",
       "        <th>identifier</th>\n",
       "        <th>analysis_id</th>\n",
       "        <th>sample_type</th>\n",
       "        <th>date_deposited</th>\n",
       "        <th>affected</th>\n",
       "        <th>family_id</th>\n",
       "        <th>father_id</th>\n",
       "        <th>mother_id</th>\n",
       "        <th>sibling_id</th>\n",
       "        <th>proband</th>\n",
       "        <th>sex</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>1</td>\n",
       "        <td>brca_allfiltered</td>\n",
       "        <td>1</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:41.614773+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>2</td>\n",
       "        <td>brca_sample_1</td>\n",
       "        <td>2</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:43.434024+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>4</td>\n",
       "        <td>brca_sample_3</td>\n",
       "        <td>4</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:49.807489+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>5</td>\n",
       "        <td>brca_sample_1</td>\n",
       "        <td>5</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:51.617945+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>6</td>\n",
       "        <td>brca_sample_master</td>\n",
       "        <td>6</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:53.272682+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "        <td>7</td>\n",
       "        <td>brca_sample_master</td>\n",
       "        <td>7</td>\n",
       "        <td>HTS</td>\n",
       "        <td>2020-10-19 10:46:54.988033+00:00</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>None</td>\n",
       "        <td>True</td>\n",
       "        <td>None</td>\n",
       "    </tr>\n",
       "</table>"
      ],
      "text/plain": [
       "[(1, 'brca_allfiltered', 1, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 41, 614773, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (2, 'brca_sample_1', 2, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 43, 434024, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (4, 'brca_sample_3', 4, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 49, 807489, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (5, 'brca_sample_1', 5, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 51, 617945, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (6, 'brca_sample_master', 6, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 53, 272682, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None),\n",
       " (7, 'brca_sample_master', 7, 'HTS', datetime.datetime(2020, 10, 19, 10, 46, 54, 988033, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), True, None, None, None, None, True, None)]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%sql\n",
    "\n",
    "SELECT * FROM sample;"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Analysis brca_sample_1.HBOCUTV_v01\n",
    "\n",
    "Since any sample can duplicate records with more than 1 analysis, make sure you are getting the correct analysis and the correct. On the UI there needs to be a filter by sample_name, sample_type, date_deposited, genepanel name, and/or genepanel version.\n",
    "\n",
    "Generally the Sample Name is going to be read in from a LIMs system somewhere anyways."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# make sure to use single quotes, '', not \"\"\n",
    "\n",
    "report_data = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>identifier</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_deposited</th>\n",
       "      <th>affected</th>\n",
       "      <th>family_id</th>\n",
       "      <th>father_id</th>\n",
       "      <th>mother_id</th>\n",
       "      <th>sibling_id</th>\n",
       "      <th>proband</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>brca_sample_1</td>\n",
       "      <td>2</td>\n",
       "      <td>HTS</td>\n",
       "      <td>2020-10-19 10:46:43.434024+00:00</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id     identifier  analysis_id sample_type  \\\n",
       "0   2  brca_sample_1            2         HTS   \n",
       "\n",
       "                    date_deposited  affected family_id father_id mother_id  \\\n",
       "0 2020-10-19 10:46:43.434024+00:00      True      None      None      None   \n",
       "\n",
       "  sibling_id  proband   sex  \n",
       "0       None     True  None  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample = %sql select * from sample where analysis_id = 2 AND identifier = 'brca_sample_1';\n",
    "sample_df = sample.DataFrame()\n",
    "sample_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['sample'] = sample_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>warnings</th>\n",
       "      <th>report</th>\n",
       "      <th>date_deposited</th>\n",
       "      <th>properties</th>\n",
       "      <th>date_requested</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>brca_sample_1.HBOCUTV_v01</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.419040+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id                       name genepanel_name genepanel_version warnings  \\\n",
       "0   2  brca_sample_1.HBOCUTV_v01        HBOCUTV               v01     None   \n",
       "\n",
       "  report                   date_deposited properties date_requested  \n",
       "0   None 2020-10-19 10:46:43.419040+00:00       None           None  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analysis = %sql SELECT * FROM analysis where name = 'brca_sample_1.HBOCUTV_v01';\n",
    "analysis_df = analysis.DataFrame()\n",
    "analysis_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['analysis'] = analysis_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "1 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>user_state</th>\n",
       "      <th>state</th>\n",
       "      <th>status</th>\n",
       "      <th>date_last_update</th>\n",
       "      <th>date_created</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>user_id</th>\n",
       "      <th>finalized</th>\n",
       "      <th>workflow_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>{'allele': {'3': {'sections': {}, 'allele_id':...</td>\n",
       "      <td>{'allele': {'3': {'report': {'included': True}...</td>\n",
       "      <td>Ongoing</td>\n",
       "      <td>2020-10-19 10:50:59.532592+00:00</td>\n",
       "      <td>2020-10-19 10:46:43.427738+00:00</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>Interpretation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id genepanel_name genepanel_version  \\\n",
       "0   2        HBOCUTV               v01   \n",
       "\n",
       "                                          user_state  \\\n",
       "0  {'allele': {'3': {'sections': {}, 'allele_id':...   \n",
       "\n",
       "                                               state   status  \\\n",
       "0  {'allele': {'3': {'report': {'included': True}...  Ongoing   \n",
       "\n",
       "                  date_last_update                     date_created  \\\n",
       "0 2020-10-19 10:50:59.532592+00:00 2020-10-19 10:46:43.427738+00:00   \n",
       "\n",
       "   analysis_id  user_id finalized workflow_status  \n",
       "0            2        1      None  Interpretation  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# At some point I will want to filter on analyses that are complete\n",
    "\n",
    "analysis_interpretation = %sql select * from analysisinterpretation where analysis_id = 2;\n",
    "analysis_interprtetation_df = analysis_interpretation.DataFrame()\n",
    "analysis_interprtetation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data['analysis_interprtetation'] = analysis_interprtetation_df.to_dict('records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * postgresql://postgres:***@postgresql/postgres\n",
      "6 rows affected.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>classification</th>\n",
       "      <th>evaluation</th>\n",
       "      <th>user_id</th>\n",
       "      <th>date_created</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>previous_assessment_id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>annotation_id</th>\n",
       "      <th>custom_annotation_id</th>\n",
       "      <th>usergroup_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:49:44.630277+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:50:00.770452+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:50:14.357702+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>U</td>\n",
       "      <td>{'acmg': {'included': [], 'suggested': [{'op':...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:50:26.990666+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:50:38.883637+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>U</td>\n",
       "      <td>{'acmg': {'included': [], 'suggested': [{'op':...</td>\n",
       "      <td>1</td>\n",
       "      <td>2020-10-19 10:50:47.757943+00:00</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>8</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id classification                                         evaluation  \\\n",
       "0   1              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "1   2              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "2   3              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "3   4              U  {'acmg': {'included': [], 'suggested': [{'op':...   \n",
       "4   5              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "5   6              U  {'acmg': {'included': [], 'suggested': [{'op':...   \n",
       "\n",
       "   user_id                     date_created date_superceeded  \\\n",
       "0        1 2020-10-19 10:49:44.630277+00:00             None   \n",
       "1        1 2020-10-19 10:50:00.770452+00:00             None   \n",
       "2        1 2020-10-19 10:50:14.357702+00:00             None   \n",
       "3        1 2020-10-19 10:50:26.990666+00:00             None   \n",
       "4        1 2020-10-19 10:50:38.883637+00:00             None   \n",
       "5        1 2020-10-19 10:50:47.757943+00:00             None   \n",
       "\n",
       "  previous_assessment_id  allele_id genepanel_name genepanel_version  \\\n",
       "0                   None          3        HBOCUTV               v01   \n",
       "1                   None          4        HBOCUTV               v01   \n",
       "2                   None          5        HBOCUTV               v01   \n",
       "3                   None          6        HBOCUTV               v01   \n",
       "4                   None          7        HBOCUTV               v01   \n",
       "5                   None          8        HBOCUTV               v01   \n",
       "\n",
       "   analysis_id  annotation_id custom_annotation_id  usergroup_id  \n",
       "0            2              3                 None             1  \n",
       "1            2              4                 None             1  \n",
       "2            2              5                 None             1  \n",
       "3            2              6                 None             1  \n",
       "4            2              7                 None             1  \n",
       "5            2              8                 None             1  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_assessment = %sql select * from alleleassessment where analysis_id = 2;\n",
    "allele_assessment_df = allele_assessment.DataFrame()\n",
    "\n",
    "report_data['allele_assessment'] = allele_assessment_df.to_dict('records')\n",
    "\n",
    "allele_assessment_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>annotations</th>\n",
       "      <th>previous_annotation_id</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>date_created</th>\n",
       "      <th>schema_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 4...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496905+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 1...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496920+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'acc_num':...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496927+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 2...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496932+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM', 'codon': 3...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496937+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>{'external': {'HGMD': {'tag': 'DM?', 'codon': ...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2020-10-19 10:46:43.496943+00:00</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  allele_id                                        annotations  \\\n",
       "0   3          3  {'external': {'HGMD': {'tag': 'DM', 'codon': 4...   \n",
       "1   4          4  {'external': {'HGMD': {'tag': 'DM', 'codon': 1...   \n",
       "2   5          5  {'external': {'HGMD': {'tag': 'DM', 'acc_num':...   \n",
       "3   6          6  {'external': {'HGMD': {'tag': 'DM', 'codon': 2...   \n",
       "4   7          7  {'external': {'HGMD': {'tag': 'DM', 'codon': 3...   \n",
       "5   8          8  {'external': {'HGMD': {'tag': 'DM?', 'codon': ...   \n",
       "\n",
       "  previous_annotation_id date_superceeded                     date_created  \\\n",
       "0                   None             None 2020-10-19 10:46:43.496905+00:00   \n",
       "1                   None             None 2020-10-19 10:46:43.496920+00:00   \n",
       "2                   None             None 2020-10-19 10:46:43.496927+00:00   \n",
       "3                   None             None 2020-10-19 10:46:43.496932+00:00   \n",
       "4                   None             None 2020-10-19 10:46:43.496937+00:00   \n",
       "5                   None             None 2020-10-19 10:46:43.496943+00:00   \n",
       "\n",
       "   schema_version  \n",
       "0               1  \n",
       "1               1  \n",
       "2               1  \n",
       "3               1  \n",
       "4               1  \n",
       "5               1  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Now let's grab the allele ids\n",
    "allele_ids = allele_assessment_df['allele_id'].tolist() \n",
    "query = sa.text(\"\"\" \n",
    "                SELECT *\n",
    "                FROM annotation\n",
    "                WHERE allele_id IN :values; \n",
    "\"\"\")\n",
    "query = query.bindparams(values=tuple(allele_ids))\n",
    "allele_annotation_df = pd.read_sql(query, engine)\n",
    "\n",
    "report_data['allele_annotation'] = allele_annotation_df.to_dict('records')\n",
    "\n",
    "allele_annotation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>genome_reference</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>open_end_position</th>\n",
       "      <th>change_from</th>\n",
       "      <th>change_to</th>\n",
       "      <th>change_type</th>\n",
       "      <th>vcf_pos</th>\n",
       "      <th>vcf_ref</th>\n",
       "      <th>vcf_alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890606</td>\n",
       "      <td>32890607</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32890607</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890645</td>\n",
       "      <td>32890647</td>\n",
       "      <td>AC</td>\n",
       "      <td></td>\n",
       "      <td>del</td>\n",
       "      <td>32890645</td>\n",
       "      <td>GAC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890665</td>\n",
       "      <td>32890666</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32890666</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893217</td>\n",
       "      <td>32893218</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893218</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893242</td>\n",
       "      <td>32893243</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893243</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893343</td>\n",
       "      <td>32893344</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893344</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id genome_reference chromosome  start_position  open_end_position  \\\n",
       "0   3           GRCh37         13        32890606           32890607   \n",
       "1   4           GRCh37         13        32890645           32890647   \n",
       "2   5           GRCh37         13        32890665           32890666   \n",
       "3   6           GRCh37         13        32893217           32893218   \n",
       "4   7           GRCh37         13        32893242           32893243   \n",
       "5   8           GRCh37         13        32893343           32893344   \n",
       "\n",
       "  change_from change_to change_type   vcf_pos vcf_ref vcf_alt  \n",
       "0           G         T         SNP  32890607       G       T  \n",
       "1          AC                   del  32890645     GAC       G  \n",
       "2           T         A         SNP  32890666       T       A  \n",
       "3           A         T         SNP  32893218       A       T  \n",
       "4           G         T         SNP  32893243       G       T  \n",
       "5           A         G         SNP  32893344       A       G  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "query = sa.text(\"\"\" \n",
    "                SELECT *\n",
    "                FROM allele\n",
    "                WHERE id IN :values; \n",
    "\"\"\")\n",
    "query = query.bindparams(values=tuple(allele_ids))\n",
    "allele_df = pd.read_sql(query, engine)\n",
    "\n",
    "\n",
    "report_data['allele'] = allele_df.to_dict('records')\n",
    "\n",
    "allele_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample Analysis Report Data Set\n",
    "\n",
    "The dataset from a sample analysis should look like:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['sample', 'analysis', 'analysis_interprtetation', 'allele_assessment', 'allele_annotation', 'allele'])\n"
     ]
    }
   ],
   "source": [
    "pprint(report_data.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
