{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example - Investigate the Ella HBOC_v01 Gene Panel\n",
    "\n",
    "In order to understand what is happening behind the scenes let's take a look at the data for Gene Panel **HBOC_v01** and reverse engineer the record for the BRCA2 gene."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import requests\n",
    "from pprint import pprint\n",
    "import numpy as np\n",
    "import tempfile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## The Transcripts and Phenotypes Files for the HBOC_v01 GenePanel\n",
    "\n",
    "For the sake of record keeping these files are included here, but they are also in the Ella github repo under `src/vardb/testdata/clinicalGenePanels/HBOC_v01`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "transcript_file=\"\"\"# Genepanel: HBOC Version: 01 Date: 2017-11-07\n",
    "#chromosome\ttxStart\ttxEnd\trefseq\tscore\tstrand\tgeneSymbol\tHGNC\tOmim gene entry\tgeneAlias\teGeneID\teTranscriptID\tcdsStart\tcdsEnd\texonsStarts\texonEnds\n",
    "17\t41196311\t41277468\tNM_007297.3\t0\t-\tBRCA1\t1100\t113705\tRNF53,BRCC1,PPP1R53,FANCS\tENSG00000012048\tENST00000309486\t41197694\t41258543\t41196311,41199659,41201137,41203079,41209068,41215349,41215890,41219624,41222944,41226347,41228504,41234420,41242960,41243451,41247862,41249260,41251791,41256138,41256884,41258472,41276033,41277293\t41197819,41199720,41201211,41203134,41209152,41215390,41215968,41219712,41223255,41226538,41228631,41234592,41243049,41246877,41247939,41249306,41251897,41256278,41256973,41258550,41276132,41277468\n",
    "13\t32889616\t32973809\tNM_000059.3\t0\t+\tBRCA2\t1101\t600185\tFAD,FAD1,BRCC2,XRCC11\tENSG00000139618\tENST00000544455\t32890597\t32972907\t32889616,32890558,32893213,32899212,32900237,32900378,32900635,32903579,32905055,32906408,32910401,32918694,32920963,32928997,32930564,32931878,32936659,32937315,32944538,32945092,32950806,32953453,32953886,32954143,32968825,32971034,32972298\t32889804,32890664,32893462,32899321,32900287,32900419,32900750,32903629,32905167,32907524,32915333,32918790,32921033,32929425,32930746,32932066,32936830,32937670,32944694,32945237,32950928,32953652,32954050,32954282,32969070,32971181,32973809\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "phenotype_file=\"\"\"# Genepanel: HBOC Version: 01 Date: 2017-11-07\n",
    "#gene symbol\tHGNC\tremove (add x)\tphenotype\tinheritance\tomim_number\tpmid\tinheritance info\tcomment\n",
    "BRCA1\t1100\t\t{Breast-ovarian cancer, familial, 1}\tAD\t604370\n",
    "BRCA2\t1101\t\t{Breast-ovarian cancer, familial, 2}\tAD\t612555\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ella provides several sample genepanels. Let's take the first `HBOC_v01.transcripts.csv` gene panel and reverse engineer it using `myGene.info`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chromosome</th>\n",
       "      <th>txStart</th>\n",
       "      <th>txEnd</th>\n",
       "      <th>refseq</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>HGNC</th>\n",
       "      <th>Omim gene entry</th>\n",
       "      <th>geneAlias</th>\n",
       "      <th>eGeneID</th>\n",
       "      <th>eTranscriptID</th>\n",
       "      <th>cdsStart</th>\n",
       "      <th>cdsEnd</th>\n",
       "      <th>exonsStarts</th>\n",
       "      <th>exonEnds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>41196311</td>\n",
       "      <td>41277468</td>\n",
       "      <td>NM_007297.3</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>1100</td>\n",
       "      <td>113705</td>\n",
       "      <td>RNF53,BRCC1,PPP1R53,FANCS</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>ENST00000309486</td>\n",
       "      <td>41197694</td>\n",
       "      <td>41258543</td>\n",
       "      <td>41196311,41199659,41201137,41203079,41209068,4...</td>\n",
       "      <td>41197819,41199720,41201211,41203134,41209152,4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13</td>\n",
       "      <td>32889616</td>\n",
       "      <td>32973809</td>\n",
       "      <td>NM_000059.3</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>1101</td>\n",
       "      <td>600185</td>\n",
       "      <td>FAD,FAD1,BRCC2,XRCC11</td>\n",
       "      <td>ENSG00000139618</td>\n",
       "      <td>ENST00000544455</td>\n",
       "      <td>32890597</td>\n",
       "      <td>32972907</td>\n",
       "      <td>32889616,32890558,32893213,32899212,32900237,3...</td>\n",
       "      <td>32889804,32890664,32893462,32899321,32900287,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #chromosome   txStart     txEnd       refseq  score strand geneSymbol  \\\n",
       "0           17  41196311  41277468  NM_007297.3      0      -      BRCA1   \n",
       "1           13  32889616  32973809  NM_000059.3      0      +      BRCA2   \n",
       "\n",
       "   HGNC  Omim gene entry                  geneAlias          eGeneID  \\\n",
       "0  1100           113705  RNF53,BRCC1,PPP1R53,FANCS  ENSG00000012048   \n",
       "1  1101           600185      FAD,FAD1,BRCC2,XRCC11  ENSG00000139618   \n",
       "\n",
       "     eTranscriptID  cdsStart    cdsEnd  \\\n",
       "0  ENST00000309486  41197694  41258543   \n",
       "1  ENST00000544455  32890597  32972907   \n",
       "\n",
       "                                         exonsStarts  \\\n",
       "0  41196311,41199659,41201137,41203079,41209068,4...   \n",
       "1  32889616,32890558,32893213,32899212,32900237,3...   \n",
       "\n",
       "                                            exonEnds  \n",
       "0  41197819,41199720,41201211,41203134,41209152,4...  \n",
       "1  32889804,32890664,32893462,32899321,32900287,3...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with tempfile.NamedTemporaryFile(mode='w+t') as temp:\n",
    "    temp.write(transcript_file)\n",
    "    temp.flush()\n",
    "    columns = [\"#chromosome\",\"txStart\", \"txEnd\",\"refseq\", \"score\", \"strand\", \"geneSymbol\", \"HGNC\", \"Omim gene entry\", \"geneAlias\", \"eGeneID\", \"eTranscriptID\", \"cdsStart\", \"cdsEnd\", \"exonsStarts\", \"exonEnds\"]\n",
    "    genepanel = pd.read_csv(temp.name, sep=\"\\t\", skiprows=2, header=None, names=columns)\n",
    "\n",
    "genepanel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Query MyGene.Info\n",
    "\n",
    "Query the mygene.info by the eGeneID. It is **VERY IMPORTANT** to note that Ella Anno uses HG19, so make sure you query the coordinates accordingly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['HGNC', 'MIM', '_id', '_version', 'accession', 'alias', 'clingen', 'ensembl', 'entrezgene', 'exac', 'exons', 'exons_hg19', 'generif', 'genomic_pos', 'genomic_pos_hg19', 'go', 'homologene', 'interpro', 'ipi', 'map_location', 'name', 'other_names', 'pantherdb', 'pathway', 'pdb', 'pfam', 'pharmgkb', 'pharos', 'pir', 'prosite', 'reagent', 'refseq', 'reporter', 'retired', 'summary', 'symbol', 'taxid', 'type_of_gene', 'umls', 'unigene', 'uniprot', 'wikipedia'])"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_data = requests.get(\"https://mygene.info/v3/gene/ENSG00000139618\")\n",
    "gene_data = gene_data.json()\n",
    "gene_data.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the Gene Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gene_data['exons_hg19']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'chr': '13', 'end': 32973805, 'start': 32889611, 'strand': 1}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_data['genomic_pos_hg19']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GenePanel TxStart:\t32889616\n",
      "MyGene.Info TxStart:\t32889644\n",
      "GenePanel TxEnd:\t32973809\n",
      "MyGene.Info TxEnd:\t32974405\n"
     ]
    }
   ],
   "source": [
    "print(\"GenePanel TxStart:\\t{}\".format(genepanel.iloc[1]['txStart']))\n",
    "print(\"MyGene.Info TxStart:\\t{}\".format(gene_data['exons_hg19'][0]['txstart']))\n",
    "\n",
    "print(\"GenePanel TxEnd:\\t{}\".format(genepanel.iloc[1]['txEnd']))\n",
    "print(\"MyGene.Info TxEnd:\\t{}\".format(gene_data['exons_hg19'][0]['txend']))\n",
    "mygene_txstart = gene_data['exons_hg19'][0]['txstart']\n",
    "mygene_txend = gene_data['exons_hg19'][0]['txend']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GenePanel CDSStart:\t32890597\n",
      "MyGene.Info CDSStart:\t32890597\n",
      "GenePanel CDSEnd:\t32972907\n",
      "MyGene.Info CDSEnd:\t32972907\n"
     ]
    }
   ],
   "source": [
    "print(\"GenePanel CDSStart:\\t{}\".format(genepanel.iloc[1]['cdsStart']))\n",
    "print(\"MyGene.Info CDSStart:\\t{}\".format(gene_data['exons_hg19'][0]['cdsstart']))\n",
    "\n",
    "\n",
    "print(\"GenePanel CDSEnd:\\t{}\".format(genepanel.iloc[1]['cdsEnd']))\n",
    "print(\"MyGene.Info CDSEnd:\\t{}\".format(gene_data['exons_hg19'][0]['cdsend']))\n",
    "\n",
    "mygene_cdsstart = gene_data['exons_hg19'][0]['cdsstart']\n",
    "mygene_cdsend = gene_data['exons_hg19'][0]['cdsend']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_start = genepanel.iloc[1]['exonsStarts'].split(',')\n",
    "exon_start = np.array(exon_start, dtype=int)\n",
    "\n",
    "exon_end = genepanel.iloc[1]['exonEnds'].split(',')\n",
    "exon_end = np.array(exon_end, dtype=int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[32889616, 32889804],\n",
       "       [32890558, 32890664],\n",
       "       [32893213, 32893462],\n",
       "       [32899212, 32899321],\n",
       "       [32900237, 32900287],\n",
       "       [32900378, 32900419],\n",
       "       [32900635, 32900750],\n",
       "       [32903579, 32903629],\n",
       "       [32905055, 32905167],\n",
       "       [32906408, 32907524],\n",
       "       [32910401, 32915333],\n",
       "       [32918694, 32918790],\n",
       "       [32920963, 32921033],\n",
       "       [32928997, 32929425],\n",
       "       [32930564, 32930746],\n",
       "       [32931878, 32932066],\n",
       "       [32936659, 32936830],\n",
       "       [32937315, 32937670],\n",
       "       [32944538, 32944694],\n",
       "       [32945092, 32945237],\n",
       "       [32950806, 32950928],\n",
       "       [32953453, 32953652],\n",
       "       [32953886, 32954050],\n",
       "       [32954143, 32954282],\n",
       "       [32968825, 32969070],\n",
       "       [32971034, 32971181],\n",
       "       [32972298, 32973809]])"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genepanel_exons = np.stack([exon_start, exon_end], axis=1)\n",
    "genepanel_exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[32889644, 32889804],\n",
       "       [32890558, 32890664],\n",
       "       [32893213, 32893462],\n",
       "       [32899212, 32899321],\n",
       "       [32900237, 32900287],\n",
       "       [32900378, 32900419],\n",
       "       [32900635, 32900750],\n",
       "       [32903579, 32903629],\n",
       "       [32905055, 32905167],\n",
       "       [32906408, 32907524],\n",
       "       [32910401, 32915333],\n",
       "       [32918694, 32918790],\n",
       "       [32920963, 32921033],\n",
       "       [32928997, 32929425],\n",
       "       [32930564, 32930746],\n",
       "       [32931878, 32932066],\n",
       "       [32936659, 32936830],\n",
       "       [32937315, 32937670],\n",
       "       [32944538, 32944694],\n",
       "       [32945092, 32945237],\n",
       "       [32950806, 32950928],\n",
       "       [32953453, 32953652],\n",
       "       [32953886, 32954050],\n",
       "       [32954143, 32954282],\n",
       "       [32968825, 32969070],\n",
       "       [32971034, 32971181],\n",
       "       [32972298, 32974405]])"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mygene_exons = np.array(gene_data['exons_hg19'][0]['position'], dtype=int)\n",
    "mygene_exons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The two arrays are almost equal, but the start of the first exon is off by a little.\n",
    "# np.testing.assert_array_equal(mygene_exons, genepanel_exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([32889644, 32890558, 32893213, 32899212, 32900237, 32900378,\n",
       "       32900635, 32903579, 32905055, 32906408, 32910401, 32918694,\n",
       "       32920963, 32928997, 32930564, 32931878, 32936659, 32937315,\n",
       "       32944538, 32945092, 32950806, 32953453, 32953886, 32954143,\n",
       "       32968825, 32971034, 32972298])"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mygene_exon_starts = mygene_exons[:,0]\n",
    "mygene_exon_starts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([32889804, 32890664, 32893462, 32899321, 32900287, 32900419,\n",
       "       32900750, 32903629, 32905167, 32907524, 32915333, 32918790,\n",
       "       32921033, 32929425, 32930746, 32932066, 32936830, 32937670,\n",
       "       32944694, 32945237, 32950928, 32953652, 32954050, 32954282,\n",
       "       32969070, 32971181, 32974405])"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mygene_exon_ends = mygene_exons[:,1]\n",
    "mygene_exon_ends"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the Chromosome and Strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr = gene_data['exons_hg19'][0]['chr']\n",
    "strand = gene_data['exons_hg19'][0]['strand']\n",
    "if strand == 1:\n",
    "    strand = '+'\n",
    "else:\n",
    "    strand = '-'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'13'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'+'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strand"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Query MyVariant.Info for BRCA2 - NM_000059.3\n",
    "\n",
    "This gives us some useful variant information that can be useful in building out the phenotype file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'took': 5,\n",
       " 'total': 10504,\n",
       " 'max_score': 5.6187677,\n",
       " 'hits': [{'_id': 'chr13:g.32953582_32953647del',\n",
       "   '_score': 5.6187677,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 568493,\n",
       "    'alt': 'G',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32953647, 'start': 32953582},\n",
       "    'hg38': {'end': 32379510, 'start': 32379445},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.8885_8950del',\n",
       "      'NM_000059.3:c.8885_8950del',\n",
       "      'NM_000059.3:c.8885_8950del',\n",
       "      'NM_000059.3:c.8885_8950del66',\n",
       "      'NM_000059.3:c.8885_8950del66',\n",
       "      'NM_000059.3:c.8885_8950delTATCAAGGGATGTCACAACCGTGTGGAAGTTGCGTATTGTAAGCTATTCAAAAAAAGAAAAAGATT'],\n",
       "     'genomic': ['LRG_293:g.68968_69033del',\n",
       "      'NC_000013.10:g.32953584_32953649del',\n",
       "      'NC_000013.11:g.32379447_32379512del',\n",
       "      'NG_012772.3:g.68968_69033del']},\n",
       "    'rcv': [{'accession': 'RCV000689905',\n",
       "      'clinical_significance': 'Uncertain significance',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2019-03-15',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8885_8950del (p.Leu2962_Asp2983del)',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV001018451',\n",
       "      'clinical_significance': 'Uncertain significance',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0027672',\n",
       "        'mondo': 'MONDO:0015356'},\n",
       "       'name': 'Hereditary cancer-predisposing syndrome',\n",
       "       'synonyms': ['Cancer predisposition',\n",
       "        'Neoplastic Syndromes, Hereditary',\n",
       "        'Tumor predisposition']},\n",
       "      'last_evaluated': '2019-05-28',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8885_8950del (p.Leu2962_Asp2983del)',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'GTTTATCAAGGGATGTCACAACCGTGTGGAAGTTGCGTATTGTAAGCTATTCAAAAAAAGAAAAAGA',\n",
       "    'rsid': 'rs1566252712',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 569309},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32953906_32954031del',\n",
       "   '_score': 5.0368056,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 102779,\n",
       "    'alt': 'G',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32954031, 'start': 32953906},\n",
       "    'hg38': {'end': 32379894, 'start': 32379769},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.8975_9100del',\n",
       "      'LRG_293t1:c.8975_9100del',\n",
       "      'NM_000059.3:c.8975_9100del',\n",
       "      'NM_000059.3:c.8975_9100del',\n",
       "      'NM_000059.3:c.8975_9100del126',\n",
       "      'NM_000059.3:c.8975_9100del126',\n",
       "      'NM_000059.3:c.8975_9100delCATCATCAGATTTATATTCTCTGTTAACAGAAGGAAAGAGATACAGAATTTATCATCTTGCAACTTCAAAATCTAAAAGTAAATCTGAAAGAGCTAACATACAGTTAGCAGCGACAAAAAAAACTC'],\n",
       "     'genomic': ['LRG_293:g.69292_69417del',\n",
       "      'NC_000013.10:g.32953908_32954033del',\n",
       "      'NC_000013.11:g.32379771_32379896del',\n",
       "      'NG_012772.3:g.69292_69417del']},\n",
       "    'rcv': [{'accession': 'RCV000082997',\n",
       "      'clinical_significance': 'Conflicting interpretations of pathogenicity',\n",
       "      'conditions': {'identifiers': {'medgen': 'C2675520',\n",
       "        'mondo': 'MONDO:0012933',\n",
       "        'omim': '612555',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Breast-ovarian cancer, familial 2 (BROVCA2)',\n",
       "       'synonyms': ['BREAST CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'Breast cancer, familial 2']},\n",
       "      'last_evaluated': '2013-06-26',\n",
       "      'number_submitters': 2,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8975_9100del (p.Pro2992_Thr3033del)',\n",
       "      'review_status': 'no assertion criteria provided'},\n",
       "     {'accession': 'RCV000522717',\n",
       "      'clinical_significance': 'Likely pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'CN517202'},\n",
       "       'name': 'not provided'},\n",
       "      'last_evaluated': '2017-03-14',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8975_9100del (p.Pro2992_Thr3033del)',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV000566941',\n",
       "      'clinical_significance': 'Likely pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0027672',\n",
       "        'mondo': 'MONDO:0015356'},\n",
       "       'name': 'Hereditary cancer-predisposing syndrome',\n",
       "       'synonyms': ['Cancer predisposition',\n",
       "        'Neoplastic Syndromes, Hereditary',\n",
       "        'Tumor predisposition']},\n",
       "      'last_evaluated': '2020-04-07',\n",
       "      'number_submitters': 2,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8975_9100del (p.Pro2992_Thr3033del)',\n",
       "      'review_status': 'criteria provided, multiple submitters, no conflicts'},\n",
       "     {'accession': 'RCV001070605',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2019-12-18',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.8975_9100del (p.Pro2992_Thr3033del)',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'GTCCATCATCAGATTTATATTCTCTGTTAACAGAAGGAAAGAGATACAGAATTTATCATCTTGCAACTTCAAAATCTAAAAGTAAATCTGAAAGAGCTAACATACAGTTAGCAGCGACAAAAAAAAC',\n",
       "    'rsid': 'rs80359736',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 96876},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32900363_32900364dup',\n",
       "   '_score': 4.728153,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 131656,\n",
       "    'alt': 'CT',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32900364, 'start': 32900363},\n",
       "    'hg38': {'end': 32326227, 'start': 32326226},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.476-9dup',\n",
       "      'NM_000059.3:c.476-9_476-8insT',\n",
       "      'NM_000059.3:c.476-9dup',\n",
       "      'NM_000059.3:c.476-9dup',\n",
       "      'NM_000059.3:c.476-9dupT',\n",
       "      'NM_000059.3:c.476-9dupT'],\n",
       "     'genomic': ['LRG_293:g.15754dup',\n",
       "      'NC_000013.10:g.32900370dup',\n",
       "      'NC_000013.11:g.32326233dup',\n",
       "      'NG_012772.3:g.15754dup']},\n",
       "    'rcv': [{'accession': 'RCV000113650',\n",
       "      'clinical_significance': 'Conflicting interpretations of pathogenicity',\n",
       "      'conditions': {'identifiers': {'medgen': 'C2675520',\n",
       "        'mondo': 'MONDO:0012933',\n",
       "        'omim': '612555',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Breast-ovarian cancer, familial 2 (BROVCA2)',\n",
       "       'synonyms': ['BREAST CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'Breast cancer, familial 2']},\n",
       "      'last_evaluated': '2016-04-21',\n",
       "      'number_submitters': 3,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.476-9dup',\n",
       "      'review_status': 'criteria provided, conflicting interpretations'},\n",
       "     {'accession': 'RCV000205599',\n",
       "      'clinical_significance': 'Benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2019-12-27',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.476-9dup',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV000478910',\n",
       "      'clinical_significance': 'Likely benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'CN169374'},\n",
       "       'name': 'not specified',\n",
       "       'synonyms': ['AllHighlyPenetrant']},\n",
       "      'last_evaluated': '2019-05-14',\n",
       "      'number_submitters': 3,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.476-9dup',\n",
       "      'review_status': 'criteria provided, multiple submitters, no conflicts'},\n",
       "     {'accession': 'RCV000580196',\n",
       "      'clinical_significance': 'Likely benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0027672',\n",
       "        'mondo': 'MONDO:0015356'},\n",
       "       'name': 'Hereditary cancer-predisposing syndrome',\n",
       "       'synonyms': ['Cancer predisposition',\n",
       "        'Neoplastic Syndromes, Hereditary',\n",
       "        'Tumor predisposition']},\n",
       "      'last_evaluated': '2015-02-23',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.476-9dup',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'C',\n",
       "    'rsid': 'rs276174849',\n",
       "    'type': 'Duplication',\n",
       "    'variant_id': 126118},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32893236_32893250del',\n",
       "   '_score': 4.7103696,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 234593,\n",
       "    'alt': 'A',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32893250, 'start': 32893236},\n",
       "    'hg38': {'end': 32319113, 'start': 32319099},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.92_106del',\n",
       "      'NM_000059.3:c.92_106del',\n",
       "      'NM_000059.3:c.92_106del15',\n",
       "      'NM_000059.3:c.92_106del15',\n",
       "      'NM_000059.3:c.92_106delGGTTTGAAGAACTTT'],\n",
       "     'genomic': ['LRG_293:g.8622_8636del',\n",
       "      'NC_000013.10:g.32893238_32893252del',\n",
       "      'NC_000013.11:g.32319101_32319115del',\n",
       "      'NG_012772.3:g.8622_8636del',\n",
       "      'NG_017006.2:g.1251_1265del']},\n",
       "    'rcv': [{'accession': 'RCV000222478',\n",
       "      'clinical_significance': 'Uncertain significance',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0027672',\n",
       "        'mondo': 'MONDO:0015356'},\n",
       "       'name': 'Hereditary cancer-predisposing syndrome',\n",
       "       'synonyms': ['Cancer predisposition',\n",
       "        'Neoplastic Syndromes, Hereditary',\n",
       "        'Tumor predisposition']},\n",
       "      'last_evaluated': '2018-02-27',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.92_106del (p.Trp31_Leu35del)',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV000814580',\n",
       "      'clinical_significance': 'Uncertain significance',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2018-10-04',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.92_106del (p.Trp31_Leu35del)',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'ATTGGTTTGAAGAACT',\n",
       "    'rsid': 'rs876658407',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 230129},\n",
       "   'observed': True},\n",
       "  {'_id': 'NM_000059.3:c.8633_8755del',\n",
       "   '_score': 4.6906223,\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 226214,\n",
       "    'coding_hgvs_only': True,\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.8633_8755del',\n",
       "      'NM_000059.3:c.8633_8755del']},\n",
       "    'rcv': {'accession': 'RCV000240780',\n",
       "     'clinical_significance': 'Pathogenic',\n",
       "     'conditions': {'identifiers': {'human_phenotype_ontology': 'HP:0010623',\n",
       "       'medgen': 'C1458155',\n",
       "       'mesh': 'D001943'},\n",
       "      'name': 'Neoplasm of the breast',\n",
       "      'synonyms': ['Breast cancer',\n",
       "       'Breast tumor',\n",
       "       'Neoplasia of the breast',\n",
       "       'Neoplasm of breast',\n",
       "       'Tumours of the breast']},\n",
       "     'last_evaluated': '2015-11-01',\n",
       "     'number_submitters': 1,\n",
       "     'origin': 'germline',\n",
       "     'preferred_name': 'NM_000059.3(BRCA2):c.8633_8755del',\n",
       "     'review_status': 'criteria provided, single submitter'},\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 224446},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32971286del',\n",
       "   '_score': 4.465083,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 67547,\n",
       "    'alt': 'A',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32971286, 'start': 32971286},\n",
       "    'hg38': {'end': 32397149, 'start': 32397149},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.9648+106del',\n",
       "      'NM_000059.3:c.9648+105del',\n",
       "      'NM_000059.3:c.9648+105delT',\n",
       "      'NM_000059.3:c.9648+106del',\n",
       "      'NM_000059.3:c.9648+106delT'],\n",
       "     'genomic': ['LRG_293:g.86671del',\n",
       "      'NC_000013.10:g.32971287del',\n",
       "      'NC_000013.11:g.32397150del',\n",
       "      'NG_012772.3:g.86671del']},\n",
       "    'rcv': [{'accession': 'RCV000114140',\n",
       "      'clinical_significance': 'Benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'C2675520',\n",
       "        'mondo': 'MONDO:0012933',\n",
       "        'omim': '612555',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Breast-ovarian cancer, familial 2 (BROVCA2)',\n",
       "       'synonyms': ['BREAST CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'Breast cancer, familial 2']},\n",
       "      'last_evaluated': '2015-01-12',\n",
       "      'number_submitters': 2,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.9648+106del',\n",
       "      'review_status': 'reviewed by expert panel'},\n",
       "     {'accession': 'RCV000205892',\n",
       "      'clinical_significance': 'Benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2016-03-19',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.9648+106del',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV000501221',\n",
       "      'clinical_significance': 'Likely benign',\n",
       "      'conditions': {'identifiers': {'medgen': 'CN169374'},\n",
       "       'name': 'not specified',\n",
       "       'synonyms': ['AllHighlyPenetrant']},\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.9648+106del',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'AT',\n",
       "    'rsid': 'rs11571824',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 52879},\n",
       "   'geno2mp': {'_license': 'http://bit.ly/2QyGCIN', 'hpo_count': 8},\n",
       "   'gnomad_genome': {'_license': 'http://bit.ly/2I1cl1I',\n",
       "    'ac': {'ac': 202,\n",
       "     'ac_afr': 198,\n",
       "     'ac_afr_female': 77,\n",
       "     'ac_afr_male': 121,\n",
       "     'ac_amr': 1,\n",
       "     'ac_amr_female': 0,\n",
       "     'ac_amr_male': 1,\n",
       "     'ac_asj': 0,\n",
       "     'ac_asj_female': 0,\n",
       "     'ac_asj_male': 0,\n",
       "     'ac_eas': 0,\n",
       "     'ac_eas_female': 0,\n",
       "     'ac_eas_male': 0,\n",
       "     'ac_female': 79,\n",
       "     'ac_fin': 0,\n",
       "     'ac_fin_female': 0,\n",
       "     'ac_fin_male': 0,\n",
       "     'ac_male': 123,\n",
       "     'ac_nfe': 1,\n",
       "     'ac_nfe_est': 0,\n",
       "     'ac_nfe_female': 1,\n",
       "     'ac_nfe_male': 0,\n",
       "     'ac_nfe_nwe': 1,\n",
       "     'ac_nfe_onf': 0,\n",
       "     'ac_nfe_seu': 0,\n",
       "     'ac_oth': 2,\n",
       "     'ac_oth_female': 1,\n",
       "     'ac_oth_male': 1,\n",
       "     'ac_popmax': 198,\n",
       "     'ac_raw': 202},\n",
       "    'af': {'af': 0.00643189,\n",
       "     'af_afr': 0.0227325,\n",
       "     'af_afr_female': 0.0209125,\n",
       "     'af_afr_male': 0.0240652,\n",
       "     'af_amr': 0.00117925,\n",
       "     'af_amr_female': 0.0,\n",
       "     'af_amr_male': 0.00230415,\n",
       "     'af_asj': 0.0,\n",
       "     'af_asj_female': 0.0,\n",
       "     'af_asj_male': 0.0,\n",
       "     'af_eas': 0.0,\n",
       "     'af_eas_female': 0.0,\n",
       "     'af_eas_male': 0.0,\n",
       "     'af_female': 0.0056704,\n",
       "     'af_fin': 0.0,\n",
       "     'af_fin_female': 0.0,\n",
       "     'af_fin_male': 0.0,\n",
       "     'af_male': 0.00703903,\n",
       "     'af_nfe': 6.47836e-05,\n",
       "     'af_nfe_est': 0.0,\n",
       "     'af_nfe_female': 0.0001468,\n",
       "     'af_nfe_male': 0.0,\n",
       "     'af_nfe_nwe': 0.000116306,\n",
       "     'af_nfe_onf': 0.0,\n",
       "     'af_nfe_seu': 0.0,\n",
       "     'af_oth': 0.00183824,\n",
       "     'af_oth_female': 0.00177936,\n",
       "     'af_oth_male': 0.00190114,\n",
       "     'af_popmax': 0.0227325,\n",
       "     'af_raw': 0.00642984},\n",
       "    'alleles': 'A',\n",
       "    'alt': 'A',\n",
       "    'an': {'an': 31406,\n",
       "     'an_afr': 8710,\n",
       "     'an_afr_female': 3682,\n",
       "     'an_afr_male': 5028,\n",
       "     'an_amr': 848,\n",
       "     'an_amr_female': 414,\n",
       "     'an_amr_male': 434,\n",
       "     'an_asj': 290,\n",
       "     'an_asj_female': 80,\n",
       "     'an_asj_male': 210,\n",
       "     'an_eas': 1560,\n",
       "     'an_eas_female': 538,\n",
       "     'an_eas_male': 1022,\n",
       "     'an_female': 13932,\n",
       "     'an_fin': 3474,\n",
       "     'an_fin_female': 1844,\n",
       "     'an_fin_male': 1630,\n",
       "     'an_male': 17474,\n",
       "     'an_nfe': 15436,\n",
       "     'an_nfe_est': 4594,\n",
       "     'an_nfe_female': 6812,\n",
       "     'an_nfe_male': 8624,\n",
       "     'an_nfe_nwe': 8598,\n",
       "     'an_nfe_onf': 2138,\n",
       "     'an_nfe_seu': 106,\n",
       "     'an_oth': 1088,\n",
       "     'an_oth_female': 562,\n",
       "     'an_oth_male': 526,\n",
       "     'an_popmax': 8710,\n",
       "     'an_raw': 31416},\n",
       "    'baseqranksum': 0.015,\n",
       "    'chrom': '13',\n",
       "    'clippingranksum': -0.055,\n",
       "    'fs': 0.0,\n",
       "    'hom': {'hom': 1,\n",
       "     'hom_afr': 1,\n",
       "     'hom_afr_female': 0,\n",
       "     'hom_afr_male': 1,\n",
       "     'hom_amr': 0,\n",
       "     'hom_amr_female': 0,\n",
       "     'hom_amr_male': 0,\n",
       "     'hom_asj': 0,\n",
       "     'hom_asj_female': 0,\n",
       "     'hom_asj_male': 0,\n",
       "     'hom_eas': 0,\n",
       "     'hom_eas_female': 0,\n",
       "     'hom_eas_male': 0,\n",
       "     'hom_female': 0,\n",
       "     'hom_fin': 0,\n",
       "     'hom_fin_female': 0,\n",
       "     'hom_fin_male': 0,\n",
       "     'hom_male': 1,\n",
       "     'hom_nfe': 0,\n",
       "     'hom_nfe_est': 0,\n",
       "     'hom_nfe_female': 0,\n",
       "     'hom_nfe_male': 0,\n",
       "     'hom_nfe_nwe': 0,\n",
       "     'hom_nfe_onf': 0,\n",
       "     'hom_nfe_seu': 0,\n",
       "     'hom_oth': 0,\n",
       "     'hom_oth_female': 0,\n",
       "     'hom_oth_male': 0,\n",
       "     'hom_popmax': 1,\n",
       "     'hom_raw': 1},\n",
       "    'inbreedingcoeff': 0.0116,\n",
       "    'mq': {'mq': 60.0, 'mqranksum': 0.037},\n",
       "    'pos': 32971285,\n",
       "    'qd': 14.68,\n",
       "    'readposranksum': 0.082,\n",
       "    'ref': 'AT',\n",
       "    'rsid': 'rs11571824',\n",
       "    'type': 'del',\n",
       "    'vqslod': 5.46,\n",
       "    'vqsr_culprit': 'FS'},\n",
       "   'hg19': {'end': 32971287, 'start': 32971286},\n",
       "   'observed': True,\n",
       "   'snpeff': {'_license': 'http://bit.ly/2suyRKt',\n",
       "    'ann': [{'distance_to_feature': '3574',\n",
       "      'effect': 'downstream_gene_variant',\n",
       "      'feature_id': 'NM_001286459.1',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'N4BP2L1',\n",
       "      'genename': 'N4BP2L1',\n",
       "      'hgvs_c': 'c.*5793del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'transcript_biotype': 'protein_coding'},\n",
       "     {'distance_to_feature': '3574',\n",
       "      'effect': 'downstream_gene_variant',\n",
       "      'feature_id': 'NM_001079691.1',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'N4BP2L1',\n",
       "      'genename': 'N4BP2L1',\n",
       "      'hgvs_c': 'c.*5995del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'transcript_biotype': 'protein_coding'},\n",
       "     {'distance_to_feature': '3574',\n",
       "      'effect': 'downstream_gene_variant',\n",
       "      'feature_id': 'NM_001286460.1',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'N4BP2L1',\n",
       "      'genename': 'N4BP2L1',\n",
       "      'hgvs_c': 'c.*4723del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'transcript_biotype': 'protein_coding'},\n",
       "     {'distance_to_feature': '3574',\n",
       "      'effect': 'downstream_gene_variant',\n",
       "      'feature_id': 'NM_001286461.1',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'N4BP2L1',\n",
       "      'genename': 'N4BP2L1',\n",
       "      'hgvs_c': 'c.*4731del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'transcript_biotype': 'protein_coding'},\n",
       "     {'distance_to_feature': '3574',\n",
       "      'effect': 'downstream_gene_variant',\n",
       "      'feature_id': 'NM_052818.2',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'N4BP2L1',\n",
       "      'genename': 'N4BP2L1',\n",
       "      'hgvs_c': 'c.*5793del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'transcript_biotype': 'protein_coding'},\n",
       "     {'effect': 'intron_variant',\n",
       "      'feature_id': 'NM_000059.3',\n",
       "      'feature_type': 'transcript',\n",
       "      'gene_id': 'BRCA2',\n",
       "      'genename': 'BRCA2',\n",
       "      'hgvs_c': 'c.9648+106del',\n",
       "      'putative_impact': 'MODIFIER',\n",
       "      'rank': '26',\n",
       "      'total': '26',\n",
       "      'transcript_biotype': 'protein_coding'}]},\n",
       "   'vcf': {'alt': 'A', 'position': '32971285', 'ref': 'AT'}},\n",
       "  {'_id': 'chr13:g.32893215_32893240del',\n",
       "   '_score': 4.390247,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 131531,\n",
       "    'alt': 'A',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32893240, 'start': 32893215},\n",
       "    'hg38': {'end': 32319103, 'start': 32319078},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.71_96del',\n",
       "      'LRG_293t1:c.71_96del',\n",
       "      'NM_000059.3:c.71_96del',\n",
       "      'NM_000059.3:c.71_96del26',\n",
       "      'NM_000059.3:c.71_96del26',\n",
       "      'NM_000059.3:c.71_96delTAGGACCAATAAGTCTTAATTGGTTT'],\n",
       "     'genomic': ['LRG_293:g.8601_8626del',\n",
       "      'NC_000013.10:g.32893217_32893242del',\n",
       "      'NC_000013.11:g.32319080_32319105del',\n",
       "      'NG_012772.3:g.8601_8626del',\n",
       "      'NG_017006.2:g.1261_1286del']},\n",
       "    'rcv': [{'accession': 'RCV000113090',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C2675520',\n",
       "        'mondo': 'MONDO:0012933',\n",
       "        'omim': '612555',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Breast-ovarian cancer, familial 2 (BROVCA2)',\n",
       "       'synonyms': ['BREAST CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'Breast cancer, familial 2']},\n",
       "      'last_evaluated': '2016-09-08',\n",
       "      'number_submitters': 3,\n",
       "      'origin': 'unknown',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.71_96del (p.Asp23_Leu24insTer)',\n",
       "      'review_status': 'reviewed by expert panel'},\n",
       "     {'accession': 'RCV000509918',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0027672',\n",
       "        'mondo': 'MONDO:0015356'},\n",
       "       'name': 'Hereditary cancer-predisposing syndrome',\n",
       "       'synonyms': ['Cancer predisposition',\n",
       "        'Neoplastic Syndromes, Hereditary',\n",
       "        'Tumor predisposition']},\n",
       "      'last_evaluated': '2016-07-09',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.71_96del (p.Asp23_Leu24insTer)',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'ATTTAGGACCAATAAGTCTTAATTGGT',\n",
       "    'rsid': 'rs80359637',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 125993},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32907399_32907417del',\n",
       "   '_score': 4.38837,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 536421,\n",
       "    'alt': 'C',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32907417, 'start': 32907399},\n",
       "    'hg38': {'end': 32333280, 'start': 32333262},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.1785_1803del',\n",
       "      'NM_000059.3:c.1785_1803del',\n",
       "      'NM_000059.3:c.1785_1803del19',\n",
       "      'NM_000059.3:c.1785_1803delTGATGAAACATCTTATAAA'],\n",
       "     'genomic': ['LRG_293:g.22784_22802del',\n",
       "      'NC_000013.10:g.32907400_32907418del',\n",
       "      'NC_000013.11:g.32333263_32333281del',\n",
       "      'NG_012772.3:g.22784_22802del']},\n",
       "    'rcv': [{'accession': 'RCV000657314',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'CN517202'},\n",
       "       'name': 'not provided'},\n",
       "      'last_evaluated': '2017-04-13',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.1785_1803del (p.His595fs)',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV000817436',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2019-11-22',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.1785_1803del (p.His595fs)',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'ref': 'CATGATGAAACATCTTATAA',\n",
       "    'rsid': 'rs1555282073',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 545781},\n",
       "   'observed': True},\n",
       "  {'_id': 'NM_000059.3:c.3407_3408insAlu',\n",
       "   '_score': 4.3596797,\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 789252,\n",
       "    'coding_hgvs_only': True,\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.3407_3408insAlu',\n",
       "      'NM_000059.3:c.3407_3408insAlu']},\n",
       "    'rcv': [{'accession': 'RCV000984804',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'CN517202'},\n",
       "       'name': 'not provided'},\n",
       "      'last_evaluated': '2019-07-24',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3:c.3407_3408insAlu',\n",
       "      'review_status': 'criteria provided, single submitter'},\n",
       "     {'accession': 'RCV001089838',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2019-10-02',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3:c.3407_3408insAlu',\n",
       "      'review_status': 'criteria provided, single submitter'}],\n",
       "    'type': 'Insertion',\n",
       "    'variant_id': 800654},\n",
       "   'observed': True},\n",
       "  {'_id': 'chr13:g.32907325_32907329del',\n",
       "   '_score': 4.3512483,\n",
       "   'chrom': '13',\n",
       "   'clinvar': {'_license': 'http://bit.ly/2SQdcI0',\n",
       "    'allele_id': 65846,\n",
       "    'alt': 'A',\n",
       "    'chrom': '13',\n",
       "    'cytogenic': '13q13.1',\n",
       "    'gene': {'id': '675', 'symbol': 'BRCA2'},\n",
       "    'hg19': {'end': 32907329, 'start': 32907325},\n",
       "    'hg38': {'end': 32333192, 'start': 32333188},\n",
       "    'hgvs': {'coding': ['LRG_293t1:c.1711_1715del',\n",
       "      'NM_000059.3:c.1710_1714delTTCTG',\n",
       "      'NM_000059.3:c.1711_1715del',\n",
       "      'NM_000059.3:c.1711_1715delTCTGT'],\n",
       "     'genomic': ['LRG_293:g.22710_22714del',\n",
       "      'NC_000013.10:g.32907326_32907330del',\n",
       "      'NC_000013.11:g.32333189_32333193del',\n",
       "      'NG_012772.3:g.22710_22714del']},\n",
       "    'rcv': [{'accession': 'RCV000496686',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C0677776',\n",
       "        'mesh': 'D061325',\n",
       "        'mondo': 'MONDO:0003582',\n",
       "        'omim': 'PS604370',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Hereditary breast and ovarian cancer syndrome (HBOC)',\n",
       "       'synonyms': ['Hereditary breast and ovarian cancer']},\n",
       "      'last_evaluated': '2014-01-31',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.1711_1715del (p.Val572fs)',\n",
       "      'review_status': 'no assertion criteria provided'},\n",
       "     {'accession': 'RCV000661545',\n",
       "      'clinical_significance': 'Pathogenic',\n",
       "      'conditions': {'identifiers': {'medgen': 'C2675520',\n",
       "        'mondo': 'MONDO:0012933',\n",
       "        'omim': '612555',\n",
       "        'orphanet': '145'},\n",
       "       'name': 'Breast-ovarian cancer, familial 2 (BROVCA2)',\n",
       "       'synonyms': ['BREAST CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'BREAST-OVARIAN CANCER, FAMILIAL, SUSCEPTIBILITY TO, 2',\n",
       "        'Breast cancer, familial 2']},\n",
       "      'last_evaluated': '2017-12-15',\n",
       "      'number_submitters': 1,\n",
       "      'origin': 'germline',\n",
       "      'preferred_name': 'NM_000059.3(BRCA2):c.1711_1715del (p.Val572fs)',\n",
       "      'review_status': 'reviewed by expert panel'}],\n",
       "    'ref': 'ATTCTG',\n",
       "    'rsid': 'rs397507602',\n",
       "    'type': 'Deletion',\n",
       "    'variant_id': 51178},\n",
       "   'observed': True}]}"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = requests.get(\"https://myvariant.info/v1/query?q=NM_000059.3\")\n",
    "data = data.json()\n",
    "data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Build out our Custom Gene Panel\n",
    "\n",
    "This assumes that you have some information about your gene of interest. If you don't, its best to go to https://gnomad.broadinstitute.org/ and search for your gene under **gnomad v2.1.1**."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "refseq = \"NM_000059.3\"\n",
    "geneSymbol = \"BRCA2\"\n",
    "HGCN = 1101\n",
    "Omim_gene_entry = 600185\n",
    "geneAlias = \"FAD,FAD1,BRCC2,XRCC11\"\n",
    "eGeneID = \"ENSG00000139618\"\n",
    "eTranscriptID = \"ENST00000544455\"\n",
    "geneSymbol = \"BRCA2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "#[\"#chromosome\",\n",
    "#\"txStart\", \"txEnd\",\n",
    "#\"refseq\", \"score\", \"strand\", \n",
    "#\"geneSymbol\", \"HGNC\", \n",
    "#\"Omim gene entry\", \n",
    "#\"geneAlias\", \n",
    "#\"eGeneID\", \n",
    "#\"eTranscriptID\", \n",
    "#\"cdsStart\", \"cdsEnd\", \n",
    "#\"exonsStarts\", \"exonEnds\"]\n",
    "\n",
    "custom_gene_panel_data = {\n",
    "    \"#chromosome\": chr,\n",
    "    \"strand\": strand,\n",
    "    \"score\": 0, #?\n",
    "    \"refseq\": refseq,\n",
    "    \"HGNC\": HGCN,\n",
    "    \"Omim gene entry\": Omim_gene_entry,\n",
    "    \"geneSymbol\": geneSymbol,\n",
    "    \"geneAlias\": geneAlias,\n",
    "    \"eGeneID\": eGeneID,\n",
    "    \"eTranscriptID\": eTranscriptID,\n",
    "    \"cdsStart\": mygene_cdsstart,\n",
    "    \"cdsEnd\": mygene_cdsend,\n",
    "    \"exonsStarts\": ','.join(np.array(mygene_exon_starts, dtype=str)),\n",
    "    \"exonEnds\": ','.join(np.array(mygene_exon_ends, dtype=str)),\n",
    "    \"txStart\" : mygene_txstart,\n",
    "    \"txEnd\" : mygene_txend\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compare the Ella Gene Panel with MyGene.Info\n",
    "\n",
    "You can see here that the data that we get from MyGene.Info is slightly off from the data we get from Ella, but in the same ballpark."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chromosome</th>\n",
       "      <th>txStart</th>\n",
       "      <th>txEnd</th>\n",
       "      <th>refseq</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>HGNC</th>\n",
       "      <th>Omim gene entry</th>\n",
       "      <th>geneAlias</th>\n",
       "      <th>eGeneID</th>\n",
       "      <th>eTranscriptID</th>\n",
       "      <th>cdsStart</th>\n",
       "      <th>cdsEnd</th>\n",
       "      <th>exonsStarts</th>\n",
       "      <th>exonEnds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>13</td>\n",
       "      <td>32889644</td>\n",
       "      <td>32974405</td>\n",
       "      <td>NM_000059.3</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>1101</td>\n",
       "      <td>600185</td>\n",
       "      <td>FAD,FAD1,BRCC2,XRCC11</td>\n",
       "      <td>ENSG00000139618</td>\n",
       "      <td>ENST00000544455</td>\n",
       "      <td>32890597</td>\n",
       "      <td>32972907</td>\n",
       "      <td>32889644,32890558,32893213,32899212,32900237,3...</td>\n",
       "      <td>32889804,32890664,32893462,32899321,32900287,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #chromosome   txStart     txEnd       refseq  score strand geneSymbol  HGNC  \\\n",
       "0          13  32889644  32974405  NM_000059.3      0      +      BRCA2  1101   \n",
       "\n",
       "   Omim gene entry              geneAlias          eGeneID    eTranscriptID  \\\n",
       "0           600185  FAD,FAD1,BRCC2,XRCC11  ENSG00000139618  ENST00000544455   \n",
       "\n",
       "   cdsStart    cdsEnd                                        exonsStarts  \\\n",
       "0  32890597  32972907  32889644,32890558,32893213,32899212,32900237,3...   \n",
       "\n",
       "                                            exonEnds  \n",
       "0  32889804,32890664,32893462,32899321,32900287,3...  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Genepanel we create from mygene.info\n",
    "df = pd.DataFrame(columns = columns, data=[custom_gene_panel_data])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chromosome</th>\n",
       "      <th>txStart</th>\n",
       "      <th>txEnd</th>\n",
       "      <th>refseq</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>geneSymbol</th>\n",
       "      <th>HGNC</th>\n",
       "      <th>Omim gene entry</th>\n",
       "      <th>geneAlias</th>\n",
       "      <th>eGeneID</th>\n",
       "      <th>eTranscriptID</th>\n",
       "      <th>cdsStart</th>\n",
       "      <th>cdsEnd</th>\n",
       "      <th>exonsStarts</th>\n",
       "      <th>exonEnds</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>17</td>\n",
       "      <td>41196311</td>\n",
       "      <td>41277468</td>\n",
       "      <td>NM_007297.3</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>1100</td>\n",
       "      <td>113705</td>\n",
       "      <td>RNF53,BRCC1,PPP1R53,FANCS</td>\n",
       "      <td>ENSG00000012048</td>\n",
       "      <td>ENST00000309486</td>\n",
       "      <td>41197694</td>\n",
       "      <td>41258543</td>\n",
       "      <td>41196311,41199659,41201137,41203079,41209068,4...</td>\n",
       "      <td>41197819,41199720,41201211,41203134,41209152,4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>13</td>\n",
       "      <td>32889616</td>\n",
       "      <td>32973809</td>\n",
       "      <td>NM_000059.3</td>\n",
       "      <td>0</td>\n",
       "      <td>+</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>1101</td>\n",
       "      <td>600185</td>\n",
       "      <td>FAD,FAD1,BRCC2,XRCC11</td>\n",
       "      <td>ENSG00000139618</td>\n",
       "      <td>ENST00000544455</td>\n",
       "      <td>32890597</td>\n",
       "      <td>32972907</td>\n",
       "      <td>32889616,32890558,32893213,32899212,32900237,3...</td>\n",
       "      <td>32889804,32890664,32893462,32899321,32900287,3...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #chromosome   txStart     txEnd       refseq  score strand geneSymbol  \\\n",
       "0           17  41196311  41277468  NM_007297.3      0      -      BRCA1   \n",
       "1           13  32889616  32973809  NM_000059.3      0      +      BRCA2   \n",
       "\n",
       "   HGNC  Omim gene entry                  geneAlias          eGeneID  \\\n",
       "0  1100           113705  RNF53,BRCC1,PPP1R53,FANCS  ENSG00000012048   \n",
       "1  1101           600185      FAD,FAD1,BRCC2,XRCC11  ENSG00000139618   \n",
       "\n",
       "     eTranscriptID  cdsStart    cdsEnd  \\\n",
       "0  ENST00000309486  41197694  41258543   \n",
       "1  ENST00000544455  32890597  32972907   \n",
       "\n",
       "                                         exonsStarts  \\\n",
       "0  41196311,41199659,41201137,41203079,41209068,4...   \n",
       "1  32889616,32890558,32893213,32899212,32900237,3...   \n",
       "\n",
       "                                            exonEnds  \n",
       "0  41197819,41199720,41201211,41203134,41209152,4...  \n",
       "1  32889804,32890664,32893462,32899321,32900287,3...  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Ella genepanel\n",
    "genepanel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Write out the gene panel name into the Ella Transcripts File Definition\n",
    "\n",
    "If you wanted to write out the Ella Transcripts file you would first write out the info lines, then you could use pandas to write out the tabular data with no index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "genepanel_name = 'BRCA2'\n",
    "genepanel_version = 'v01'\n",
    "\n",
    "info_line = \"# Genepanel: {} Version: {} Date: 2020-10-09\\n\".format(genepanel_name, genepanel_version)\n",
    "genepanel_transcripts_file_name = \"{name}_{version}.transcripts.csv\".format(name=genepanel_name,\n",
    "                                                                             version=genepanel_version)\n",
    "\n",
    "with open(os.path.join(genepanel_transcripts_file_name), 'w') as fp:\n",
    "    fp.write(info_line)\n",
    "    df.to_csv(fp, index=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then the `genepanel_transcripts_file_name` will have the correct output for the transcripts. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add the HBOC_v01 Gene Panel to the Ella Database\n",
    "\n",
    "The HBOC_v01 gene panel is included with the Ella codebase and is in the ella docker container.\n",
    "\n",
    "```bash\n",
    "docker run -it dabbleofdevops/ella:1.11.1 bash -c \"ella-cli deposit genepanel \\\n",
    "    --genepanel_name HBOC \\\n",
    "    --genepanel_version v01 \\\n",
    "    --transcripts_path /ella/src/vardb/testdata/clinicalGenePanels/HBOC_v01/HBOC_v01.transcripts.csv\n",
    "    --phenotypes_path /ella/src/vardb/testdata/clinicalGenePanels/HBOC_v01/HBOC_v01.phenotypes.csv\"\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
