{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate Report From the ELLA REST API\n",
    "\n",
    "Let's load in the data we fetched from `Creating-Custom-Reports/Parse-Analysis-From-REST-API.ipynb`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import os\n",
    "import copy\n",
    "from pprint import pprint\n",
    "\n",
    "## All data will be loaded in asyncronously from REST APIs\n",
    "\n",
    "with open('/data/analysis_rest_api_data.json', 'r') as reader:\n",
    "    analysis_data = json.load(reader)\n",
    "\n",
    "with open('/data/mock_patient_data.json', 'r') as reader:\n",
    "    mock_patient_data = json.load(reader)\n",
    "    \n",
    "acmg_codes = {\n",
    "    '1': 'Benign',\n",
    "    '2': 'Likely Benign',\n",
    "    '3': 'Variant of Uncertain Significance',\n",
    "    '4': 'Likely Pathogenic',\n",
    "    '5': 'Pathogenic',\n",
    "    'U': 'Unknown'\n",
    "}\n",
    "ranks = ['U', '1', '2', '3', '4', '5']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! pip install tabulate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'allele_assessment': {'allele_id': 3,\n",
       "  'analysis_id': 2,\n",
       "  'annotation_id': 3,\n",
       "  'attachment_ids': [],\n",
       "  'classification': '4',\n",
       "  'custom_annotation_id': None,\n",
       "  'date_created': '2020-10-19T10:49:44.630277+00:00',\n",
       "  'date_superceeded': None,\n",
       "  'evaluation': {'acmg': {'included': [{'code': 'PVS1',\n",
       "      'comment': '',\n",
       "      'match': None,\n",
       "      'op': None,\n",
       "      'source': 'suggested',\n",
       "      'uuid': '5104d287-683d-4391-a75b-5694c65f2b88'},\n",
       "     {'code': 'PPxPM2',\n",
       "      'comment': '',\n",
       "      'match': None,\n",
       "      'op': None,\n",
       "      'source': 'suggested',\n",
       "      'uuid': 'b44315f8-2db9-4605-b824-acf51e50ac31'}],\n",
       "    'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
       "      'match': ['LENI'],\n",
       "      'op': '$in',\n",
       "      'source': 'genepanel.last_exon_important'},\n",
       "     {'code': 'REQ_GP_LOF_missense',\n",
       "      'match': ['ANY'],\n",
       "      'op': '$in',\n",
       "      'source': 'genepanel.disease_mode'},\n",
       "     {'code': 'REQ_not_in_last_exon',\n",
       "      'match': ['no'],\n",
       "      'op': '$in',\n",
       "      'source': 'transcript.in_last_exon'},\n",
       "     {'code': 'REQ_null_variant',\n",
       "      'match': ['stop_gained'],\n",
       "      'op': '$in',\n",
       "      'source': 'transcript.consequences'},\n",
       "     {'code': 'REQ_no_freq',\n",
       "      'match': ['null_freq'],\n",
       "      'op': '$in',\n",
       "      'source': 'frequencies.commonness'},\n",
       "     {'code': 'PVS1', 'match': None, 'op': None, 'source': None},\n",
       "     {'code': 'PPxPM2', 'match': None, 'op': None, 'source': None}],\n",
       "    'suggested_classification': 4},\n",
       "   'classification': {'comment': ''},\n",
       "   'external': {'comment': ''},\n",
       "   'frequency': {'comment': ''},\n",
       "   'prediction': {'comment': ''},\n",
       "   'reference': {'comment': ''}},\n",
       "  'genepanel_name': 'HBOCUTV',\n",
       "  'genepanel_version': 'v01',\n",
       "  'id': 1,\n",
       "  'previous_assessment_id': None,\n",
       "  'seconds_since_update': 3447886.719649,\n",
       "  'user': {'abbrev_name': 'H. Ibsen',\n",
       "   'active': True,\n",
       "   'first_name': 'Henrik',\n",
       "   'full_name': 'Henrik Ibsen',\n",
       "   'id': 1,\n",
       "   'last_name': 'Ibsen',\n",
       "   'user_group_name': 'testgroup01',\n",
       "   'username': 'testuser1'},\n",
       "  'user_id': 1,\n",
       "  'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
       "  'usergroup_id': 1},\n",
       " 'allele_report': {'allele_id': 3,\n",
       "  'analysis_id': 2,\n",
       "  'date_created': '2020-10-19T10:49:44.640931+00:00',\n",
       "  'date_superceeded': None,\n",
       "  'evaluation': {'comment': ''},\n",
       "  'id': 1,\n",
       "  'previous_report_id': None,\n",
       "  'seconds_since_update': 3447886.733316,\n",
       "  'user': {'abbrev_name': 'H. Ibsen',\n",
       "   'active': True,\n",
       "   'first_name': 'Henrik',\n",
       "   'full_name': 'Henrik Ibsen',\n",
       "   'id': 1,\n",
       "   'last_name': 'Ibsen',\n",
       "   'user_group_name': 'testgroup01',\n",
       "   'username': 'testuser1'},\n",
       "  'user_id': 1,\n",
       "  'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
       "  'usergroup_id': 1},\n",
       " 'annotation': {'annotation_id': 3,\n",
       "  'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Pathogenic',\n",
       "      'last_evaluated': '01/11/2017',\n",
       "      'rcv': 'SCV000296805',\n",
       "      'submitter': 'GeneKor MSA',\n",
       "      'traitnames': '',\n",
       "      'variant_id': ''},\n",
       "     {'clinical_significance_descr': 'Pathogenic',\n",
       "      'last_evaluated': '18/10/2016',\n",
       "      'rcv': 'SCV000323955',\n",
       "      'submitter': 'ENIGMA',\n",
       "      'traitnames': '',\n",
       "      'variant_id': ''},\n",
       "     {'clinical_significance_descr': 'Pathogenic',\n",
       "      'last_evaluated': '18/10/2016',\n",
       "      'rcv': 'RCV000257718',\n",
       "      'submitter': '',\n",
       "      'traitnames': 'Breast-ovarian cancer, familial 2',\n",
       "      'variant_id': '51063'},\n",
       "     {'clinical_significance_descr': 'Pathogenic',\n",
       "      'last_evaluated': '01/11/2017',\n",
       "      'rcv': 'RCV000239090',\n",
       "      'submitter': '',\n",
       "      'traitnames': 'not provided',\n",
       "      'variant_id': '51063'},\n",
       "     {'clinical_significance_descr': 'Pathogenic',\n",
       "      'last_evaluated': '02/10/2015',\n",
       "      'rcv': 'SCV000326495',\n",
       "      'submitter': 'CIMBA',\n",
       "      'traitnames': '',\n",
       "      'variant_id': ''}],\n",
       "    'variant_description': 'reviewed by expert panel',\n",
       "    'variant_id': 51063},\n",
       "   'HGMD': {'acc_num': 'CM082514',\n",
       "    'codon': 4,\n",
       "    'disease': 'Breast and/or ovarian cancer',\n",
       "    'tag': 'DM'}},\n",
       "  'filtered_transcripts': ['NM_000059.3'],\n",
       "  'frequencies': {},\n",
       "  'prediction': {},\n",
       "  'references': [{'pubmed_id': 17453335,\n",
       "    'source_info': {'HGMD': 'Primary literature report. No comments.'},\n",
       "    'sources': ['HGMD']}],\n",
       "  'schema_version': 1,\n",
       "  'transcripts': [{'HGVSc': 'c.10G>T',\n",
       "    'HGVSc_short': 'c.10G>T',\n",
       "    'HGVSp': 'p.Gly4Ter',\n",
       "    'amino_acids': 'G/*',\n",
       "    'coding_region_distance': 0,\n",
       "    'codons': 'Gga/Tga',\n",
       "    'consequences': ['stop_gained'],\n",
       "    'dbsnp': ['rs397507571'],\n",
       "    'exon': '2/27',\n",
       "    'exon_distance': 0,\n",
       "    'hgnc_id': 1101,\n",
       "    'in_last_exon': 'no',\n",
       "    'is_canonical': True,\n",
       "    'protein': 'NP_000050.2',\n",
       "    'strand': 1,\n",
       "    'symbol': 'BRCA2',\n",
       "    'transcript': 'NM_000059.3'},\n",
       "   {'consequences': ['upstream_gene_variant'],\n",
       "    'dbsnp': ['rs397507571'],\n",
       "    'hgnc_id': 37116,\n",
       "    'in_last_exon': 'no',\n",
       "    'is_canonical': True,\n",
       "    'strand': -1,\n",
       "    'symbol': 'ZAR1L',\n",
       "    'transcript': 'NM_001136571.1'}],\n",
       "  'worst_consequence': ['NM_000059.3']},\n",
       " 'change_from': 'G',\n",
       " 'change_to': 'T',\n",
       " 'change_type': 'SNP',\n",
       " 'chromosome': '13',\n",
       " 'genome_reference': 'GRCh37',\n",
       " 'id': 3,\n",
       " 'open_end_position': 32890607,\n",
       " 'samples': [{'affected': True,\n",
       "   'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
       "   'family_id': None,\n",
       "   'father_id': None,\n",
       "   'genotype': {'allele_depth': {'REF (G)': 107, 'T': 80},\n",
       "    'allele_ratio': 0.42780748663101603,\n",
       "    'filter_status': 'PASS',\n",
       "    'formatted': 'G/T',\n",
       "    'genotype_quality': 99,\n",
       "    'id': 3,\n",
       "    'multiallelic': False,\n",
       "    'needs_verification': False,\n",
       "    'needs_verification_checks': {'allele_ratio': True,\n",
       "     'dp': True,\n",
       "     'hts': True,\n",
       "     'pass': True,\n",
       "     'snp': True},\n",
       "    'sequencing_depth': 187,\n",
       "    'type': 'Heterozygous',\n",
       "    'variant_quality': 5000},\n",
       "   'id': 2,\n",
       "   'identifier': 'brca_sample_1',\n",
       "   'mother_id': None,\n",
       "   'proband': True,\n",
       "   'sample_type': 'HTS',\n",
       "   'sex': None,\n",
       "   'sibling_id': None}],\n",
       " 'start_position': 32890606,\n",
       " 'tags': ['has_references'],\n",
       " 'vcf_alt': 'T',\n",
       " 'vcf_pos': 32890607,\n",
       " 'vcf_ref': 'G',\n",
       " 'filterd_transcripts': [{'HGVSc': 'c.10G>T',\n",
       "   'HGVSc_short': 'c.10G>T',\n",
       "   'HGVSp': 'p.Gly4Ter',\n",
       "   'amino_acids': 'G/*',\n",
       "   'coding_region_distance': 0,\n",
       "   'codons': 'Gga/Tga',\n",
       "   'consequences': ['stop_gained'],\n",
       "   'dbsnp': ['rs397507571'],\n",
       "   'exon': '2/27',\n",
       "   'exon_distance': 0,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'is_canonical': True,\n",
       "   'protein': 'NP_000050.2',\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'},\n",
       "  {'HGVSc': 'c.51_52delAC',\n",
       "   'HGVSc_short': 'c.51_52delAC',\n",
       "   'HGVSp': 'p.Arg18LeufsTer12',\n",
       "   'amino_acids': 'T/X',\n",
       "   'coding_region_distance': 0,\n",
       "   'codons': 'ACa/a',\n",
       "   'consequences': ['frameshift_variant'],\n",
       "   'exon': '2/27',\n",
       "   'exon_distance': 0,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'is_canonical': True,\n",
       "   'protein': 'NP_000050.2',\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'},\n",
       "  {'HGVSc': 'c.67+2T>A',\n",
       "   'HGVSc_short': 'c.67+2T>A',\n",
       "   'coding_region_distance': None,\n",
       "   'consequences': ['splice_donor_variant'],\n",
       "   'dbsnp': ['rs81002885'],\n",
       "   'exon_distance': 2,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'intron': '2/26',\n",
       "   'is_canonical': True,\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'},\n",
       "  {'HGVSc': 'c.72A>T',\n",
       "   'HGVSc_short': 'c.72A>T',\n",
       "   'HGVSp': 'p.Leu24Phe',\n",
       "   'amino_acids': 'L/F',\n",
       "   'coding_region_distance': 0,\n",
       "   'codons': 'ttA/ttT',\n",
       "   'consequences': ['missense_variant'],\n",
       "   'dbsnp': ['rs397507909'],\n",
       "   'exon': '3/27',\n",
       "   'exon_distance': 0,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'is_canonical': True,\n",
       "   'protein': 'NP_000050.2',\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'},\n",
       "  {'HGVSc': 'c.97G>T',\n",
       "   'HGVSc_short': 'c.97G>T',\n",
       "   'HGVSp': 'p.Glu33Ter',\n",
       "   'amino_acids': 'E/*',\n",
       "   'coding_region_distance': 0,\n",
       "   'codons': 'Gaa/Taa',\n",
       "   'consequences': ['stop_gained'],\n",
       "   'dbsnp': ['rs397508065'],\n",
       "   'exon': '3/27',\n",
       "   'exon_distance': 0,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'is_canonical': True,\n",
       "   'protein': 'NP_000050.2',\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'},\n",
       "  {'HGVSc': 'c.198A>G',\n",
       "   'HGVSc_short': 'c.198A>G',\n",
       "   'HGVSp': 'c.198A>G(p.=)',\n",
       "   'amino_acids': 'Q',\n",
       "   'coding_region_distance': 0,\n",
       "   'codons': 'caA/caG',\n",
       "   'consequences': ['synonymous_variant'],\n",
       "   'dbsnp': ['rs28897700'],\n",
       "   'exon': '3/27',\n",
       "   'exon_distance': 0,\n",
       "   'hgnc_id': 1101,\n",
       "   'in_last_exon': 'no',\n",
       "   'is_canonical': True,\n",
       "   'protein': 'NM_000059.3',\n",
       "   'strand': 1,\n",
       "   'symbol': 'BRCA2',\n",
       "   'transcript': 'NM_000059.3'}]}"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_transcripts_report_data = []\n",
    "for allele_assessment in analysis_data['allele_assessments']:\n",
    "    classification_int_code = allele_assessment['allele_assessment']['classification']\n",
    "    allele_report = allele_assessment['allele_report']\n",
    "    transcripts = allele_assessment['annotation']['transcripts']\n",
    "    # This just gives the name of the filtered transcripts\n",
    "    filtered_transcripts_ids = allele_assessment['annotation']['filtered_transcripts']\n",
    "    filtered_transcripts_data =  list(filter(lambda n : n['transcript'] in filtered_transcripts_ids, transcripts))\n",
    "    for n in filtered_transcripts_data:\n",
    "        filtered_transcripts_report_data.append(n)\n",
    "    allele_assessment['filterd_transcripts'] = filtered_transcripts_report_data\n",
    "    \n",
    "analysis_data['allele_assessments'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>Variant Classification Code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.198A&gt;G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.97G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.72A&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.67+2T&gt;A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.72A&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.51_52delAC</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.97G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.51_52delAC</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.67+2T&gt;A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.198A&gt;G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>U</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene       Variant      Zygosity Variant Classification  \\\n",
       "0   BRCA2       c.10G>T  Heterozygous      Likely Pathogenic   \n",
       "1   BRCA2      c.198A>G  Heterozygous      Likely Pathogenic   \n",
       "2   BRCA2       c.97G>T  Heterozygous      Likely Pathogenic   \n",
       "3   BRCA2       c.72A>T  Heterozygous                Unknown   \n",
       "4   BRCA2     c.67+2T>A  Heterozygous                Unknown   \n",
       "5   BRCA2       c.72A>T  Heterozygous      Likely Pathogenic   \n",
       "7   BRCA2       c.10G>T  Heterozygous                Unknown   \n",
       "9   BRCA2  c.51_52delAC  Heterozygous                Unknown   \n",
       "15  BRCA2       c.97G>T  Heterozygous                Unknown   \n",
       "17  BRCA2  c.51_52delAC  Heterozygous      Likely Pathogenic   \n",
       "18  BRCA2     c.67+2T>A  Heterozygous      Likely Pathogenic   \n",
       "23  BRCA2      c.198A>G  Heterozygous                Unknown   \n",
       "\n",
       "   Variant Classification Code  \n",
       "0                            4  \n",
       "1                            4  \n",
       "2                            4  \n",
       "3                            U  \n",
       "4                            U  \n",
       "5                            4  \n",
       "7                            U  \n",
       "9                            U  \n",
       "15                           U  \n",
       "17                           4  \n",
       "18                           4  \n",
       "23                           U  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# We are assuming we only need data for the first sample!\n",
    "columns = ['Gene', 'Variant', 'Zygosity', 'Variant Classification']\n",
    "\n",
    "# We also need to rank the variants and go for the most conservative classification\n",
    "gene_data = []\n",
    "for allele_assessment in analysis_data['allele_assessments']:\n",
    "    transcripts = allele_assessment['filterd_transcripts']\n",
    "    for transcript in transcripts:\n",
    "        d = {\n",
    "            'Gene': transcript['symbol'],\n",
    "            'Variant': transcript['HGVSc'],\n",
    "            'Zygosity': allele_assessment['samples'][0]['genotype']['type'],\n",
    "            'Variant Classification': acmg_codes[str(allele_assessment['allele_assessment']['classification'])],\n",
    "            'Variant Classification Code': allele_assessment['allele_assessment']['classification']\n",
    "        }\n",
    "        gene_data.append(d)\n",
    "\n",
    "#print(gene_data)\n",
    "import random\n",
    "random.shuffle(gene_data)\n",
    "gene_data\n",
    "df = pd.DataFrame.from_records(gene_data)\n",
    "\n",
    "df = df.drop_duplicates()\n",
    "df\n",
    "#df['Variant Classififica']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>Variant Classification Code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.198A&gt;G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.97G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.72A&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.51_52delAC</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.67+2T&gt;A</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene       Variant      Zygosity Variant Classification  \\\n",
       "0   BRCA2       c.10G>T  Heterozygous      Likely Pathogenic   \n",
       "1   BRCA2      c.198A>G  Heterozygous      Likely Pathogenic   \n",
       "2   BRCA2       c.97G>T  Heterozygous      Likely Pathogenic   \n",
       "5   BRCA2       c.72A>T  Heterozygous      Likely Pathogenic   \n",
       "17  BRCA2  c.51_52delAC  Heterozygous      Likely Pathogenic   \n",
       "18  BRCA2     c.67+2T>A  Heterozygous      Likely Pathogenic   \n",
       "\n",
       "   Variant Classification Code  \n",
       "0                            4  \n",
       "1                            4  \n",
       "2                            4  \n",
       "5                            4  \n",
       "17                           4  \n",
       "18                           4  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# We also need to rank the variants and go for the most conservative classification\n",
    "# I think this will work, but it should be tested more\n",
    "\n",
    "df = df.sort_values(by=['Variant Classification Code'], key=lambda x: x.map(acmg_codes))\n",
    "df = df.drop_duplicates(subset=['Variant'])\n",
    "df.drop(columns=['Variant Classification Code'])\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tabulate import tabulate\n",
    "markdown_table_str = tabulate(df, headers='keys', tablefmt='pipe', showindex=False)\n",
    "mock_patient_data['markdown_table_str'] = markdown_table_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "# Report\n",
      "\n",
      "## Sample-1\n",
      "\n",
      "---\n",
      "\n",
      "|  |  |  |  |\n",
      "| --- | --- | --- | --- |\n",
      "| **Patient Name:** | Smith, Todd | **Ordered By:** | Dr. Who |\n",
      "| **Date of Birth:** | 12/11/1977 | **Sample Collection Date:** | 12/9/2020 | \n",
      "| **Gender**: | Male | **Sample Accession Date:** | 12/9/2020 |\n",
      "| **Sample Type:** | blood | **Report Data:** | 12/9/2020 |\n",
      "| **Sample ID/MRN:** | 1234567 | | |\n",
      "| **Test Accession:** | BRCA122345 | | | \n",
      "| **Test Ordered:** | BRCA2 sequencing & Deletion/Duplication analysis | | | \n",
      "| **Test Code:** | ABCD | | |\n",
      "| **Test Indictation** | Patient has family history of breast cancer. | | |\n",
      "\n",
      "---\n",
      "\n",
      "## Result: Postitive\n",
      "\n",
      "---\n",
      "\n",
      "| Gene   | Variant      | Zygosity     | Variant Classification   |\n",
      "|:-------|:-------------|:-------------|:-------------------------|\n",
      "| BRCA2  | c.10G>T      | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.51_52delAC | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.67+2T>A    | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.72A>T      | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.97G>T      | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.198A>G     | Heterozygous | Likely Pathogenic        |\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from jinja2 import Environment, BaseLoader\n",
    "\n",
    "\n",
    "report_header_table = \"\"\"\n",
    "# Report\n",
    "\n",
    "## Sample-1\n",
    "\n",
    "---\n",
    "\n",
    "|  |  |  |  |\n",
    "| --- | --- | --- | --- |\n",
    "| **Patient Name:** | {{ patient_data.patient_last_name }}, {{ patient_data.patient_first_name }} | **Ordered By:** | {{test_data.test_requesting_physician}} |\n",
    "| **Date of Birth:** | {{patient_data.patient_dob}} | **Sample Collection Date:** | {{ sample_data.sample_collection_date }} | \n",
    "| **Gender**: | {{patient_data.patient_gender}} | **Sample Accession Date:** | {{ sample_data.sample_accession_date }} |\n",
    "| **Sample Type:** | {{sample_data.sample_type}} | **Report Data:** | {{report_data.report_date}} |\n",
    "| **Sample ID/MRN:** | {{ sample_data.sample_id }} | | |\n",
    "| **Test Accession:** | {{ test_data.test_accession }} | | | \n",
    "| **Test Ordered:** | {{test_data.test_ordered}} | | | \n",
    "| **Test Code:** | {{ test_data.test_code }} | | |\n",
    "| **Test Indictation** | {{ test_data.test_indication }} | | |\n",
    "\n",
    "---\n",
    "\n",
    "## Result: Postitive\n",
    "\n",
    "---\n",
    "\n",
    "{{ markdown_table_str }}\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "rtemplate = Environment(loader=BaseLoader()).from_string(report_header_table)\n",
    "rendered_report_header_table = rtemplate.render(**mock_patient_data)\n",
    "print(rendered_report_header_table)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "# Report\n",
       "\n",
       "## Sample-1\n",
       "\n",
       "---\n",
       "\n",
       "|  |  |  |  |\n",
       "| --- | --- | --- | --- |\n",
       "| **Patient Name:** | Smith, Todd | **Ordered By:** | Dr. Who |\n",
       "| **Date of Birth:** | 12/11/1977 | **Sample Collection Date:** | 12/9/2020 | \n",
       "| **Gender**: | Male | **Sample Accession Date:** | 12/9/2020 |\n",
       "| **Sample Type:** | blood | **Report Data:** | 12/9/2020 |\n",
       "| **Sample ID/MRN:** | 1234567 | | |\n",
       "| **Test Accession:** | BRCA122345 | | | \n",
       "| **Test Ordered:** | BRCA2 sequencing & Deletion/Duplication analysis | | | \n",
       "| **Test Code:** | ABCD | | |\n",
       "| **Test Indictation** | Patient has family history of breast cancer. | | |\n",
       "\n",
       "---\n",
       "\n",
       "## Result: Postitive\n",
       "\n",
       "---\n",
       "\n",
       "| Gene   | Variant      | Zygosity     | Variant Classification   |\n",
       "|:-------|:-------------|:-------------|:-------------------------|\n",
       "| BRCA2  | c.10G>T      | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.51_52delAC | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.67+2T>A    | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.72A>T      | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.97G>T      | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.198A>G     | Heterozygous | Likely Pathogenic        |\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# This doesn't work in the jupyterhub output\n",
    "from IPython.display import display, Markdown, Latex, HTML\n",
    "display(Markdown(rendered_report_header_table))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
