{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Parse the Ella Analysis\n",
    "\n",
    "In the previous notebook I demonstrated how to query the database directly to create a custom report. This is an example of you could then query and output the data.\n",
    "\n",
    "Note that the analysis output format may change!\n",
    "\n",
    "This assumes that you ran the 'Analysis-brca_sample_1.HB0C' notebook to generate the data file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import pandas as pd\n",
    "import os\n",
    "import copy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/data/analysis_data.json'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# There was an illegal escape sequence in the original file \n",
    "# Make sure not to have any slashes in front of analysis_data. (not /analysis_data)\n",
    "analysis_data_file = '/data/analysis_data.json'\n",
    "analysis_data_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['sample', 'alleleassessment', 'allelereport', 'geneassessment', 'analysisinterpretation', 'analysis', 'annotation', 'annotationshadowtranscript', 'allele', 'genotype', 'genotypesampledata', 'filtered_allele_data', 'filtered_allele_ids'])"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(analysis_data_file, 'r') as reader:\n",
    "    analysis_data = json.load(reader)\n",
    "    \n",
    "acmg_config = []\n",
    "#Class 1 benign\n",
    "#Class 2 likely benign\n",
    "#Class 3 variant of uncertain significance (VUS)\n",
    "#Class 4 likely pathogenic\n",
    "#Class 5 pathogenic\n",
    "\n",
    "acmg_codes = {\n",
    "    '1': 'Benign',\n",
    "    '2': 'Likely Benign',\n",
    "    '3': 'Variant of Uncertain Significance',\n",
    "    '4': 'Likely Pathogenic',\n",
    "    '5': 'Pathogenic',\n",
    "    'U': 'Unknown'\n",
    "}\n",
    "\n",
    "for idx, assessment in enumerate(analysis_data['alleleassessment']):\n",
    "    acmg_config_t = {\n",
    "        'id': assessment['id'],\n",
    "        'acmg_code' : assessment['classification'],\n",
    "        'acmg_class' : acmg_codes[assessment['classification']],\n",
    "        'allele_id': assessment['allele_id']\n",
    "    }\n",
    "    acmg_eval = assessment['evaluation']['acmg']['included']\n",
    "    codes = map(lambda x: x['code'], acmg_eval)\n",
    "    codes = list(codes)\n",
    "    acmg_config_t['codes'] = ', '.join(codes)\n",
    "    acmg_config.append(acmg_config_t)\n",
    "    assessment['acmg_class'] = acmg_config_t['acmg_class']\n",
    "    \n",
    "analysis_data.keys()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>identifier</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>date_deposited</th>\n",
       "      <th>affected</th>\n",
       "      <th>family_id</th>\n",
       "      <th>father_id</th>\n",
       "      <th>mother_id</th>\n",
       "      <th>sibling_id</th>\n",
       "      <th>proband</th>\n",
       "      <th>sex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>brca_sample_1</td>\n",
       "      <td>2</td>\n",
       "      <td>HTS</td>\n",
       "      <td>1603104403434</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id     identifier  analysis_id sample_type  date_deposited  affected  \\\n",
       "0   2  brca_sample_1            2         HTS   1603104403434      True   \n",
       "\n",
       "  family_id father_id mother_id sibling_id  proband   sex  \n",
       "0      None      None      None       None     True  None  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame.from_records(analysis_data['sample'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Panel : HB0C v01"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>secondallele_id</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>variant_quality</th>\n",
       "      <th>filter_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  allele_id secondallele_id  sample_id  variant_quality filter_status\n",
       "0   3          3            None          2             5000          PASS\n",
       "1   4          4            None          2             5000          PASS\n",
       "2   5          5            None          2             5000          PASS\n",
       "3   6          6            None          2             5000          PASS\n",
       "4   7          7            None          2             5000          PASS"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotype_df = pd.DataFrame.from_records(analysis_data['genotype'])\n",
    "genotype_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alleles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>genome_reference</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_position</th>\n",
       "      <th>open_end_position</th>\n",
       "      <th>change_from</th>\n",
       "      <th>change_to</th>\n",
       "      <th>change_type</th>\n",
       "      <th>vcf_pos</th>\n",
       "      <th>vcf_ref</th>\n",
       "      <th>vcf_alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890606</td>\n",
       "      <td>32890607</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32890607</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890645</td>\n",
       "      <td>32890647</td>\n",
       "      <td>AC</td>\n",
       "      <td></td>\n",
       "      <td>del</td>\n",
       "      <td>32890645</td>\n",
       "      <td>GAC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32890665</td>\n",
       "      <td>32890666</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32890666</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893217</td>\n",
       "      <td>32893218</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893218</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893242</td>\n",
       "      <td>32893243</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893243</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>8</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>32893343</td>\n",
       "      <td>32893344</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>32893344</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id genome_reference chromosome  start_position  open_end_position  \\\n",
       "0   3           GRCh37         13        32890606           32890607   \n",
       "1   4           GRCh37         13        32890645           32890647   \n",
       "2   5           GRCh37         13        32890665           32890666   \n",
       "3   6           GRCh37         13        32893217           32893218   \n",
       "4   7           GRCh37         13        32893242           32893243   \n",
       "5   8           GRCh37         13        32893343           32893344   \n",
       "\n",
       "  change_from change_to change_type   vcf_pos vcf_ref vcf_alt  \n",
       "0           G         T         SNP  32890607       G       T  \n",
       "1          AC                   del  32890645     GAC       G  \n",
       "2           T         A         SNP  32890666       T       A  \n",
       "3           A         T         SNP  32893218       A       T  \n",
       "4           G         T         SNP  32893243       G       T  \n",
       "5           A         G         SNP  32893344       A       G  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_df = pd.DataFrame.from_records(analysis_data['allele'])\n",
    "allele_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# just get the allele ids\n",
    "\n",
    "#allele_df.id.isin(filtered_allele_ids)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>symbol</th>\n",
       "      <th>transcript</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>protein</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>consequences</th>\n",
       "      <th>exon_distance</th>\n",
       "      <th>coding_region_distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>49</td>\n",
       "      <td>3</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "      <td>ENST00000345108</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[upstream_gene_variant]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>50</td>\n",
       "      <td>3</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000380152</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>ENSP00000369497.3</td>\n",
       "      <td>p.Gly4Ter</td>\n",
       "      <td>[stop_gained]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>51</td>\n",
       "      <td>3</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000530893</td>\n",
       "      <td>c.-356G&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[5_prime_UTR_variant]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-356.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>52</td>\n",
       "      <td>3</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "      <td>ENST00000533490</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[upstream_gene_variant]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>53</td>\n",
       "      <td>3</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000544455</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>ENSP00000439902.1</td>\n",
       "      <td>p.Gly4Ter</td>\n",
       "      <td>[stop_gained]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  allele_id  hgnc_id symbol       transcript      hgvsc  \\\n",
       "0  49          3    37116  ZAR1L  ENST00000345108       None   \n",
       "1  50          3     1101  BRCA2  ENST00000380152    c.10G>T   \n",
       "2  51          3     1101  BRCA2  ENST00000530893  c.-356G>T   \n",
       "3  52          3    37116  ZAR1L  ENST00000533490       None   \n",
       "4  53          3     1101  BRCA2  ENST00000544455    c.10G>T   \n",
       "\n",
       "             protein      hgvsp             consequences  exon_distance  \\\n",
       "0               None       None  [upstream_gene_variant]            NaN   \n",
       "1  ENSP00000369497.3  p.Gly4Ter            [stop_gained]            0.0   \n",
       "2               None       None    [5_prime_UTR_variant]            0.0   \n",
       "3               None       None  [upstream_gene_variant]            NaN   \n",
       "4  ENSP00000439902.1  p.Gly4Ter            [stop_gained]            0.0   \n",
       "\n",
       "   coding_region_distance  \n",
       "0                     NaN  \n",
       "1                     0.0  \n",
       "2                  -356.0  \n",
       "3                     NaN  \n",
       "4                     0.0  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_shadow_transcript_df =  pd.DataFrame.from_records(analysis_data['annotationshadowtranscript'])\n",
    "annotation_shadow_transcript_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Gene Symbols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>allele_id</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>4</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    allele_id  hgnc_id symbol\n",
       "0           3    37116  ZAR1L\n",
       "1           3     1101  BRCA2\n",
       "7           4    37116  ZAR1L\n",
       "8           4     1101  BRCA2\n",
       "14          5    37116  ZAR1L"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_symbols_df = annotation_shadow_transcript_df[['allele_id', 'hgnc_id', 'symbol']].copy()\n",
    "gene_symbols_df = gene_symbols_df.drop_duplicates()\n",
    "gene_symbols_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ACMG Assessments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>acmg_code</th>\n",
       "      <th>acmg_class</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>codes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>U</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>6</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>7</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id acmg_code         acmg_class  allele_id         codes\n",
       "0   1         4  Likely Pathogenic          3  PVS1, PPxPM2\n",
       "1   2         4  Likely Pathogenic          4  PVS1, PPxPM2\n",
       "2   3         4  Likely Pathogenic          5  PVS1, PPxPM2\n",
       "3   4         U            Unknown          6              \n",
       "4   5         4  Likely Pathogenic          7  PVS1, PPxPM2"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# TODO Get Gene Symbol from transcripts\n",
    "acmg_config_df = pd.DataFrame.from_records(acmg_config)\n",
    "acmg_config_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>acmg_code</th>\n",
       "      <th>acmg_class</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>codes</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>3</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>4</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>5</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id acmg_code         acmg_class  allele_id         codes  hgnc_id symbol\n",
       "0   1         4  Likely Pathogenic          3  PVS1, PPxPM2    37116  ZAR1L\n",
       "0   1         4  Likely Pathogenic          3  PVS1, PPxPM2     1101  BRCA2\n",
       "1   2         4  Likely Pathogenic          4  PVS1, PPxPM2    37116  ZAR1L\n",
       "1   2         4  Likely Pathogenic          4  PVS1, PPxPM2     1101  BRCA2\n",
       "2   3         4  Likely Pathogenic          5  PVS1, PPxPM2    37116  ZAR1L"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_data_df = acmg_config_df.join(gene_symbols_df.set_index('allele_id'), on='allele_id')\n",
    "allele_data_df.join(allele_df.set_index('id'), on='allele_id')\n",
    "allele_data_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Allele Report\n",
    "\n",
    "These contain custom comments by the Genetic Counselor."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>evaluation</th>\n",
       "      <th>user_id</th>\n",
       "      <th>date_created</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>previous_report_id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>alleleassessment_id</th>\n",
       "      <th>usergroup_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>{'comment': ''}</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104584640</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>{'comment': ''}</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104600778</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>{'comment': ''}</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104614366</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>{'comment': ''}</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104626999</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>{'comment': ''}</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104638892</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id       evaluation  user_id   date_created date_superceeded  \\\n",
       "0   1  {'comment': ''}        1  1603104584640             None   \n",
       "1   2  {'comment': ''}        1  1603104600778             None   \n",
       "2   3  {'comment': ''}        1  1603104614366             None   \n",
       "3   4  {'comment': ''}        1  1603104626999             None   \n",
       "4   5  {'comment': ''}        1  1603104638892             None   \n",
       "\n",
       "  previous_report_id  allele_id  analysis_id alleleassessment_id  usergroup_id  \n",
       "0               None          3            2                None             1  \n",
       "1               None          4            2                None             1  \n",
       "2               None          5            2                None             1  \n",
       "3               None          6            2                None             1  \n",
       "4               None          7            2                None             1  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_report_df = pd.DataFrame.from_records(analysis_data['allelereport'])\n",
    "allele_report_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genotype Sample Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>genotype_id</th>\n",
       "      <th>secondallele</th>\n",
       "      <th>multiallelic</th>\n",
       "      <th>type</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genotype_quality</th>\n",
       "      <th>sequencing_depth</th>\n",
       "      <th>genotype_likelihood</th>\n",
       "      <th>allele_depth</th>\n",
       "      <th>allele_ratio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (GAC)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'A': 80, 'REF (T)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (A)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  genotype_id  secondallele  multiallelic          type  sample_id  \\\n",
       "0   3            3         False         False  Heterozygous          2   \n",
       "1   4            4         False         False  Heterozygous          2   \n",
       "2   5            5         False         False  Heterozygous          2   \n",
       "3   6            6         False         False  Heterozygous          2   \n",
       "4   7            7         False         False  Heterozygous          2   \n",
       "\n",
       "   genotype_quality  sequencing_depth genotype_likelihood  \\\n",
       "0                99               187     [2048, 0, 2917]   \n",
       "1                99               187     [2048, 0, 2917]   \n",
       "2                99               187     [2048, 0, 2917]   \n",
       "3                99               187     [2048, 0, 2917]   \n",
       "4                99               187     [2048, 0, 2917]   \n",
       "\n",
       "                  allele_depth  allele_ratio  \n",
       "0    {'T': 80, 'REF (G)': 107}      0.427807  \n",
       "1  {'G': 80, 'REF (GAC)': 107}      0.427807  \n",
       "2    {'A': 80, 'REF (T)': 107}      0.427807  \n",
       "3    {'T': 80, 'REF (A)': 107}      0.427807  \n",
       "4    {'T': 80, 'REF (G)': 107}      0.427807  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotype_sample_data_report_df = pd.DataFrame.from_records(analysis_data['genotypesampledata'])\n",
    "genotype_sample_data_report_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate a Merged table with ALL THE ALLELE THINGS\n",
    "\n",
    "Obviously,throwing everything into a single table is not suitable for very large datasets. Then we want to use something smarter like Xarray + Dask or proper SQL queries."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_x</th>\n",
       "      <th>genotype_id</th>\n",
       "      <th>secondallele</th>\n",
       "      <th>multiallelic</th>\n",
       "      <th>type</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genotype_quality</th>\n",
       "      <th>sequencing_depth</th>\n",
       "      <th>genotype_likelihood</th>\n",
       "      <th>allele_depth</th>\n",
       "      <th>allele_ratio</th>\n",
       "      <th>id_y</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>secondallele_id</th>\n",
       "      <th>variant_quality</th>\n",
       "      <th>filter_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (GAC)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'A': 80, 'REF (T)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (A)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>0.427807</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id_x  genotype_id  secondallele  multiallelic          type  sample_id  \\\n",
       "0     3            3         False         False  Heterozygous          2   \n",
       "1     4            4         False         False  Heterozygous          2   \n",
       "2     5            5         False         False  Heterozygous          2   \n",
       "3     6            6         False         False  Heterozygous          2   \n",
       "4     7            7         False         False  Heterozygous          2   \n",
       "\n",
       "   genotype_quality  sequencing_depth genotype_likelihood  \\\n",
       "0                99               187     [2048, 0, 2917]   \n",
       "1                99               187     [2048, 0, 2917]   \n",
       "2                99               187     [2048, 0, 2917]   \n",
       "3                99               187     [2048, 0, 2917]   \n",
       "4                99               187     [2048, 0, 2917]   \n",
       "\n",
       "                  allele_depth  allele_ratio  id_y  allele_id secondallele_id  \\\n",
       "0    {'T': 80, 'REF (G)': 107}      0.427807     3          3            None   \n",
       "1  {'G': 80, 'REF (GAC)': 107}      0.427807     4          4            None   \n",
       "2    {'A': 80, 'REF (T)': 107}      0.427807     5          5            None   \n",
       "3    {'T': 80, 'REF (A)': 107}      0.427807     6          6            None   \n",
       "4    {'T': 80, 'REF (G)': 107}      0.427807     7          7            None   \n",
       "\n",
       "   variant_quality filter_status  \n",
       "0             5000          PASS  \n",
       "1             5000          PASS  \n",
       "2             5000          PASS  \n",
       "3             5000          PASS  \n",
       "4             5000          PASS  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotype_data_df = pd.merge(genotype_sample_data_report_df, genotype_df, left_on=['genotype_id','sample_id'], right_on=['id', 'sample_id'])\n",
    "genotype_data_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_x</th>\n",
       "      <th>genotype_id</th>\n",
       "      <th>secondallele</th>\n",
       "      <th>multiallelic</th>\n",
       "      <th>type</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genotype_quality</th>\n",
       "      <th>sequencing_depth</th>\n",
       "      <th>genotype_likelihood</th>\n",
       "      <th>allele_depth</th>\n",
       "      <th>...</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>secondallele_id</th>\n",
       "      <th>variant_quality</th>\n",
       "      <th>filter_status</th>\n",
       "      <th>id</th>\n",
       "      <th>acmg_code</th>\n",
       "      <th>acmg_class</th>\n",
       "      <th>codes</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (GAC)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (GAC)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'A': 80, 'REF (T)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>5000</td>\n",
       "      <td>PASS</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "      <td>PVS1, PPxPM2</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   id_x  genotype_id  secondallele  multiallelic          type  sample_id  \\\n",
       "0     3            3         False         False  Heterozygous          2   \n",
       "1     3            3         False         False  Heterozygous          2   \n",
       "2     4            4         False         False  Heterozygous          2   \n",
       "3     4            4         False         False  Heterozygous          2   \n",
       "4     5            5         False         False  Heterozygous          2   \n",
       "\n",
       "   genotype_quality  sequencing_depth genotype_likelihood  \\\n",
       "0                99               187     [2048, 0, 2917]   \n",
       "1                99               187     [2048, 0, 2917]   \n",
       "2                99               187     [2048, 0, 2917]   \n",
       "3                99               187     [2048, 0, 2917]   \n",
       "4                99               187     [2048, 0, 2917]   \n",
       "\n",
       "                  allele_depth  ...  allele_id  secondallele_id  \\\n",
       "0    {'T': 80, 'REF (G)': 107}  ...          3             None   \n",
       "1    {'T': 80, 'REF (G)': 107}  ...          3             None   \n",
       "2  {'G': 80, 'REF (GAC)': 107}  ...          4             None   \n",
       "3  {'G': 80, 'REF (GAC)': 107}  ...          4             None   \n",
       "4    {'A': 80, 'REF (T)': 107}  ...          5             None   \n",
       "\n",
       "   variant_quality filter_status  id acmg_code         acmg_class  \\\n",
       "0             5000          PASS   1         4  Likely Pathogenic   \n",
       "1             5000          PASS   1         4  Likely Pathogenic   \n",
       "2             5000          PASS   2         4  Likely Pathogenic   \n",
       "3             5000          PASS   2         4  Likely Pathogenic   \n",
       "4             5000          PASS   3         4  Likely Pathogenic   \n",
       "\n",
       "          codes hgnc_id symbol  \n",
       "0  PVS1, PPxPM2   37116  ZAR1L  \n",
       "1  PVS1, PPxPM2    1101  BRCA2  \n",
       "2  PVS1, PPxPM2   37116  ZAR1L  \n",
       "3  PVS1, PPxPM2    1101  BRCA2  \n",
       "4  PVS1, PPxPM2   37116  ZAR1L  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analysis_data_merged_report = pd.merge(genotype_data_df, allele_data_df, left_on='allele_id', right_on='allele_id')\n",
    "analysis_data_merged_report.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_x</th>\n",
       "      <th>genotype_id</th>\n",
       "      <th>secondallele</th>\n",
       "      <th>multiallelic</th>\n",
       "      <th>type</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genotype_quality</th>\n",
       "      <th>sequencing_depth</th>\n",
       "      <th>genotype_likelihood</th>\n",
       "      <th>allele_depth</th>\n",
       "      <th>...</th>\n",
       "      <th>id_y</th>\n",
       "      <th>hgnc_id_y</th>\n",
       "      <th>symbol_y</th>\n",
       "      <th>transcript</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>protein</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>consequences</th>\n",
       "      <th>exon_distance</th>\n",
       "      <th>coding_region_distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>49</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "      <td>ENST00000345108</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[upstream_gene_variant]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>50</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000380152</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>ENSP00000369497.3</td>\n",
       "      <td>p.Gly4Ter</td>\n",
       "      <td>[stop_gained]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>51</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000530893</td>\n",
       "      <td>c.-356G&gt;T</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[5_prime_UTR_variant]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-356.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>52</td>\n",
       "      <td>37116</td>\n",
       "      <td>ZAR1L</td>\n",
       "      <td>ENST00000533490</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>[upstream_gene_variant]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>53</td>\n",
       "      <td>1101</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>ENST00000544455</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>ENSP00000439902.1</td>\n",
       "      <td>p.Gly4Ter</td>\n",
       "      <td>[stop_gained]</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   id_x  genotype_id  secondallele  multiallelic          type  sample_id  \\\n",
       "0     3            3         False         False  Heterozygous          2   \n",
       "1     3            3         False         False  Heterozygous          2   \n",
       "2     3            3         False         False  Heterozygous          2   \n",
       "3     3            3         False         False  Heterozygous          2   \n",
       "4     3            3         False         False  Heterozygous          2   \n",
       "\n",
       "   genotype_quality  sequencing_depth genotype_likelihood  \\\n",
       "0                99               187     [2048, 0, 2917]   \n",
       "1                99               187     [2048, 0, 2917]   \n",
       "2                99               187     [2048, 0, 2917]   \n",
       "3                99               187     [2048, 0, 2917]   \n",
       "4                99               187     [2048, 0, 2917]   \n",
       "\n",
       "                allele_depth  ...  id_y  hgnc_id_y  symbol_y       transcript  \\\n",
       "0  {'T': 80, 'REF (G)': 107}  ...    49      37116     ZAR1L  ENST00000345108   \n",
       "1  {'T': 80, 'REF (G)': 107}  ...    50       1101     BRCA2  ENST00000380152   \n",
       "2  {'T': 80, 'REF (G)': 107}  ...    51       1101     BRCA2  ENST00000530893   \n",
       "3  {'T': 80, 'REF (G)': 107}  ...    52      37116     ZAR1L  ENST00000533490   \n",
       "4  {'T': 80, 'REF (G)': 107}  ...    53       1101     BRCA2  ENST00000544455   \n",
       "\n",
       "       hgvsc            protein      hgvsp             consequences  \\\n",
       "0       None               None       None  [upstream_gene_variant]   \n",
       "1    c.10G>T  ENSP00000369497.3  p.Gly4Ter            [stop_gained]   \n",
       "2  c.-356G>T               None       None    [5_prime_UTR_variant]   \n",
       "3       None               None       None  [upstream_gene_variant]   \n",
       "4    c.10G>T  ENSP00000439902.1  p.Gly4Ter            [stop_gained]   \n",
       "\n",
       "  exon_distance coding_region_distance  \n",
       "0           NaN                    NaN  \n",
       "1           0.0                    0.0  \n",
       "2           0.0                 -356.0  \n",
       "3           NaN                    NaN  \n",
       "4           0.0                    0.0  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analysis_data_merged_report = pd.merge(analysis_data_merged_report, annotation_shadow_transcript_df, left_on='allele_id', right_on='allele_id')\n",
    "analysis_data_merged_report.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>classification</th>\n",
       "      <th>evaluation</th>\n",
       "      <th>user_id</th>\n",
       "      <th>date_created</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>previous_assessment_id</th>\n",
       "      <th>allele_id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>annotation_id</th>\n",
       "      <th>custom_annotation_id</th>\n",
       "      <th>usergroup_id</th>\n",
       "      <th>acmg_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104600770</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104614357</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>U</td>\n",
       "      <td>{'acmg': {'included': [], 'suggested': [{'op':...</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104626990</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>{'acmg': {'included': [{'op': None, 'code': 'P...</td>\n",
       "      <td>1</td>\n",
       "      <td>1603104638883</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id classification                                         evaluation  \\\n",
       "0   1              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "1   2              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "2   3              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "3   4              U  {'acmg': {'included': [], 'suggested': [{'op':...   \n",
       "4   5              4  {'acmg': {'included': [{'op': None, 'code': 'P...   \n",
       "\n",
       "   user_id   date_created date_superceeded previous_assessment_id  allele_id  \\\n",
       "0        1  1603104584630             None                   None          3   \n",
       "1        1  1603104600770             None                   None          4   \n",
       "2        1  1603104614357             None                   None          5   \n",
       "3        1  1603104626990             None                   None          6   \n",
       "4        1  1603104638883             None                   None          7   \n",
       "\n",
       "  genepanel_name genepanel_version  analysis_id  annotation_id  \\\n",
       "0        HBOCUTV               v01            2              3   \n",
       "1        HBOCUTV               v01            2              4   \n",
       "2        HBOCUTV               v01            2              5   \n",
       "3        HBOCUTV               v01            2              6   \n",
       "4        HBOCUTV               v01            2              7   \n",
       "\n",
       "  custom_annotation_id  usergroup_id         acmg_class  \n",
       "0                 None             1  Likely Pathogenic  \n",
       "1                 None             1  Likely Pathogenic  \n",
       "2                 None             1  Likely Pathogenic  \n",
       "3                 None             1            Unknown  \n",
       "4                 None             1  Likely Pathogenic  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_assessment_df = pd.DataFrame.from_records(analysis_data['alleleassessment']) \n",
    "allele_assessment_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'4'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allele_assessment_df.iloc[0]['classification']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id_x</th>\n",
       "      <th>genotype_id</th>\n",
       "      <th>secondallele</th>\n",
       "      <th>multiallelic</th>\n",
       "      <th>type</th>\n",
       "      <th>sample_id</th>\n",
       "      <th>genotype_quality</th>\n",
       "      <th>sequencing_depth</th>\n",
       "      <th>genotype_likelihood</th>\n",
       "      <th>allele_depth</th>\n",
       "      <th>...</th>\n",
       "      <th>date_created</th>\n",
       "      <th>date_superceeded</th>\n",
       "      <th>previous_assessment_id</th>\n",
       "      <th>genepanel_name</th>\n",
       "      <th>genepanel_version</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>annotation_id</th>\n",
       "      <th>custom_annotation_id</th>\n",
       "      <th>usergroup_id</th>\n",
       "      <th>acmg_class_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'T': 80, 'REF (G)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104584630</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (A)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104647757</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (A)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104647757</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (A)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104647757</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (A)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104647757</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>2</td>\n",
       "      <td>99</td>\n",
       "      <td>187</td>\n",
       "      <td>[2048, 0, 2917]</td>\n",
       "      <td>{'G': 80, 'REF (A)': 107}</td>\n",
       "      <td>...</td>\n",
       "      <td>1603104647757</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>HBOCUTV</td>\n",
       "      <td>v01</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows  46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    id_x  genotype_id  secondallele  multiallelic          type  sample_id  \\\n",
       "0      3            3         False         False  Heterozygous          2   \n",
       "1      3            3         False         False  Heterozygous          2   \n",
       "2      3            3         False         False  Heterozygous          2   \n",
       "3      3            3         False         False  Heterozygous          2   \n",
       "4      3            3         False         False  Heterozygous          2   \n",
       "..   ...          ...           ...           ...           ...        ...   \n",
       "67     8            8         False         False  Heterozygous          2   \n",
       "68     8            8         False         False  Heterozygous          2   \n",
       "69     8            8         False         False  Heterozygous          2   \n",
       "70     8            8         False         False  Heterozygous          2   \n",
       "71     8            8         False         False  Heterozygous          2   \n",
       "\n",
       "    genotype_quality  sequencing_depth genotype_likelihood  \\\n",
       "0                 99               187     [2048, 0, 2917]   \n",
       "1                 99               187     [2048, 0, 2917]   \n",
       "2                 99               187     [2048, 0, 2917]   \n",
       "3                 99               187     [2048, 0, 2917]   \n",
       "4                 99               187     [2048, 0, 2917]   \n",
       "..               ...               ...                 ...   \n",
       "67                99               187     [2048, 0, 2917]   \n",
       "68                99               187     [2048, 0, 2917]   \n",
       "69                99               187     [2048, 0, 2917]   \n",
       "70                99               187     [2048, 0, 2917]   \n",
       "71                99               187     [2048, 0, 2917]   \n",
       "\n",
       "                 allele_depth  ...   date_created  date_superceeded  \\\n",
       "0   {'T': 80, 'REF (G)': 107}  ...  1603104584630              None   \n",
       "1   {'T': 80, 'REF (G)': 107}  ...  1603104584630              None   \n",
       "2   {'T': 80, 'REF (G)': 107}  ...  1603104584630              None   \n",
       "3   {'T': 80, 'REF (G)': 107}  ...  1603104584630              None   \n",
       "4   {'T': 80, 'REF (G)': 107}  ...  1603104584630              None   \n",
       "..                        ...  ...            ...               ...   \n",
       "67  {'G': 80, 'REF (A)': 107}  ...  1603104647757              None   \n",
       "68  {'G': 80, 'REF (A)': 107}  ...  1603104647757              None   \n",
       "69  {'G': 80, 'REF (A)': 107}  ...  1603104647757              None   \n",
       "70  {'G': 80, 'REF (A)': 107}  ...  1603104647757              None   \n",
       "71  {'G': 80, 'REF (A)': 107}  ...  1603104647757              None   \n",
       "\n",
       "    previous_assessment_id genepanel_name  genepanel_version analysis_id  \\\n",
       "0                     None        HBOCUTV                v01           2   \n",
       "1                     None        HBOCUTV                v01           2   \n",
       "2                     None        HBOCUTV                v01           2   \n",
       "3                     None        HBOCUTV                v01           2   \n",
       "4                     None        HBOCUTV                v01           2   \n",
       "..                     ...            ...                ...         ...   \n",
       "67                    None        HBOCUTV                v01           2   \n",
       "68                    None        HBOCUTV                v01           2   \n",
       "69                    None        HBOCUTV                v01           2   \n",
       "70                    None        HBOCUTV                v01           2   \n",
       "71                    None        HBOCUTV                v01           2   \n",
       "\n",
       "    annotation_id custom_annotation_id usergroup_id       acmg_class_y  \n",
       "0               3                 None            1  Likely Pathogenic  \n",
       "1               3                 None            1  Likely Pathogenic  \n",
       "2               3                 None            1  Likely Pathogenic  \n",
       "3               3                 None            1  Likely Pathogenic  \n",
       "4               3                 None            1  Likely Pathogenic  \n",
       "..            ...                  ...          ...                ...  \n",
       "67              8                 None            1            Unknown  \n",
       "68              8                 None            1            Unknown  \n",
       "69              8                 None            1            Unknown  \n",
       "70              8                 None            1            Unknown  \n",
       "71              8                 None            1            Unknown  \n",
       "\n",
       "[72 rows x 46 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "analysis_data_merged_report = pd.merge(analysis_data_merged_report, allele_assessment_df, left_on='allele_id', right_on='allele_id')\n",
    "analysis_data_merged_report"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is a lot of data, so let's subset just a few to keep in our report.\n",
    "\n",
    "Disclaimer: I am not a variant scientist. The examples given are meant to help to give an understanding of the data structures for people who *know what they are doing* to generate reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['id_x', 'genotype_id', 'secondallele', 'multiallelic', 'type',\n",
       "       'sample_id', 'genotype_quality', 'sequencing_depth',\n",
       "       'genotype_likelihood', 'allele_depth', 'allele_ratio', 'id_y',\n",
       "       'allele_id', 'secondallele_id', 'variant_quality', 'filter_status',\n",
       "       'id_x', 'acmg_code', 'acmg_class_x', 'codes', 'hgnc_id_x', 'symbol_x',\n",
       "       'id_y', 'hgnc_id_y', 'symbol_y', 'transcript', 'hgvsc', 'protein',\n",
       "       'hgvsp', 'consequences', 'exon_distance', 'coding_region_distance',\n",
       "       'id', 'classification', 'evaluation', 'user_id', 'date_created',\n",
       "       'date_superceeded', 'previous_assessment_id', 'genepanel_name',\n",
       "       'genepanel_version', 'analysis_id', 'annotation_id',\n",
       "       'custom_annotation_id', 'usergroup_id', 'acmg_class_y'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "keep_these = ['c.10G>T', 'c.-172A>G']\n",
    "\n",
    "subset_df = analysis_data_merged_report[analysis_data_merged_report[\"hgvsc\"].isin(keep_these)]\n",
    "#subset_df = subset_df[0:2]\n",
    "subset_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>symbol_y</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>type</th>\n",
       "      <th>acmg_class_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.-172A&gt;G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   symbol_y      hgvsc          type       acmg_class_y\n",
       "1     BRCA2    c.10G>T  Heterozygous  Likely Pathogenic\n",
       "63    BRCA2  c.-172A>G  Heterozygous            Unknown"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1 = subset_df[['symbol_y',  'hgvsc', 'type', 'acmg_class_y']]\n",
    "df1 = df1.drop_duplicates()\n",
    "df1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Variant Classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.10G&gt;T</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Likely Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>c.-172A&gt;G</td>\n",
       "      <td>Heterozygous</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene    Variant      Zygosity Variant Classification\n",
       "1   BRCA2    c.10G>T  Heterozygous      Likely Pathogenic\n",
       "63  BRCA2  c.-172A>G  Heterozygous                Unknown"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.columns = ['Gene', 'Variant', 'Zygosity', 'Variant Classification']\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: tabulate in /dist/ella-python/lib/python3.7/site-packages\r\n"
     ]
    }
   ],
   "source": [
    "! pip install tabulate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tabulate import tabulate\n",
    "markdown_table_str = tabulate(df1, headers='keys', tablefmt='pipe', showindex=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create a Custom Report Template\n",
    "\n",
    "This is just an example of how you could create a custom report. Every institute is going to have it's own custom reporting."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "report_data = {\n",
    "    \"sample_name\": \"Sample-N\",\n",
    "    \"patient_data\": {\n",
    "        \"patient_first_name\": \"Todd\",\n",
    "        \"patient_last_name\": \"Smith\",\n",
    "        \"patient_dob\": \"12/11/1977\",\n",
    "        \"patient_gender\": \"Male\",\n",
    "    },\n",
    "    \"sample_data\": {\n",
    "        \"sample_type\": \"blood\",\n",
    "        \"sample_collection_date\": \"12/9/2020\",\n",
    "        \"sample_id\": \"1234567\",\n",
    "        \"sample_accession\": \"1234567\",\n",
    "        \"sample_accession_date\": \"12/9/2020\",\n",
    "    },\n",
    "    \"test_data\": {\n",
    "        \"test_ordered\": \"BRCA2 sequencing & Deletion/Duplication analysis\",\n",
    "        \"test_code\": \"ABCD\",\n",
    "        \"test_indication\": \"Patient has family history of breast cancer.\",\n",
    "        \"test_accession\": \"BRCA122345\",\n",
    "        \"test_requesting_physician\": \"Dr. Who\",\n",
    "    },\n",
    "    \"report_data\": {\n",
    "        \"report_date\": \"12/9/2020\",\n",
    "    },\n",
    "    \"methods\": {\n",
    "      \"mlpa\": \"\"\"MLPA.\"\"\",\n",
    "      \"amplicon\": \"\"\"AMPLICON\"\"\",  \n",
    "    },\n",
    "    #\"mlpa_analysis_data\": None,\n",
    "    #\"ngs_analysis_data\": None,\n",
    "    #\"analysis_data\": df1,\n",
    "    \"markdown_table_str\" : markdown_table_str,\n",
    "}\n",
    "\n",
    "with open('/data/mock_patient_data.json', 'w') as fp:\n",
    "    json.dump(report_data, fp, sort_keys=True, indent=4)\n",
    "    \n",
    "report_data['analysis_data'] = df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "# Report\n",
      "\n",
      "## Sample-1\n",
      "\n",
      "---\n",
      "\n",
      "|  |  |  |  |\n",
      "| --- | --- | --- | --- |\n",
      "| **Patient Name:** | Smith, Todd | **Ordered By:** | Dr. Who |\n",
      "| **Date of Birth:** | 12/11/1977 | **Sample Collection Date:** | 12/9/2020 | \n",
      "| **Gender**: | Male | **Sample Accession Date:** | 12/9/2020 |\n",
      "| **Sample Type:** | blood | **Report Data:** | 12/9/2020 |\n",
      "| **Sample ID/MRN:** | 1234567 | | |\n",
      "| **Test Accession:** | BRCA122345 | | | \n",
      "| **Test Ordered:** | BRCA2 sequencing & Deletion/Duplication analysis | | | \n",
      "| **Test Code:** | ABCD | | |\n",
      "| **Test Indictation** | Patient has family history of breast cancer. | | |\n",
      "\n",
      "---\n",
      "\n",
      "## Result: Postitive\n",
      "\n",
      "---\n",
      "\n",
      "| Gene   | Variant   | Zygosity     | Variant Classification   |\n",
      "|:-------|:----------|:-------------|:-------------------------|\n",
      "| BRCA2  | c.10G>T   | Heterozygous | Likely Pathogenic        |\n",
      "| BRCA2  | c.-172A>G | Heterozygous | Unknown                  |\n",
      "\n"
     ]
    }
   ],
   "source": [
    "from jinja2 import Environment, BaseLoader\n",
    "\n",
    "\n",
    "report_header_table = \"\"\"\n",
    "# Report\n",
    "\n",
    "## Sample-1\n",
    "\n",
    "---\n",
    "\n",
    "|  |  |  |  |\n",
    "| --- | --- | --- | --- |\n",
    "| **Patient Name:** | {{ patient_data.patient_last_name }}, {{ patient_data.patient_first_name }} | **Ordered By:** | {{test_data.test_requesting_physician}} |\n",
    "| **Date of Birth:** | {{patient_data.patient_dob}} | **Sample Collection Date:** | {{ sample_data.sample_collection_date }} | \n",
    "| **Gender**: | {{patient_data.patient_gender}} | **Sample Accession Date:** | {{ sample_data.sample_accession_date }} |\n",
    "| **Sample Type:** | {{sample_data.sample_type}} | **Report Data:** | {{report_data.report_date}} |\n",
    "| **Sample ID/MRN:** | {{ sample_data.sample_id }} | | |\n",
    "| **Test Accession:** | {{ test_data.test_accession }} | | | \n",
    "| **Test Ordered:** | {{test_data.test_ordered}} | | | \n",
    "| **Test Code:** | {{ test_data.test_code }} | | |\n",
    "| **Test Indictation** | {{ test_data.test_indication }} | | |\n",
    "\n",
    "---\n",
    "\n",
    "## Result: Postitive\n",
    "\n",
    "---\n",
    "\n",
    "{{ markdown_table_str }}\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "rtemplate = Environment(loader=BaseLoader()).from_string(report_header_table)\n",
    "rendered_report_header_table = rtemplate.render(**report_data)\n",
    "print(rendered_report_header_table)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "# Report\n",
       "\n",
       "## Sample-1\n",
       "\n",
       "---\n",
       "\n",
       "|  |  |  |  |\n",
       "| --- | --- | --- | --- |\n",
       "| **Patient Name:** | Smith, Todd | **Ordered By:** | Dr. Who |\n",
       "| **Date of Birth:** | 12/11/1977 | **Sample Collection Date:** | 12/9/2020 | \n",
       "| **Gender**: | Male | **Sample Accession Date:** | 12/9/2020 |\n",
       "| **Sample Type:** | blood | **Report Data:** | 12/9/2020 |\n",
       "| **Sample ID/MRN:** | 1234567 | | |\n",
       "| **Test Accession:** | BRCA122345 | | | \n",
       "| **Test Ordered:** | BRCA2 sequencing & Deletion/Duplication analysis | | | \n",
       "| **Test Code:** | ABCD | | |\n",
       "| **Test Indictation** | Patient has family history of breast cancer. | | |\n",
       "\n",
       "---\n",
       "\n",
       "## Result: Postitive\n",
       "\n",
       "---\n",
       "\n",
       "| Gene   | Variant   | Zygosity     | Variant Classification   |\n",
       "|:-------|:----------|:-------------|:-------------------------|\n",
       "| BRCA2  | c.10G>T   | Heterozygous | Likely Pathogenic        |\n",
       "| BRCA2  | c.-172A>G | Heterozygous | Unknown                  |\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# This doesn't work in the jupyterhub output\n",
    "from IPython.display import display, Markdown, Latex, HTML\n",
    "display(Markdown(rendered_report_header_table))\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
