{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Understanding the Ella CLI\n",
    "\n",
    "The Ella CLI is very robust and you can learn alot just by running subcommands and `--help`. You can manipulate any of the datamodels through the CLI.\n",
    "\n",
    "## Bring up the Dev Stack\n",
    "\n",
    "Bring up the dev stack and make the database schema.\n",
    "\n",
    "```bash\n",
    "# From the project root\n",
    "make dev\n",
    "make db\n",
    "```\n",
    "\n",
    "## Users, Groups, and FilterConfigs\n",
    "\n",
    "At a minimum to use Ella you need users, groups, and filter configs.\n",
    "\n",
    "### Add User Groups\n",
    "\n",
    "Since we'll be using the demo data in Ella we need to add the User Groups first, or else adding our users will fail because it will point to user groups that don't exist.\n",
    "\n",
    "```bash\n",
    "echo \"Adding user groups\"\n",
    "docker-compose -f local.yml exec \\\n",
    "\t\tella-web bash -c \"ella-cli users add_groups /ella/src/vardb/testdata/usergroups.json\"\n",
    "```\n",
    "\n",
    "### Add Users\n",
    "\n",
    "Now that we have the user groups we can add the users.\n",
    "\n",
    "```bash\n",
    "echo \"Adding users\"\n",
    "docker-compose -f local.yml \\\n",
    "\t\texec ella-web bash -c \"ella-cli users add_many /ella/src/vardb/testdata/users.json\"\n",
    "```\n",
    "\n",
    "### Add Filter Configs\n",
    "\n",
    "Filter configs can be very complex. The idea is to use a computer to filter what the variant scientist sees on the Ella UI. To get started we'll just add in the test data filter config\n",
    "\n",
    "```bash\n",
    "echo \"Adding filter configs\"\n",
    "docker-compose -f local.yml \\\n",
    "\t\texec ella-web bash -c \"ella-cli filterconfigs update /ella/src/vardb/testdata/filterconfigs.json\"\n",
    "```\n",
    "\n",
    "## Gene Panels\n",
    "\n",
    "If you hammer on the test data you'll see that there are plenty of gene panels you can use.\n",
    "\n",
    "\n",
    "```bash\n",
    "echo \"Create HBOC v01 gene panel\"\n",
    "docker-compose -f local.yml exec ella-web \\\n",
    "    bash -c \\\n",
    "    \"ella-cli deposit genepanel --genepanel_name HBOC --genepanel_version v01 \\\n",
    "        --transcripts_path /ella/src/vardb/testdata/clinicalGenePanels/HBOC_v01/HBOC_v01.transcripts.csv \\\n",
    "        --phenotypes_path /ella/src/vardb/testdata/clinicalGenePanels/HBOC_v01/HBOC_v01.phenotypes.csv\"\n",
    "\n",
    "echo \"Create HBOCUTV v01 gene panel\"\n",
    "docker-compose -f local.yml exec  ella-web \\\n",
    "    bash -c \\\n",
    "    \"ella-cli deposit genepanel --genepanel_name HBOCUTV --genepanel_version v01 \\\n",
    "        --transcripts_path /ella/src/vardb/testdata/clinicalGenePanels/HBOCUTV_v01/HBOCUTV_v01.transcripts.csv \\\n",
    "        --phenotypes_path /ella/src/vardb/testdata/clinicalGenePanels/HBOCUTV_v01/HBOCUTV_v01.phenotypes.csv\"\n",
    "\n",
    "echo \"Create Mendeliome v01 gene panel\"\n",
    "docker-compose -f local.yml exec  ella-web \\\n",
    "    bash -c \\\n",
    "    \"ella-cli deposit genepanel --genepanel_name Mendeliome --genepanel_version v01 \\\n",
    "        --transcripts_path /ella/src/vardb/testdata/clinicalGenePanels/Mendeliome_v01/Mendeliome_v01.transcripts.csv \\\n",
    "        --phenotypes_path /ella/src/vardb/testdata/clinicalGenePanels/Mendeliome_v01/Mendeliome_v01.phenotypes.csv\"\n",
    "\n",
    "echo \"Create Ciliopati v05 gene panel\"\n",
    "docker-compose -f local.yml exec ella-web \\\n",
    "    bash -c \\\n",
    "    \"ella-cli deposit genepanel --genepanel_name Ciliopati --genepanel_version v05 \\\n",
    "        --transcripts_path /ella/src/vardb/testdata/clinicalGenePanels/Ciliopati_v05/Ciliopati_v05.transcripts.csv \\\n",
    "        --phenotypes_path /ella/src/vardb/testdata/clinicalGenePanels/Ciliopati_v05/Ciliopati_v05.phenotypes.csv\"\n",
    "```\n",
    "\n",
    "## Analyses\n",
    "\n",
    "Analyses are associated to a genepanel, which is associated to a User group, and the User Group is associated to a filter config.\n",
    "\n",
    "The easiest way to load an analysis into Ella is to use the CLI and to load a VCF file that was annotated with the Ella Annotation Service. The VCF file should be named with the format `{analysis_name}.{genepanel_name}_{genepanel_version}.vcf`. If its compressed it should have a tabix index.\n",
    "\n",
    "```bash\n",
    "docker-compose -f local.yml \\\n",
    "\texec ella-web \\\n",
    "\tbash -c \"find /ella/src/vardb/testdata/analyses/default -name '*vcf' | grep -v Trio | grep -v brca_sample_2 | xargs -I {} ella-cli deposit analysis {}\"\n",
    "```\n",
    "\n",
    "If you see an error like:\n",
    "\n",
    "```bash\n",
    "psycopg2.OperationalError: stack depth limit exceeded\n",
    "HINT:  Increase the configuration parameter \"max_stack_depth\" (currently 2048kB), after ensuring the platform's stack depth limit is adequate.\n",
    "```\n",
    " \n",
    "This means that the machine you are on is not powerful enough to handle the database transaction. For the purposes of demonstration this doesn't really matter as long as you can load in at least a few analyses to look at. \n",
    "\n",
    "**Coming Soon** If you are planning on loading large analyses (whole exome, whole genome) I will soon have a reworked function to do that in Ella Admin.\n",
    "\n",
    "You'll notice we reverse grep a few vcf files. This is because they are mutli sample VCF files that require a ped file.\n",
    "\n",
    "```bash\n",
    "echo \"Adding trio analyses\"\n",
    "docker-compose -f local.yml \\\n",
    "\texec ella-web \\\n",
    "\tbash -c \"ella-cli deposit analysis \\\n",
    "\t\t--ped /ella/src/vardb/testdata/analyses/default/HG002-Trio.Mendeliome_v01/HG002-Trio.Mendeliome_v01.ped \\\n",
    "\t\t/ella/src/vardb/testdata/analyses/default/HG002-Trio.Mendeliome_v01/HG002-Trio.Mendeliome_v01.vcf\"\n",
    "```"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
