{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# The ELLA REST API\n",
    "\n",
    "Ella has a large REST API that can be used to fetch all kinds of data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "import json\n",
    "from pprint import pprint"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Querying the ELLA API\n",
    "\n",
    "In order to get the REST API you'll need to know what port Ella is running on. \n",
    "\n",
    "This can get a bit tricky because you need to pay attention to whether or not you are running within the docker container or the port exposed outside.\n",
    "\n",
    "In this case I am running in the jupyterhub notebook provided within the `ella-admin` jupyterhub stack. This means that I am using the network created by docker, so I query ella using the service name `ella-web` on port `5000`.\n",
    "\n",
    "(For the sake of brevity I got rid of some of the variables in the ella-web service.)\n",
    "\n",
    "```yaml\n",
    "  ella-web:\n",
    "    image: dabbleofdevops/ella:1.11.1\n",
    "    ports:\n",
    "      # External:Internal\n",
    "      # Expose the port running in the container as 5000 to my localhost as 5001\n",
    "      - 5001:5000\n",
    "```\n",
    "\n",
    "### Query Internally from within a Docker Container\n",
    "\n",
    "You could exec (`docker-compose exec -f local.yml ella-web` or `docker-compose exec -f local.yml jupyter`) into a docker container, or use the notebook to query the Ella API. If you do that you should use the url `http://{service-name}/{internal-port}`.\n",
    "\n",
    "### Query Externally from the Docker Host\n",
    "\n",
    "If you want to run this command externally, such as if you are using Postman, you need to use the external url. `http://localhost:5001` or `http://{ip-of-server}/{external-port}`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--------------------------\n",
      "dict_keys(['info', 'paths', 'tags', 'swagger', 'definitions', 'parameters'])\n",
      "--------------------------\n",
      "dict_keys(['Analysis', 'AnalysisInterpretation', 'AlleleInterpretation', 'Interpretation', 'Allele', 'Reference', 'ReferenceAssessment', 'AlleleAssessment', 'AlleleAssessmentInput', 'AlleleReport', 'User', 'Classification', 'Rule', 'Genepanel', 'Annotation', 'CustomAnnotation', 'Genotype'])\n",
      "--------------------------\n",
      "['/api/v1/acmg/alleles/',\n",
      " '/api/v1/acmg/classifications/',\n",
      " '/api/v1/alleles/',\n",
      " '/api/v1/alleles/by-gene/',\n",
      " '/api/v1/alleles/{allele_id}/analyses/',\n",
      " '/api/v1/alleleassessments/{aa_id}/',\n",
      " '/api/v1/alleleassessments/',\n",
      " '/api/v1/allelereports/',\n",
      " '/api/v1/allelereports/{ar_id}/',\n",
      " '/api/v1/analyses/',\n",
      " '/api/v1/analyses/{analysis_id}/',\n",
      " '/api/v1/broadcasts/',\n",
      " '/api/v1/customannotations/',\n",
      " '/api/v1/config/',\n",
      " '/api/v1/filterconfigs/{filterconfig_id}',\n",
      " '/api/v1/geneassessments/',\n",
      " '/api/v1/genepanels/',\n",
      " '/api/v1/genepanels/{name}/{version}/',\n",
      " '/api/v1/genepanels/{name}/{version}/stats/',\n",
      " '/api/v1/references/',\n",
      " '/api/v1/referenceassessments/{ra_id}/',\n",
      " '/api/v1/referenceassessments/',\n",
      " '/api/v1/search/',\n",
      " '/api/v1/search/options/',\n",
      " '/api/v1/users/',\n",
      " '/api/v1/users/{user_id}/',\n",
      " '/api/v1/overviews/alleles/',\n",
      " '/api/v1/overviews/alleles/finalized/',\n",
      " '/api/v1/overviews/analyses/',\n",
      " '/api/v1/overviews/analyses/finalized/',\n",
      " '/api/v1/overviews/userstats/',\n",
      " '/api/v1/import/service/jobs/{id}/',\n",
      " '/api/v1/import/service/jobs/',\n",
      " '/api/v1/import/service/running/',\n",
      " '/api/v1/import/service/samples/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/genepanels/{gp_name}/{gp_version}/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/genepanels/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/interpretations/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/interpretations/{interpretation_id}/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/interpretations/{interpretation_id}/alleles/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/start/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/markinterpretation/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/markreview/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/finalize/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/finalizeallele/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/override/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/actions/reopen/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/collisions/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/logs/',\n",
      " '/api/v1/workflows/alleles/{allele_id}/logs/{log_id}/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/stats/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/interpretations/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/genepanels/{gp_name}/{gp_version}/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/interpretations/{interpretation_id}/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/interpretations/{interpretation_id}/finishallowed',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/interpretations/{interpretation_id}/filteredalleles/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/interpretations/{interpretation_id}/alleles/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/start/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/marknotready/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/markinterpretation/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/markreview/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/markmedicalreview/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/finalize/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/finalizeallele/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/override/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/actions/reopen/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/collisions/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/logs/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/logs/{log_id}/',\n",
      " '/api/v1/workflows/analyses/{analysis_id}/filterconfigs/',\n",
      " '/api/v1/users/actions/login/',\n",
      " '/api/v1/users/actions/logout/',\n",
      " '/api/v1/users/actions/changepassword/',\n",
      " '/api/v1/users/currentuser/',\n",
      " '/api/v1/attachments/upload/',\n",
      " '/api/v1/attachments/',\n",
      " '/api/v1/igv/{filename}',\n",
      " '/api/v1/igv/search/',\n",
      " '/api/v1/igv/genepanel/{gp_name}/{gp_version}/',\n",
      " '/api/v1/igv/classifications/',\n",
      " '/api/v1/igv/variants/{analysis_id}/',\n",
      " '/api/v1/igv/tracks/{analysis_id}/',\n",
      " '/api/v1/igv/tracks/global/{filename}',\n",
      " '/api/v1/igv/tracks/usergroups/{usergroup_id}/{filename}',\n",
      " '/api/v1/igv/tracks/analyses/{analysis_id}/{filename}']\n"
     ]
    }
   ],
   "source": [
    "response = requests.get('http://ella-web:5000/api/v1/specs')\n",
    "content = response.content.decode('utf-8')\n",
    "api_spec_data = json.loads(content)\n",
    "\n",
    "print('--------------------------')\n",
    "pprint(api_spec_data.keys())\n",
    "print('--------------------------')\n",
    "pprint(api_spec_data['definitions'].keys())\n",
    "print('--------------------------')\n",
    "pprint(list(api_spec_data['paths'].keys()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Authenticating Against the REST API\n",
    "\n",
    "In order to authenticate against the REST API we'll have to pass in our username and password. If you ran the dev stack as \n",
    "\n",
    "```bash\n",
    "make dev\n",
    "make load\n",
    "```\n",
    "\n",
    "You should already have a test user setup.\n",
    "\n",
    "(Please note that the dev stack is for dev purposes only and you should change your own usernames and passwords!)\n",
    "\n",
    "Once we've authenticated we should have a cookie called 'AuthenticationToken'. Using the Python Requests session this is saved for us. Otherwise you will have to add the token to your cookies manually."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'AuthenticationToken': '0b5a3cc5679cd7bcdb82206b46b4d21c6546b1ba835568e44047fc5910385a7c'}\n"
     ]
    }
   ],
   "source": [
    "session = requests.Session()\n",
    "\n",
    "url = 'http://ella-web:5000'\n",
    "user_data = {\"username\": \"testuser1\", \"password\": \"Password#123\"}\n",
    "session.post('{}/api/v1/users/actions/login/'.format(url), json=user_data)\n",
    "print(session.cookies.get_dict())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that we're authenticated we can start running queries! "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Run Authenticated Queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'AuthenticationToken': '0b5a3cc5679cd7bcdb82206b46b4d21c6546b1ba835568e44047fc5910385a7c'}\n"
     ]
    }
   ],
   "source": [
    "response = session.get('{}/api/v1/workflows/analyses/2/interpretations/2/'.format(url))\n",
    "print(session.cookies.get_dict())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'date_last_update': '2020-11-28T06:44:25.638124+00:00',\n",
      " 'finalized': True,\n",
      " 'genepanel_name': 'HBOCUTV',\n",
      " 'genepanel_version': 'v01',\n",
      " 'id': 2,\n",
      " 'state': {'allele': {'3': {'allele_id': 3,\n",
      "                            'alleleassessment': {'allele_id': 3,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'},\n",
      "                                                                                       {'code': 'REQ_null_variant',\n",
      "                                                                                        'match': ['stop_gained'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_no_freq',\n",
      "                                                                                        'match': ['null_freq'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'frequencies.commonness'},\n",
      "                                                                                       {'code': 'PVS1',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None},\n",
      "                                                                                       {'code': 'PPxPM2',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 1},\n",
      "                            'allelereport': {'copiedFromId': 1,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': True},\n",
      "                            'workflow': {'reviewed': True}},\n",
      "                      '4': {'allele_id': 4,\n",
      "                            'alleleassessment': {'allele_id': 4,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'},\n",
      "                                                                                       {'code': 'REQ_null_variant',\n",
      "                                                                                        'match': ['frameshift_variant'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_no_freq',\n",
      "                                                                                        'match': ['null_freq'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'frequencies.commonness'},\n",
      "                                                                                       {'code': 'PVS1',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None},\n",
      "                                                                                       {'code': 'PPxPM2',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 2},\n",
      "                            'allelereport': {'copiedFromId': 2,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': True},\n",
      "                            'workflow': {'reviewed': True}},\n",
      "                      '5': {'allele_id': 5,\n",
      "                            'alleleassessment': {'allele_id': 5,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'},\n",
      "                                                                                       {'code': 'REQ_null_variant',\n",
      "                                                                                        'match': ['splice_donor_variant'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_no_freq',\n",
      "                                                                                        'match': ['null_freq'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'frequencies.commonness'},\n",
      "                                                                                       {'code': 'PVS1',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None},\n",
      "                                                                                       {'code': 'PPxPM2',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 3},\n",
      "                            'allelereport': {'copiedFromId': 3,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': True},\n",
      "                            'workflow': {'reviewed': True}},\n",
      "                      '6': {'allele_id': 6,\n",
      "                            'alleleassessment': {'allele_id': 6,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_missense',\n",
      "                                                                                        'match': ['missense_variant'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 4},\n",
      "                            'allelereport': {'copiedFromId': 4,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': False},\n",
      "                            'workflow': {'reviewed': True}},\n",
      "                      '7': {'allele_id': 7,\n",
      "                            'alleleassessment': {'allele_id': 7,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'},\n",
      "                                                                                       {'code': 'REQ_null_variant',\n",
      "                                                                                        'match': ['stop_gained'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_no_freq',\n",
      "                                                                                        'match': ['null_freq'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'frequencies.commonness'},\n",
      "                                                                                       {'code': 'PVS1',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None},\n",
      "                                                                                       {'code': 'PPxPM2',\n",
      "                                                                                        'match': None,\n",
      "                                                                                        'op': None,\n",
      "                                                                                        'source': None}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 5},\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                            'allelereport': {'copiedFromId': 5,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': True},\n",
      "                            'workflow': {'reviewed': True}},\n",
      "                      '8': {'allele_id': 8,\n",
      "                            'alleleassessment': {'allele_id': 8,\n",
      "                                                 'evaluation': {'acmg': {'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                                                        'match': ['LENI'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.last_exon_important'},\n",
      "                                                                                       {'code': 'REQ_GP_LOF_missense',\n",
      "                                                                                        'match': ['ANY'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'genepanel.disease_mode'},\n",
      "                                                                                       {'code': 'REQ_less_common',\n",
      "                                                                                        'match': ['less_common'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'frequencies.commonness'},\n",
      "                                                                                       {'code': 'REQ_no_aa_change',\n",
      "                                                                                        'match': ['synonymous_variant'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.consequences'},\n",
      "                                                                                       {'code': 'REQ_not_in_last_exon',\n",
      "                                                                                        'match': ['no'],\n",
      "                                                                                        'op': '$in',\n",
      "                                                                                        'source': 'transcript.in_last_exon'}]}},\n",
      "                                                 'reuse': True,\n",
      "                                                 'reuseCheckedId': 6},\n",
      "                            'allelereport': {'copiedFromId': 6,\n",
      "                                             'evaluation': {'comment': ''}},\n",
      "                            'analysis': {'comment': '',\n",
      "                                         'notrelevant': None,\n",
      "                                         'verification': 'verified'},\n",
      "                            'referenceassessments': [],\n",
      "                            'report': {'included': False},\n",
      "                            'workflow': {'reviewed': True}}},\n",
      "           'filterconfigId': 1,\n",
      "           'manuallyAddedAlleles': []},\n",
      " 'status': 'Done',\n",
      " 'user': {'abbrev_name': 'H. Ibsen',\n",
      "          'active': True,\n",
      "          'first_name': 'Henrik',\n",
      "          'full_name': 'Henrik Ibsen',\n",
      "          'id': 1,\n",
      "          'last_name': 'Ibsen',\n",
      "          'user_group_name': 'testgroup01',\n",
      "          'username': 'testuser1'},\n",
      " 'user_id': 1,\n",
      " 'user_state': {'allele': {'3': {'allele_id': 3,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False},\n",
      "                           '4': {'allele_id': 4,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False},\n",
      "                           '5': {'allele_id': 5,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False},\n",
      "                           '6': {'allele_id': 6,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False},\n",
      "                           '7': {'allele_id': 7,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False},\n",
      "                           '8': {'allele_id': 8,\n",
      "                                 'sections': {},\n",
      "                                 'showExcludedReferences': False}}},\n",
      " 'workflow_status': 'Interpretation'}\n"
     ]
    }
   ],
   "source": [
    "pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'date_deposited': '2020-10-19T10:46:43.419040+00:00',\n",
      " 'date_requested': None,\n",
      " 'genepanel': {'name': 'HBOCUTV', 'version': 'v01'},\n",
      " 'id': 2,\n",
      " 'interpretations': [{'analysis_id': 2,\n",
      "                      'date_last_update': '2020-11-28T06:44:25.638124+00:00',\n",
      "                      'finalized': True,\n",
      "                      'genepanel_name': 'HBOCUTV',\n",
      "                      'genepanel_version': 'v01',\n",
      "                      'id': 2,\n",
      "                      'status': 'Done',\n",
      "                      'user': {'abbrev_name': 'H. Ibsen',\n",
      "                               'active': True,\n",
      "                               'first_name': 'Henrik',\n",
      "                               'full_name': 'Henrik Ibsen',\n",
      "                               'id': 1,\n",
      "                               'last_name': 'Ibsen',\n",
      "                               'user_group_name': 'testgroup01',\n",
      "                               'username': 'testuser1'},\n",
      "                      'user_id': 1,\n",
      "                      'workflow_status': 'Interpretation'}],\n",
      " 'name': 'brca_sample_1.HBOCUTV_v01',\n",
      " 'report': None,\n",
      " 'samples': [{'affected': True,\n",
      "              'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "              'family_id': None,\n",
      "              'father_id': None,\n",
      "              'id': 2,\n",
      "              'identifier': 'brca_sample_1',\n",
      "              'mother_id': None,\n",
      "              'proband': True,\n",
      "              'sample_type': 'HTS',\n",
      "              'sex': None,\n",
      "              'sibling_id': None}],\n",
      " 'warnings': None}\n"
     ]
    }
   ],
   "source": [
    "response = session.get('{}/api/v1/analyses/2'.format(url))\n",
    "pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ella REST API Analysis Overview\n",
    "\n",
    "This is the sequence of events the REST API goes through when generating the Allele Interpretation Interface."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You can get this sequence by checking out the ella-web logs when you bring up an analysis page.\n",
    "\n",
    "```\n",
    "#GET /api/v1/workflows/analyses/2/interpretations/2/alleles/ \n",
    "GET /workflows/analyses/2/ HTTP/1.1\"\n",
    "\n",
    "GET /api/v1/config/\n",
    "GET /api/v1/users/currentuser/\n",
    "GET /api/v1/workflows/analyses/2/stats/\n",
    "GET /api/v1/analyses/2/\n",
    "GET /api/v1/workflows/analyses/2/filterconfigs -\n",
    "GET /api/v1/workflows/analyses/2/filterconfigs/ -\n",
    "GET /api/v1/workflows/analyses/2/interpretations/  -\n",
    "GET /api/v1/workflows/analyses/2/interpretations/2/filteredalleles/?filterconfig_id=1 -\n",
    "GET /api/v1/workflows/analyses/2/genepanels/HBOCUTV/v01/?allele_ids=3%2C4%2C5%2C6%2C7%2C8  -\n",
    "GET /api/v1/workflows/analyses/2/interpretations/2/alleles/?allele_ids=3%2C4%2C5%2C6%2C7%2C8&current=true&filterconfig_id=1 HTTP/1.1\" 200 -\n",
    "GET /api/v1/workflows/analyses/2/collisions/?allele_ids=3%2C4%2C5%2C6%2C7%2C8 HTTP/1.1\" 200 -\n",
    "GET /api/v1/references/?q=%7B%22id%22%3A%5B%5D%7D HTTP/1.1\" 200 -\n",
    "GET /api/v1/attachments/?q=%7B%22id%22%3A%5B%5D%7D HTTP/1.1\" 200 -\n",
    "GET /api/v1/references/?q=%7B%22pubmed_id%22%3A%5B17453335%2C8589730%2C18607349%2C28324225%2C18092194%2C9245992%2C10070953%2C22430266%2C20927582%2C25525159%2C21356067%2C16760289%2C16847550%2C18844490%2C28492532%2C19471317%2C10923033%5D%7D HTTP/1.1\" 200 -\n",
    "POST - http://localhost:5001/api/v1/acmg/alleles/ - {\"allele_ids\":[\"3\",\"4\",\"5\",\"6\",\"7\",\"8\"],\"gp_name\":\"HBOCUTV\",\"gp_version\":\"v01\",\"referenceassessments\":[]} - 4705 - 650ms\n",
    "\"POST /api/v1/acmg/alleles/ HTTP/1.1\" 200 -\n",
    "```\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#query_url = 'api/v1/workflows/analyses/2'\n",
    "#response = session.get('{}/{}'.format(url, query_url))\n",
    "#pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_url = 'api/v1/config/'\n",
    "response = session.get('{}/{}'.format(url, query_url))\n",
    "#pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'allele_count': 6}\n"
     ]
    }
   ],
   "source": [
    "query_url = 'api/v1/workflows/analyses/2/stats/'\n",
    "response = session.get('{}/{}'.format(url, query_url))\n",
    "pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'date_deposited': '2020-10-19T10:46:43.419040+00:00',\n",
      " 'date_requested': None,\n",
      " 'genepanel': {'name': 'HBOCUTV', 'version': 'v01'},\n",
      " 'id': 2,\n",
      " 'interpretations': [{'analysis_id': 2,\n",
      "                      'date_last_update': '2020-11-28T06:44:25.638124+00:00',\n",
      "                      'finalized': True,\n",
      "                      'genepanel_name': 'HBOCUTV',\n",
      "                      'genepanel_version': 'v01',\n",
      "                      'id': 2,\n",
      "                      'status': 'Done',\n",
      "                      'user': {'abbrev_name': 'H. Ibsen',\n",
      "                               'active': True,\n",
      "                               'first_name': 'Henrik',\n",
      "                               'full_name': 'Henrik Ibsen',\n",
      "                               'id': 1,\n",
      "                               'last_name': 'Ibsen',\n",
      "                               'user_group_name': 'testgroup01',\n",
      "                               'username': 'testuser1'},\n",
      "                      'user_id': 1,\n",
      "                      'workflow_status': 'Interpretation'}],\n",
      " 'name': 'brca_sample_1.HBOCUTV_v01',\n",
      " 'report': None,\n",
      " 'samples': [{'affected': True,\n",
      "              'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "              'family_id': None,\n",
      "              'father_id': None,\n",
      "              'id': 2,\n",
      "              'identifier': 'brca_sample_1',\n",
      "              'mother_id': None,\n",
      "              'proband': True,\n",
      "              'sample_type': 'HTS',\n",
      "              'sex': None,\n",
      "              'sibling_id': None}],\n",
      " 'warnings': None}\n"
     ]
    }
   ],
   "source": [
    "query_url = 'api/v1/analyses/2/'\n",
    "response = session.get('{}/{}'.format(url, query_url))\n",
    "pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "query_url = 'api/v1/workflows/analyses/2/filterconfigs/'\n",
    "response = session.get('{}/{}'.format(url, query_url))\n",
    "#pprint(json.loads(response.content.decode('utf-8')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'allele_ids': [3, 4, 5, 6, 7, 8],\n",
      " 'excluded_allele_ids': {'classification': [],\n",
      "                         'consequence': [],\n",
      "                         'frequency': [],\n",
      "                         'gene': [],\n",
      "                         'inheritancemodel': [],\n",
      "                         'ppy': [],\n",
      "                         'quality': [],\n",
      "                         'region': [],\n",
      "                         'segregation': []}}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[3, 4, 5, 6, 7, 8]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Get some filtered alleles\n",
    "\n",
    "query_url = 'api/v1/workflows/analyses/2/interpretations/2/filteredalleles/'\n",
    "payload = {'filterconfig': 1}\n",
    "response = session.get('{}/{}'.format(url, query_url), params=payload)\n",
    "response = json.loads(response.content.decode('utf-8'))\n",
    "pprint(response)\n",
    "allele_ids = response['allele_ids']\n",
    "allele_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'geneassessments': [],\n",
      " 'name': 'HBOCUTV',\n",
      " 'phenotypes': [{'description': '{Breast cancer, male, susceptibility to}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 6,\n",
      "                 'inheritance': 'AD',\n",
      "                 'omim_id': 114480},\n",
      "                {'description': '{Breast-ovarian cancer, familial, 2}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 2,\n",
      "                 'inheritance': 'AD',\n",
      "                 'omim_id': 612555},\n",
      "                {'description': 'Fanconi anemia, complementation group D1',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 4,\n",
      "                 'inheritance': 'AR',\n",
      "                 'omim_id': 605724},\n",
      "                {'description': '{Glioblastoma 3}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 7,\n",
      "                 'inheritance': 'AR',\n",
      "                 'omim_id': 613029},\n",
      "                {'description': '{Medulloblastoma}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 8,\n",
      "                 'inheritance': 'AD;AR',\n",
      "                 'omim_id': 155255},\n",
      "                {'description': '{Pancreatic cancer 2}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 9,\n",
      "                 'inheritance': '',\n",
      "                 'omim_id': 613347},\n",
      "                {'description': '{Prostate cancer}',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 10,\n",
      "                 'inheritance': 'AD',\n",
      "                 'omim_id': 176807},\n",
      "                {'description': 'Wilms tumor',\n",
      "                 'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                          'hgnc_id': 1101,\n",
      "                          'hgnc_symbol': 'BRCA2',\n",
      "                          'omim_entry_id': 600185},\n",
      "                 'id': 5,\n",
      "                 'inheritance': 'AD;SMu',\n",
      "                 'omim_id': 194070}],\n",
      " 'transcripts': [{'cds_end': 32972907,\n",
      "                  'cds_start': 32890597,\n",
      "                  'chromosome': '13',\n",
      "                  'corresponding_ensembl': 'ENST00000544455',\n",
      "                  'corresponding_lrg': None,\n",
      "                  'corresponding_refseq': None,\n",
      "                  'exon_ends': [32889804,\n",
      "                                32890664,\n",
      "                                32893462,\n",
      "                                32899321,\n",
      "                                32900287,\n",
      "                                32900419,\n",
      "                                32900750,\n",
      "                                32903629,\n",
      "                                32905167,\n",
      "                                32907524,\n",
      "                                32915333,\n",
      "                                32918790,\n",
      "                                32921033,\n",
      "                                32929425,\n",
      "                                32930746,\n",
      "                                32932066,\n",
      "                                32936830,\n",
      "                                32937670,\n",
      "                                32944694,\n",
      "                                32945237,\n",
      "                                32950928,\n",
      "                                32953652,\n",
      "                                32954050,\n",
      "                                32954282,\n",
      "                                32969070,\n",
      "                                32971181,\n",
      "                                32973809],\n",
      "                  'exon_starts': [32889616,\n",
      "                                  32890558,\n",
      "                                  32893213,\n",
      "                                  32899212,\n",
      "                                  32900237,\n",
      "                                  32900378,\n",
      "                                  32900635,\n",
      "                                  32903579,\n",
      "                                  32905055,\n",
      "                                  32906408,\n",
      "                                  32910401,\n",
      "                                  32918694,\n",
      "                                  32920963,\n",
      "                                  32928997,\n",
      "                                  32930564,\n",
      "                                  32931878,\n",
      "                                  32936659,\n",
      "                                  32937315,\n",
      "                                  32944538,\n",
      "                                  32945092,\n",
      "                                  32950806,\n",
      "                                  32953453,\n",
      "                                  32953886,\n",
      "                                  32954143,\n",
      "                                  32968825,\n",
      "                                  32971034,\n",
      "                                  32972298],\n",
      "                  'gene': {'ensembl_gene_id': 'ENSG00000139618',\n",
      "                           'hgnc_id': 1101,\n",
      "                           'hgnc_symbol': 'BRCA2',\n",
      "                           'omim_entry_id': 600185},\n",
      "                  'genome_reference': 'GRCh37',\n",
      "                  'strand': '+',\n",
      "                  'transcript_name': 'NM_000059.3',\n",
      "                  'tx_end': 32973809,\n",
      "                  'tx_start': 32889616}],\n",
      " 'version': 'v01'}\n"
     ]
    }
   ],
   "source": [
    "# /api/v1/workflows/analyses/2/genepanels/HBOCUTV/v01/?allele_ids=3%2C4%2C5%2C6%2C7%2C8 \n",
    "\n",
    "# TODO Get the Genepanel from the analysis def\n",
    "query_url = 'api/v1/workflows/analyses/2/genepanels/HBOCUTV/v01/'\n",
    "payload = {'allele_ids': allele_ids}\n",
    "response = session.get('{}/{}'.format(url, query_url), params=payload)\n",
    "response = json.loads(response.content.decode('utf-8'))\n",
    "pprint(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "http://ella-web:5000/api/v1/workflows/analyses/2/interpretations/2/alleles/?allele_ids=3%2C4%2C5%2C6%2C7%2C8&current=True&filterconfig_id=1\n",
      "6\n",
      "[{'allele_assessment': {'allele_id': 3,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 3,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': '4',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:49:44.630277+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [{'code': 'PVS1',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': '5104d287-683d-4391-a75b-5694c65f2b88'},\n",
      "                                                             {'code': 'PPxPM2',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': 'b44315f8-2db9-4605-b824-acf51e50ac31'}],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'},\n",
      "                                                              {'code': 'REQ_null_variant',\n",
      "                                                               'match': ['stop_gained'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_no_freq',\n",
      "                                                               'match': ['null_freq'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'frequencies.commonness'},\n",
      "                                                              {'code': 'PVS1',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None},\n",
      "                                                              {'code': 'PPxPM2',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None}],\n",
      "                                                'suggested_classification': 4},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 1,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448206.50482,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n",
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 3,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:49:44.640931+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 1,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448206.53819,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 3,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '01/11/2017',\n",
      "                                                     'rcv': 'SCV000296805',\n",
      "                                                     'submitter': 'GeneKor MSA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '18/10/2016',\n",
      "                                                     'rcv': 'SCV000323955',\n",
      "                                                     'submitter': 'ENIGMA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '18/10/2016',\n",
      "                                                     'rcv': 'RCV000257718',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': '51063'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '01/11/2017',\n",
      "                                                     'rcv': 'RCV000239090',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'not '\n",
      "                                                                   'provided',\n",
      "                                                     'variant_id': '51063'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '02/10/2015',\n",
      "                                                     'rcv': 'SCV000326495',\n",
      "                                                     'submitter': 'CIMBA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''}],\n",
      "                                          'variant_description': 'reviewed by '\n",
      "                                                                 'expert panel',\n",
      "                                          'variant_id': 51063},\n",
      "                              'HGMD': {'acc_num': 'CM082514',\n",
      "                                       'codon': 4,\n",
      "                                       'disease': 'Breast and/or ovarian '\n",
      "                                                  'cancer',\n",
      "                                       'tag': 'DM'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 17453335,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. No '\n",
      "                                                         'comments.'},\n",
      "                                 'sources': ['HGMD']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.10G>T',\n",
      "                                  'HGVSc_short': 'c.10G>T',\n",
      "                                  'HGVSp': 'p.Gly4Ter',\n",
      "                                  'amino_acids': 'G/*',\n",
      "                                  'coding_region_distance': 0,\n",
      "                                  'codons': 'Gga/Tga',\n",
      "                                  'consequences': ['stop_gained'],\n",
      "                                  'dbsnp': ['rs397507571'],\n",
      "                                  'exon': '2/27',\n",
      "                                  'exon_distance': 0,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'protein': 'NP_000050.2',\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'},\n",
      "                                 {'consequences': ['upstream_gene_variant'],\n",
      "                                  'dbsnp': ['rs397507571'],\n",
      "                                  'hgnc_id': 37116,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'strand': -1,\n",
      "                                  'symbol': 'ZAR1L',\n",
      "                                  'transcript': 'NM_001136571.1'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'G',\n",
      "  'change_to': 'T',\n",
      "  'change_type': 'SNP',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 3,\n",
      "  'open_end_position': 32890607,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'REF (G)': 107, 'T': 80},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'G/T',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 3,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': False,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': True},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32890606,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'T',\n",
      "  'vcf_pos': 32890607,\n",
      "  'vcf_ref': 'G'},\n",
      " {'allele_assessment': {'allele_id': 4,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 4,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': '4',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:50:00.770452+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [{'code': 'PVS1',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': '9f3d29e2-aa9c-477a-9203-ad2419b56212'},\n",
      "                                                             {'code': 'PPxPM2',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': 'a3d24926-c0c4-407f-af82-8d6f5b4bb6f4'}],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'},\n",
      "                                                              {'code': 'REQ_null_variant',\n",
      "                                                               'match': ['frameshift_variant'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_no_freq',\n",
      "                                                               'match': ['null_freq'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'frequencies.commonness'},\n",
      "                                                              {'code': 'PVS1',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None},\n",
      "                                                              {'code': 'PPxPM2',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None}],\n",
      "                                                'suggested_classification': 4},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 2,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448190.371559,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n",
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 4,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:50:00.778677+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 2,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448190.402037,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 4,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '12/12/2016',\n",
      "                                                     'rcv': 'RCV000131871',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'cancer-predisposing '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': '51814'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '21/01/2016',\n",
      "                                                     'rcv': 'SCV000279262',\n",
      "                                                     'submitter': 'GeneDx',\n",
      "                                                     'traitnames': 'Not '\n",
      "                                                                   'Provided',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '22/04/2016',\n",
      "                                                     'rcv': 'SCV000282398',\n",
      "                                                     'submitter': 'ENIGMA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '22/04/2016',\n",
      "                                                     'rcv': 'RCV000113063',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': '51814'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '06/01/2016',\n",
      "                                                     'rcv': 'SCV000694829',\n",
      "                                                     'submitter': 'Integrated '\n",
      "                                                                  'Genetics/Laboratory '\n",
      "                                                                  'Corporation '\n",
      "                                                                  'of America',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '02/10/2015',\n",
      "                                                     'rcv': 'SCV000327129',\n",
      "                                                     'submitter': 'CIMBA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '23/09/2016',\n",
      "                                                     'rcv': 'SCV000186926',\n",
      "                                                     'submitter': 'Ambry '\n",
      "                                                                  'Genetics',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'cancer-predisposing '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '29/05/2002',\n",
      "                                                     'rcv': 'SCV000146071',\n",
      "                                                     'submitter': 'BIC (BRCA2)',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '21/01/2016',\n",
      "                                                     'rcv': 'RCV000219019',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'not '\n",
      "                                                                   'provided',\n",
      "                                                     'variant_id': '51814'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '12/12/2016',\n",
      "                                                     'rcv': 'SCV000683670',\n",
      "                                                     'submitter': 'Color '\n",
      "                                                                  'Genomics, '\n",
      "                                                                  'Inc.',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '17/02/2015',\n",
      "                                                     'rcv': 'SCV000591655',\n",
      "                                                     'submitter': 'Department '\n",
      "                                                                  'of '\n",
      "                                                                  'Pathology '\n",
      "                                                                  'and '\n",
      "                                                                  'Laboratory '\n",
      "                                                                  'Medicine,Sinai '\n",
      "                                                                  'Health '\n",
      "                                                                  'System',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '31/01/2014',\n",
      "                                                     'rcv': 'SCV000587530',\n",
      "                                                     'submitter': 'Research '\n",
      "                                                                  'Molecular '\n",
      "                                                                  'Genetics '\n",
      "                                                                  \"Laboratory,Women's \"\n",
      "                                                                  'College '\n",
      "                                                                  'Hospital, '\n",
      "                                                                  'University '\n",
      "                                                                  'of Toronto',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '06/01/2016',\n",
      "                                                     'rcv': 'RCV000496563',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'breast and '\n",
      "                                                                   'ovarian '\n",
      "                                                                   'cancer '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': '51814'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '01/07/2016',\n",
      "                                                     'rcv': 'RCV000044604',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Familial '\n",
      "                                                                   'cancer of '\n",
      "                                                                   'breast',\n",
      "                                                     'variant_id': '51814'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '01/07/2016',\n",
      "                                                     'rcv': 'SCV000296839',\n",
      "                                                     'submitter': 'GeneKor MSA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''}],\n",
      "                                          'variant_description': 'reviewed by '\n",
      "                                                                 'expert panel',\n",
      "                                          'variant_id': 51814},\n",
      "                              'HGMD': {'acc_num': 'CD961847',\n",
      "                                       'codon': 16,\n",
      "                                       'disease': 'Breast cancer',\n",
      "                                       'tag': 'DM'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 8589730,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. aka '\n",
      "                                                         '279delAC.'},\n",
      "                                 'sources': ['HGMD', 'CLINVAR']},\n",
      "                                {'pubmed_id': 18607349,\n",
      "                                 'source_info': {'HGMD': 'Functional '\n",
      "                                                         'characterisation. No '\n",
      "                                                         'comments.'},\n",
      "                                 'sources': ['HGMD', 'CLINVAR']},\n",
      "                                {'pubmed_id': 28324225,\n",
      "                                 'source_info': {'HGMD': 'Additional '\n",
      "                                                         'phenotype. Breast '\n",
      "                                                         'and/or ovarian '\n",
      "                                                         'cancer.'},\n",
      "                                 'sources': ['HGMD']},\n",
      "                                {'pubmed_id': 18092194,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 9245992,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 10070953,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 22430266,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 20927582,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.51_52delAC',\n",
      "                                  'HGVSc_short': 'c.51_52delAC',\n",
      "                                  'HGVSp': 'p.Arg18LeufsTer12',\n",
      "                                  'amino_acids': 'T/X',\n",
      "                                  'coding_region_distance': 0,\n",
      "                                  'codons': 'ACa/a',\n",
      "                                  'consequences': ['frameshift_variant'],\n",
      "                                  'exon': '2/27',\n",
      "                                  'exon_distance': 0,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'protein': 'NP_000050.2',\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'},\n",
      "                                 {'consequences': ['upstream_gene_variant'],\n",
      "                                  'hgnc_id': 37116,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'strand': -1,\n",
      "                                  'symbol': 'ZAR1L',\n",
      "                                  'transcript': 'NM_001136571.1'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'AC',\n",
      "  'change_to': '',\n",
      "  'change_type': 'del',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 4,\n",
      "  'open_end_position': 32890647,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'G': 80, 'REF (GAC)': 107},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'AC/-',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 4,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': True,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': False},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32890645,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'G',\n",
      "  'vcf_pos': 32890645,\n",
      "  'vcf_ref': 'GAC'},\n",
      " {'allele_assessment': {'allele_id': 5,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 5,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': '4',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:50:14.357702+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [{'code': 'PVS1',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': 'a6640f97-6c04-4d6c-80d8-d7316097c2b3'},\n",
      "                                                             {'code': 'PPxPM2',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': '78328515-d040-45c6-8e23-21fee75294c5'}],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'},\n",
      "                                                              {'code': 'REQ_null_variant',\n",
      "                                                               'match': ['splice_donor_variant'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_no_freq',\n",
      "                                                               'match': ['null_freq'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'frequencies.commonness'},\n",
      "                                                              {'code': 'PVS1',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None},\n",
      "                                                              {'code': 'PPxPM2',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None}],\n",
      "                                                'suggested_classification': 4},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 3,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448176.792547,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 5,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:50:14.366927+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 3,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448176.81548,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 5,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '24/02/2003',\n",
      "                                                     'rcv': 'RCV000113080',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': '52162'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '24/02/2003',\n",
      "                                                     'rcv': 'SCV000146095',\n",
      "                                                     'submitter': 'BIC (BRCA2)',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': ''}],\n",
      "                                          'variant_description': 'no assertion '\n",
      "                                                                 'criteria '\n",
      "                                                                 'provided',\n",
      "                                          'variant_id': 52162},\n",
      "                              'HGMD': {'acc_num': 'CS087186',\n",
      "                                       'disease': 'Breast cancer',\n",
      "                                       'tag': 'DM'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 18092194,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. Descr. in '\n",
      "                                                         'Suppl. Table 1 '\n",
      "                                                         '(online).'},\n",
      "                                 'sources': ['HGMD']},\n",
      "                                {'pubmed_id': 25525159,\n",
      "                                 'source_info': {'HGMD': 'Additional report. '\n",
      "                                                         'predicted to induce '\n",
      "                                                         'a large splicing '\n",
      "                                                         'change - Table S4.'},\n",
      "                                 'sources': ['HGMD']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.67+2T>A',\n",
      "                                  'HGVSc_short': 'c.67+2T>A',\n",
      "                                  'coding_region_distance': None,\n",
      "                                  'consequences': ['splice_donor_variant'],\n",
      "                                  'dbsnp': ['rs81002885'],\n",
      "                                  'exon_distance': 2,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'intron': '2/26',\n",
      "                                  'is_canonical': True,\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'},\n",
      "                                 {'consequences': ['upstream_gene_variant'],\n",
      "                                  'dbsnp': ['rs81002885'],\n",
      "                                  'hgnc_id': 37116,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'strand': -1,\n",
      "                                  'symbol': 'ZAR1L',\n",
      "                                  'transcript': 'NM_001136571.1'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'T',\n",
      "  'change_to': 'A',\n",
      "  'change_type': 'SNP',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 5,\n",
      "  'open_end_position': 32890666,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'A': 80, 'REF (T)': 107},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'T/A',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 5,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': False,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': True},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32890665,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'A',\n",
      "  'vcf_pos': 32890666,\n",
      "  'vcf_ref': 'T'},\n",
      " {'allele_assessment': {'allele_id': 6,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 6,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': 'U',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:50:26.990666+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_missense',\n",
      "                                                               'match': ['missense_variant'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'}],\n",
      "                                                'suggested_classification': None},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 4,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448164.170551,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n",
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 6,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:50:26.999585+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 4,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448164.184426,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 6,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'not '\n",
      "                                                                                    'provided',\n",
      "                                                     'last_evaluated': '',\n",
      "                                                     'rcv': 'SCV000678953',\n",
      "                                                     'submitter': 'ClinVar '\n",
      "                                                                  'Staff, '\n",
      "                                                                  'National '\n",
      "                                                                  'Center for '\n",
      "                                                                  'Biotechnology '\n",
      "                                                                  'Information '\n",
      "                                                                  '(NCBI)',\n",
      "                                                     'traitnames': 'Familial '\n",
      "                                                                   'cancer of '\n",
      "                                                                   'breast',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'not '\n",
      "                                                                                    'provided',\n",
      "                                                     'last_evaluated': '',\n",
      "                                                     'rcv': 'RCV000577171',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Familial '\n",
      "                                                                   'cancer of '\n",
      "                                                                   'breast',\n",
      "                                                     'variant_id': '52303'}],\n",
      "                                          'variant_description': 'no assertion '\n",
      "                                                                 'criteria '\n",
      "                                                                 'provided',\n",
      "                                          'variant_id': 52303},\n",
      "                              'HGMD': {'acc_num': 'CM111973',\n",
      "                                       'codon': 24,\n",
      "                                       'disease': 'Breast and/or ovarian '\n",
      "                                                  'cancer',\n",
      "                                       'tag': 'DM'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {'ExAC': {'count': {'AFR': 0,\n",
      "                                                    'AMR': 0,\n",
      "                                                    'EAS': 0,\n",
      "                                                    'FIN': 2,\n",
      "                                                    'G': 2,\n",
      "                                                    'Het': 2,\n",
      "                                                    'NFE': 0,\n",
      "                                                    'OTH': 0,\n",
      "                                                    'SAS': 0},\n",
      "                                          'freq': {'AFR': 0.0,\n",
      "                                                   'AMR': 0.0,\n",
      "                                                   'EAS': 0.0,\n",
      "                                                   'FIN': 0.00030303030303030303,\n",
      "                                                   'G': 1.6784155756965425e-05,\n",
      "                                                   'NFE': 0.0,\n",
      "                                                   'OTH': 0.0,\n",
      "                                                   'SAS': 0.0},\n",
      "                                          'hom': {'AFR': 0,\n",
      "                                                  'AMR': 0,\n",
      "                                                  'EAS': 0,\n",
      "                                                  'FIN': 0,\n",
      "                                                  'G': 0,\n",
      "                                                  'NFE': 0,\n",
      "                                                  'OTH': 0,\n",
      "                                                  'SAS': 0},\n",
      "                                          'num': {'AFR': 9500,\n",
      "                                                  'AMR': 11530,\n",
      "                                                  'EAS': 8550,\n",
      "                                                  'FIN': 6600,\n",
      "                                                  'G': 119160,\n",
      "                                                  'NFE': 65594,\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                  'OTH': 894,\n",
      "                                                  'SAS': 16492}},\n",
      "                                 'GNOMAD_EXOMES': {'count': {'AFR': 0,\n",
      "                                                             'AMR': 0,\n",
      "                                                             'ASJ': 0,\n",
      "                                                             'EAS': 0,\n",
      "                                                             'FIN': 5,\n",
      "                                                             'Female': 4,\n",
      "                                                             'G': 5,\n",
      "                                                             'Male': 1,\n",
      "                                                             'NFE': 0,\n",
      "                                                             'OTH': 0,\n",
      "                                                             'SAS': 0},\n",
      "                                                   'filter': {'G': ['PASS']},\n",
      "                                                   'freq': {'AFR': 0.0,\n",
      "                                                            'AMR': 0.0,\n",
      "                                                            'ASJ': 0.0,\n",
      "                                                            'EAS': 0.0,\n",
      "                                                            'FIN': 0.0002243560979987436,\n",
      "                                                            'Female': 3.604382929642445e-05,\n",
      "                                                            'G': 2.0363778539835625e-05,\n",
      "                                                            'Male': 7.431739472941036e-06,\n",
      "                                                            'NFE': 0.0,\n",
      "                                                            'OTH': 0.0,\n",
      "                                                            'SAS': 0.0},\n",
      "                                                   'hom': {'AFR': 0,\n",
      "                                                           'AMR': 0,\n",
      "                                                           'ASJ': 0,\n",
      "                                                           'EAS': 0,\n",
      "                                                           'FIN': 0,\n",
      "                                                           'Female': 0,\n",
      "                                                           'G': 0,\n",
      "                                                           'Male': 0,\n",
      "                                                           'NFE': 0,\n",
      "                                                           'OTH': 0,\n",
      "                                                           'SAS': 0},\n",
      "                                                   'num': {'AFR': 15200,\n",
      "                                                           'AMR': 33558,\n",
      "                                                           'ASJ': 9828,\n",
      "                                                           'EAS': 17220,\n",
      "                                                           'FIN': 22286,\n",
      "                                                           'Female': 110976,\n",
      "                                                           'G': 245534,\n",
      "                                                           'Male': 134558,\n",
      "                                                           'NFE': 111296,\n",
      "                                                           'OTH': 5474,\n",
      "                                                           'SAS': 30672}}},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 21356067,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. No '\n",
      "                                                         'comments.'},\n",
      "                                 'sources': ['HGMD', 'CLINVAR']},\n",
      "                                {'pubmed_id': 25525159,\n",
      "                                 'source_info': {'HGMD': 'Additional report. '\n",
      "                                                         'predicted to induce '\n",
      "                                                         'a large splicing '\n",
      "                                                         'change - Table S4.'},\n",
      "                                 'sources': ['HGMD']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.72A>T',\n",
      "                                  'HGVSc_short': 'c.72A>T',\n",
      "                                  'HGVSp': 'p.Leu24Phe',\n",
      "                                  'amino_acids': 'L/F',\n",
      "                                  'coding_region_distance': 0,\n",
      "                                  'codons': 'ttA/ttT',\n",
      "                                  'consequences': ['missense_variant'],\n",
      "                                  'dbsnp': ['rs397507909'],\n",
      "                                  'exon': '3/27',\n",
      "                                  'exon_distance': 0,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'protein': 'NP_000050.2',\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'A',\n",
      "  'change_to': 'T',\n",
      "  'change_type': 'SNP',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 6,\n",
      "  'open_end_position': 32893218,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'REF (A)': 107, 'T': 80},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'A/T',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 6,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': False,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': True},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32893217,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'T',\n",
      "  'vcf_pos': 32893218,\n",
      "  'vcf_ref': 'A'},\n",
      " {'allele_assessment': {'allele_id': 7,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 7,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': '4',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:50:38.883637+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [{'code': 'PVS1',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': '6fb71fb5-b847-49c4-b787-f51f9a5a4aaf'},\n",
      "                                                             {'code': 'PPxPM2',\n",
      "                                                              'comment': '',\n",
      "                                                              'match': None,\n",
      "                                                              'op': None,\n",
      "                                                              'source': 'suggested',\n",
      "                                                              'uuid': '3cfaf648-2f7b-4863-b85f-884ac4bb3f99'}],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'},\n",
      "                                                              {'code': 'REQ_null_variant',\n",
      "                                                               'match': ['stop_gained'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_no_freq',\n",
      "                                                               'match': ['null_freq'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'frequencies.commonness'},\n",
      "                                                              {'code': 'PVS1',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None},\n",
      "                                                              {'code': 'PPxPM2',\n",
      "                                                               'match': None,\n",
      "                                                               'op': None,\n",
      "                                                               'source': None}],\n",
      "                                                'suggested_classification': 4},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 5,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448152.284614,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n",
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 7,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:50:38.892853+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 5,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448152.292575,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 7,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '18/10/2016',\n",
      "                                                     'rcv': 'SCV000324794',\n",
      "                                                     'submitter': 'ENIGMA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                     'last_evaluated': '18/10/2016',\n",
      "                                                     'rcv': 'RCV000256555',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': '52900'},\n",
      "                                                    {'clinical_significance_descr': 'Pathogenic',\n",
      "                                                     'last_evaluated': '02/10/2015',\n",
      "                                                     'rcv': 'SCV000328170',\n",
      "                                                     'submitter': 'CIMBA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''}],\n",
      "                                          'variant_description': 'reviewed by '\n",
      "                                                                 'expert panel',\n",
      "                                          'variant_id': 52900},\n",
      "                              'HGMD': {'acc_num': 'CM067653',\n",
      "                                       'codon': 33,\n",
      "                                       'disease': 'Breast and/or ovarian '\n",
      "                                                  'cancer',\n",
      "                                       'tag': 'DM'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 16760289,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. No '\n",
      "                                                         'comments.'},\n",
      "                                 'sources': ['HGMD']},\n",
      "                                {'pubmed_id': 16847550,\n",
      "                                 'source_info': {'HGMD': 'Additional report. '\n",
      "                                                         'No comments.'},\n",
      "                                 'sources': ['HGMD']},\n",
      "                                {'pubmed_id': 25525159,\n",
      "                                 'source_info': {'HGMD': 'Additional report. '\n",
      "                                                         'predicted to induce '\n",
      "                                                         'a large splicing '\n",
      "                                                         'change - Table S4.'},\n",
      "                                 'sources': ['HGMD']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.97G>T',\n",
      "                                  'HGVSc_short': 'c.97G>T',\n",
      "                                  'HGVSp': 'p.Glu33Ter',\n",
      "                                  'amino_acids': 'E/*',\n",
      "                                  'coding_region_distance': 0,\n",
      "                                  'codons': 'Gaa/Taa',\n",
      "                                  'consequences': ['stop_gained'],\n",
      "                                  'dbsnp': ['rs397508065'],\n",
      "                                  'exon': '3/27',\n",
      "                                  'exon_distance': 0,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'protein': 'NP_000050.2',\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'G',\n",
      "  'change_to': 'T',\n",
      "  'change_type': 'SNP',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 7,\n",
      "  'open_end_position': 32893243,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'REF (G)': 107, 'T': 80},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'G/T',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 7,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': False,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': True},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32893242,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'T',\n",
      "  'vcf_pos': 32893243,\n",
      "  'vcf_ref': 'G'},\n",
      " {'allele_assessment': {'allele_id': 8,\n",
      "                        'analysis_id': 2,\n",
      "                        'annotation_id': 8,\n",
      "                        'attachment_ids': [],\n",
      "                        'classification': 'U',\n",
      "                        'custom_annotation_id': None,\n",
      "                        'date_created': '2020-10-19T10:50:47.757943+00:00',\n",
      "                        'date_superceeded': None,\n",
      "                        'evaluation': {'acmg': {'included': [],\n",
      "                                                'suggested': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                                                               'match': ['LENI'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.last_exon_important'},\n",
      "                                                              {'code': 'REQ_GP_LOF_missense',\n",
      "                                                               'match': ['ANY'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'genepanel.disease_mode'},\n",
      "                                                              {'code': 'REQ_less_common',\n",
      "                                                               'match': ['less_common'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'frequencies.commonness'},\n",
      "                                                              {'code': 'REQ_no_aa_change',\n",
      "                                                               'match': ['synonymous_variant'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.consequences'},\n",
      "                                                              {'code': 'REQ_not_in_last_exon',\n",
      "                                                               'match': ['no'],\n",
      "                                                               'op': '$in',\n",
      "                                                               'source': 'transcript.in_last_exon'}],\n",
      "                                                'suggested_classification': None},\n",
      "                                       'classification': {'comment': ''},\n",
      "                                       'external': {'comment': ''},\n",
      "                                       'frequency': {'comment': ''},\n",
      "                                       'prediction': {'comment': ''},\n",
      "                                       'reference': {'comment': ''}},\n",
      "                        'genepanel_name': 'HBOCUTV',\n",
      "                        'genepanel_version': 'v01',\n",
      "                        'id': 6,\n",
      "                        'previous_assessment_id': None,\n",
      "                        'seconds_since_update': 3448143.416361,\n",
      "                        'user': {'abbrev_name': 'H. Ibsen',\n",
      "                                 'active': True,\n",
      "                                 'first_name': 'Henrik',\n",
      "                                 'full_name': 'Henrik Ibsen',\n",
      "                                 'id': 1,\n",
      "                                 'last_name': 'Ibsen',\n",
      "                                 'user_group_name': 'testgroup01',\n",
      "                                 'username': 'testuser1'},\n",
      "                        'user_id': 1,\n",
      "                        'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                        'usergroup_id': 1},\n",
      "  'allele_report': {'allele_id': 8,\n",
      "                    'analysis_id': 2,\n",
      "                    'date_created': '2020-10-19T10:50:47.766311+00:00',\n",
      "                    'date_superceeded': None,\n",
      "                    'evaluation': {'comment': ''},\n",
      "                    'id': 6,\n",
      "                    'previous_report_id': None,\n",
      "                    'seconds_since_update': 3448143.420534,\n",
      "                    'user': {'abbrev_name': 'H. Ibsen',\n",
      "                             'active': True,\n",
      "                             'first_name': 'Henrik',\n",
      "                             'full_name': 'Henrik Ibsen',\n",
      "                             'id': 1,\n",
      "                             'last_name': 'Ibsen',\n",
      "                             'user_group_name': 'testgroup01',\n",
      "                             'username': 'testuser1'},\n",
      "                    'user_id': 1,\n",
      "                    'usergroup': {'id': 1, 'name': 'testgroup01'},\n",
      "                    'usergroup_id': 1},\n",
      "  'annotation': {'annotation_id': 8,\n",
      "                 'external': {'CLINVAR': {'items': [{'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '14/06/2016',\n",
      "                                                     'rcv': 'RCV000377649',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Fanconi '\n",
      "                                                                   'anemia',\n",
      "                                                     'variant_id': '51234'},\n",
      "                                                    {'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '02/04/2015',\n",
      "                                                     'rcv': 'RCV000162901',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'cancer-predisposing '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': '51234'},\n",
      "                                                    {'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '14/06/2016',\n",
      "                                                     'rcv': 'SCV000383604',\n",
      "                                                     'submitter': 'Illumina',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'Breast and '\n",
      "                                                                   'Ovarian '\n",
      "                                                                   'Cancer',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '29/06/2017',\n",
      "                                                     'rcv': 'SCV000578005',\n",
      "                                                     'submitter': 'ENIGMA',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '15/11/2012',\n",
      "                                                     'rcv': 'RCV000499809',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'not '\n",
      "                                                                   'specified',\n",
      "                                                     'variant_id': '51234'},\n",
      "                                                    {'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '04/07/2014',\n",
      "                                                     'rcv': 'SCV000213388',\n",
      "                                                     'submitter': 'Ambry '\n",
      "                                                                  'Genetics',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'cancer-predisposing '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '29/06/2017',\n",
      "                                                     'rcv': 'RCV000113265',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': '51234'},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '03/01/2018',\n",
      "                                                     'rcv': 'SCV000071946',\n",
      "                                                     'submitter': 'Invitae',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '18/09/2010',\n",
      "                                                     'rcv': 'SCV000146362',\n",
      "                                                     'submitter': 'BIC (BRCA2)',\n",
      "                                                     'traitnames': 'Breast-ovarian '\n",
      "                                                                   'cancer, '\n",
      "                                                                   'familial 2',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign',\n",
      "                                                     'last_evaluated': '15/11/2012',\n",
      "                                                     'rcv': 'SCV000591663',\n",
      "                                                     'submitter': 'Department '\n",
      "                                                                  'of '\n",
      "                                                                  'Pathology '\n",
      "                                                                  'and '\n",
      "                                                                  'Laboratory '\n",
      "                                                                  'Medicine,Sinai '\n",
      "                                                                  'Health '\n",
      "                                                                  'System',\n",
      "                                                     'traitnames': 'not '\n",
      "                                                                   'specified',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '14/06/2016',\n",
      "                                                     'rcv': 'SCV000383605',\n",
      "                                                     'submitter': 'Illumina',\n",
      "                                                     'traitnames': 'Fanconi '\n",
      "                                                                   'Anemia',\n",
      "                                                     'variant_id': ''},\n",
      "                                                    {'clinical_significance_descr': 'Benign/Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '03/01/2018',\n",
      "                                                     'rcv': 'RCV000167850',\n",
      "                                                     'submitter': '',\n",
      "                                                     'traitnames': 'Hereditary '\n",
      "                                                                   'breast and '\n",
      "                                                                   'ovarian '\n",
      "                                                                   'cancer '\n",
      "                                                                   'syndrome',\n",
      "                                                     'variant_id': '51234'},\n",
      "                                                    {'clinical_significance_descr': 'Likely '\n",
      "                                                                                    'benign',\n",
      "                                                     'last_evaluated': '02/04/2015',\n",
      "                                                     'rcv': 'SCV000683467',\n",
      "                                                     'submitter': 'Color '\n",
      "                                                                  'Genomics, '\n",
      "                                                                  'Inc.',\n",
      "                                                     'traitnames': '',\n",
      "                                                     'variant_id': ''}],\n",
      "                                          'variant_description': 'reviewed by '\n",
      "                                                                 'expert panel',\n",
      "                                          'variant_id': 51234},\n",
      "                              'HGMD': {'acc_num': 'CM173583',\n",
      "                                       'codon': 66,\n",
      "                                       'disease': 'Breast and/or ovarian '\n",
      "                                                  'cancer',\n",
      "                                       'tag': 'DM?'}},\n",
      "                 'filtered_transcripts': ['NM_000059.3'],\n",
      "                 'frequencies': {'1000g': {'freq': {'AFR': 0.0,\n",
      "                                                    'AMR': 0.0,\n",
      "                                                    'EAS': 0.0,\n",
      "                                                    'EUR': 0.001,\n",
      "                                                    'G': 0.002,\n",
      "                                                    'SAS': 0.0092}},\n",
      "                                 'ExAC': {'count': {'AFR': 0,\n",
      "                                                    'AMR': 1,\n",
      "                                                    'EAS': 0,\n",
      "                                                    'FIN': 0,\n",
      "                                                    'G': 130,\n",
      "                                                    'Het': 124,\n",
      "                                                    'NFE': 13,\n",
      "                                                    'OTH': 0,\n",
      "                                                    'SAS': 116},\n",
      "                                          'freq': {'AFR': 0.0,\n",
      "                                                   'AMR': 8.644536652835408e-05,\n",
      "                                                   'EAS': 0.0,\n",
      "                                                   'FIN': 0.0,\n",
      "                                                   'G': 0.0010724125983732325,\n",
      "                                                   'NFE': 0.00019503705704083775,\n",
      "                                                   'OTH': 0.0,\n",
      "                                                   'SAS': 0.007027747485762753},\n",
      "                                          'hom': {'AFR': 0,\n",
      "                                                  'AMR': 0,\n",
      "                                                  'EAS': 0,\n",
      "                                                  'FIN': 0,\n",
      "                                                  'G': 3,\n",
      "                                                  'NFE': 0,\n",
      "                                                  'OTH': 0,\n",
      "                                                  'SAS': 3},\n",
      "                                          'num': {'AFR': 10322,\n",
      "                                                  'AMR': 11568,\n",
      "                                                  'EAS': 8652,\n",
      "                                                  'FIN': 6614,\n",
      "                                                  'G': 121222,\n",
      "                                                  'NFE': 66654,\n",
      "                                                  'OTH': 906,\n",
      "                                                  'SAS': 16506}},\n",
      "                                 'GNOMAD_EXOMES': {'count': {'AFR': 0,\n",
      "                                                             'AMR': 3,\n",
      "                                                             'ASJ': 0,\n",
      "                                                             'EAS': 0,\n",
      "                                                             'FIN': 0,\n",
      "                                                             'Female': 73,\n",
      "                                                             'G': 273,\n",
      "                                                             'Male': 200,\n",
      "                                                             'NFE': 30,\n",
      "                                                             'OTH': 3,\n",
      "                                                             'SAS': 237},\n",
      "                                                   'filter': {'G': ['PASS']},\n",
      "                                                   'freq': {'AFR': 0.0,\n",
      "                                                            'AMR': 8.933889219773674e-05,\n",
      "                                                            'ASJ': 0.0,\n",
      "                                                            'EAS': 0.0,\n",
      "                                                            'FIN': 0.0,\n",
      "                                                            'Female': 0.0006557082547381659,\n",
      "                                                            'G': 0.001108836574547936,\n",
      "                                                            'Male': 0.0014828654892714683,\n",
      "                                                            'NFE': 0.00026863907444884215,\n",
      "                                                            'OTH': 0.0005470459518599562,\n",
      "                                                            'SAS': 0.007699805068226121},\n",
      "                                                   'hom': {'AFR': 0,\n",
      "                                                           'AMR': 0,\n",
      "                                                           'ASJ': 0,\n",
      "                                                           'EAS': 0,\n",
      "                                                           'FIN': 0,\n",
      "                                                           'Female': 3,\n",
      "                                                           'G': 5,\n",
      "                                                           'Male': 2,\n",
      "                                                           'NFE': 0,\n",
      "                                                           'OTH': 0,\n",
      "                                                           'SAS': 5},\n",
      "                                                   'num': {'AFR': 15296,\n",
      "                                                           'AMR': 33580,\n",
      "                                                           'ASJ': 9848,\n",
      "                                                           'EAS': 17246,\n",
      "                                                           'FIN': 22296,\n",
      "                                                           'Female': 111330,\n",
      "                                                           'G': 246204,\n",
      "                                                           'Male': 134874,\n",
      "                                                           'NFE': 111674,\n",
      "                                                           'OTH': 5484,\n",
      "                                                           'SAS': 30780}},\n",
      "                                 'GNOMAD_GENOMES': {'count': {'AFR': 0,\n",
      "                                                              'AMR': 0,\n",
      "                                                              'ASJ': 0,\n",
      "                                                              'EAS': 0,\n",
      "                                                              'FIN': 0,\n",
      "                                                              'Female': 2,\n",
      "                                                              'G': 4,\n",
      "                                                              'Male': 2,\n",
      "                                                              'NFE': 4,\n",
      "                                                              'OTH': 0},\n",
      "                                                    'filter': {'G': ['PASS']},\n",
      "                                                    'freq': {'AFR': 0.0,\n",
      "                                                             'AMR': 0.0,\n",
      "                                                             'ASJ': 0.0,\n",
      "                                                             'EAS': 0.0,\n",
      "                                                             'FIN': 0.0,\n",
      "                                                             'Female': 0.0001443209698369173,\n",
      "                                                             'G': 0.0001291155584247902,\n",
      "                                                             'Male': 0.00011680878402055835,\n",
      "                                                             'NFE': 0.00026652452025586353,\n",
      "                                                             'OTH': 0.0},\n",
      "                                                    'hom': {'AFR': 0,\n",
      "                                                            'AMR': 0,\n",
      "                                                            'ASJ': 0,\n",
      "                                                            'EAS': 0,\n",
      "                                                            'FIN': 0,\n",
      "                                                            'Female': 0,\n",
      "                                                            'G': 0,\n",
      "                                                            'Male': 0,\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                            'NFE': 0,\n",
      "                                                            'OTH': 0},\n",
      "                                                    'num': {'AFR': 8736,\n",
      "                                                            'AMR': 838,\n",
      "                                                            'ASJ': 302,\n",
      "                                                            'EAS': 1622,\n",
      "                                                            'FIN': 3494,\n",
      "                                                            'Female': 13858,\n",
      "                                                            'G': 30980,\n",
      "                                                            'Male': 17122,\n",
      "                                                            'NFE': 15008,\n",
      "                                                            'OTH': 980}},\n",
      "                                 'esp6500': {'freq': {'AA': 0.0, 'EA': 0.0003}},\n",
      "                                 'inDB': {'count': {'OUST1': 1},\n",
      "                                          'filter': {'OUST1': ['PASS']},\n",
      "                                          'freq': {'OUST1': 0.0017006802721088435},\n",
      "                                          'hom': {'OUST1': 0},\n",
      "                                          'indications': {'OUSWES': {'OUST1': {'TrioPU': '1'}}},\n",
      "                                          'num': {'OUST1': 588}}},\n",
      "                 'prediction': {},\n",
      "                 'references': [{'pubmed_id': 28324225,\n",
      "                                 'source_info': {'HGMD': 'Primary literature '\n",
      "                                                         'report. Classified '\n",
      "                                                         'as VUS3 (IARC).'},\n",
      "                                 'sources': ['HGMD']},\n",
      "                                {'pubmed_id': 18844490,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 28492532,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 19471317,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']},\n",
      "                                {'pubmed_id': 10923033,\n",
      "                                 'source_info': {},\n",
      "                                 'sources': ['CLINVAR']}],\n",
      "                 'schema_version': 1,\n",
      "                 'transcripts': [{'HGVSc': 'c.198A>G',\n",
      "                                  'HGVSc_short': 'c.198A>G',\n",
      "                                  'HGVSp': 'c.198A>G(p.=)',\n",
      "                                  'amino_acids': 'Q',\n",
      "                                  'coding_region_distance': 0,\n",
      "                                  'codons': 'caA/caG',\n",
      "                                  'consequences': ['synonymous_variant'],\n",
      "                                  'dbsnp': ['rs28897700'],\n",
      "                                  'exon': '3/27',\n",
      "                                  'exon_distance': 0,\n",
      "                                  'hgnc_id': 1101,\n",
      "                                  'in_last_exon': 'no',\n",
      "                                  'is_canonical': True,\n",
      "                                  'protein': 'NM_000059.3',\n",
      "                                  'strand': 1,\n",
      "                                  'symbol': 'BRCA2',\n",
      "                                  'transcript': 'NM_000059.3'}],\n",
      "                 'worst_consequence': ['NM_000059.3']},\n",
      "  'change_from': 'A',\n",
      "  'change_to': 'G',\n",
      "  'change_type': 'SNP',\n",
      "  'chromosome': '13',\n",
      "  'genome_reference': 'GRCh37',\n",
      "  'id': 8,\n",
      "  'open_end_position': 32893344,\n",
      "  'samples': [{'affected': True,\n",
      "               'date_deposited': '2020-10-19T10:46:43.434024+00:00',\n",
      "               'family_id': None,\n",
      "               'father_id': None,\n",
      "               'genotype': {'allele_depth': {'G': 80, 'REF (A)': 107},\n",
      "                            'allele_ratio': 0.42780748663101603,\n",
      "                            'filter_status': 'PASS',\n",
      "                            'formatted': 'A/G',\n",
      "                            'genotype_quality': 99,\n",
      "                            'id': 8,\n",
      "                            'multiallelic': False,\n",
      "                            'needs_verification': False,\n",
      "                            'needs_verification_checks': {'allele_ratio': True,\n",
      "                                                          'dp': True,\n",
      "                                                          'hts': True,\n",
      "                                                          'pass': True,\n",
      "                                                          'snp': True},\n",
      "                            'sequencing_depth': 187,\n",
      "                            'type': 'Heterozygous',\n",
      "                            'variant_quality': 5000},\n",
      "               'id': 2,\n",
      "               'identifier': 'brca_sample_1',\n",
      "               'mother_id': None,\n",
      "               'proband': True,\n",
      "               'sample_type': 'HTS',\n",
      "               'sex': None,\n",
      "               'sibling_id': None}],\n",
      "  'start_position': 32893343,\n",
      "  'tags': ['has_references'],\n",
      "  'vcf_alt': 'G',\n",
      "  'vcf_pos': 32893344,\n",
      "  'vcf_ref': 'A'}]\n"
     ]
    }
   ],
   "source": [
    "# /api/v1/workflows/analyses/2/interpretations/2/alleles/\n",
    "# ?allele_ids=3%2C4%2C5%2C6%2C7%2C8&current=true&filterconfig_id=1\n",
    "\n",
    "query_url = 'api/v1/workflows/analyses/2/interpretations/2/alleles/'\n",
    "# Allele Ids needs to be a string with comma separated values, not a list!\n",
    "payload = {'allele_ids': ','.join([str(x) for x in allele_ids]), \n",
    "           'current': True, \n",
    "           'filterconfig_id': 1\n",
    "          }\n",
    "response = session.get('{}/{}'.format(url, query_url), params=payload)\n",
    "\n",
    "print(response.url)\n",
    "response = json.loads(response.content.decode('utf-8'))\n",
    "pprint(len(response))\n",
    "pprint(response)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[]\n"
     ]
    }
   ],
   "source": [
    "# api/v1/workflows/analyses/2/collisions/?allele_ids=3%2C4%2C5%2C6%2C7%2C8\n",
    "query_url = 'api/v1/workflows/analyses/2/collisions/'\n",
    "payload = {'allele_ids': allele_ids}\n",
    "response = session.get('{}/{}'.format(url, query_url), params=payload)\n",
    "response = json.loads(response.content.decode('utf-8'))\n",
    "pprint(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'3': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']},\n",
      "                 {'code': 'REQ_null_variant',\n",
      "                  'match': ['stop_gained'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['start_lost',\n",
      "                            'initiator_codon_variant',\n",
      "                            'transcript_ablation',\n",
      "                            'splice_donor_variant',\n",
      "                            'splice_acceptor_variant',\n",
      "                            'stop_gained',\n",
      "                            'frameshift_variant']},\n",
      "                 {'code': 'REQ_no_freq',\n",
      "                  'match': ['null_freq'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'frequencies.commonness',\n",
      "                  'value': ['null_freq']},\n",
      "                 {'code': 'PVS1', 'source': None},\n",
      "                 {'code': 'PPxPM2', 'source': None}]},\n",
      " '4': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']},\n",
      "                 {'code': 'REQ_null_variant',\n",
      "                  'match': ['frameshift_variant'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['start_lost',\n",
      "                            'initiator_codon_variant',\n",
      "                            'transcript_ablation',\n",
      "                            'splice_donor_variant',\n",
      "                            'splice_acceptor_variant',\n",
      "                            'stop_gained',\n",
      "                            'frameshift_variant']},\n",
      "                 {'code': 'REQ_no_freq',\n",
      "                  'match': ['null_freq'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'frequencies.commonness',\n",
      "                  'value': ['null_freq']},\n",
      "                 {'code': 'PVS1', 'source': None},\n",
      "                 {'code': 'PPxPM2', 'source': None}]},\n",
      " '5': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']},\n",
      "                 {'code': 'REQ_null_variant',\n",
      "                  'match': ['splice_donor_variant'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['start_lost',\n",
      "                            'initiator_codon_variant',\n",
      "                            'transcript_ablation',\n",
      "                            'splice_donor_variant',\n",
      "                            'splice_acceptor_variant',\n",
      "                            'stop_gained',\n",
      "                            'frameshift_variant']},\n",
      "                 {'code': 'REQ_no_freq',\n",
      "                  'match': ['null_freq'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'frequencies.commonness',\n",
      "                  'value': ['null_freq']},\n",
      "                 {'code': 'PVS1', 'source': None},\n",
      "                 {'code': 'PPxPM2', 'source': None}]},\n",
      " '6': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_missense',\n",
      "                  'match': ['missense_variant'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['missense_variant', 'protein_altering_variant']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']}]},\n",
      " '7': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']},\n",
      "                 {'code': 'REQ_null_variant',\n",
      "                  'match': ['stop_gained'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['start_lost',\n",
      "                            'initiator_codon_variant',\n",
      "                            'transcript_ablation',\n",
      "                            'splice_donor_variant',\n",
      "                            'splice_acceptor_variant',\n",
      "                            'stop_gained',\n",
      "                            'frameshift_variant']},\n",
      "                 {'code': 'REQ_no_freq',\n",
      "                  'match': ['null_freq'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'frequencies.commonness',\n",
      "                  'value': ['null_freq']},\n",
      "                 {'code': 'PVS1', 'source': None},\n",
      "                 {'code': 'PPxPM2', 'source': None}]},\n",
      " '8': {'codes': [{'code': 'REQ_GP_last_exon_not_important',\n",
      "                  'match': ['LENI'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.last_exon_important',\n",
      "                  'value': ['LENI']},\n",
      "                 {'code': 'REQ_GP_LOF_missense',\n",
      "                  'match': ['ANY'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'genepanel.disease_mode',\n",
      "                  'value': ['ANY']},\n",
      "                 {'code': 'REQ_less_common',\n",
      "                  'match': ['less_common'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'frequencies.commonness',\n",
      "                  'value': ['less_common']},\n",
      "                 {'code': 'REQ_no_aa_change',\n",
      "                  'match': ['synonymous_variant'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.consequences',\n",
      "                  'value': ['stop_retained_variant',\n",
      "                            'start_retained_variant',\n",
      "                            'synonymous_variant']},\n",
      "                 {'code': 'REQ_not_in_last_exon',\n",
      "                  'match': ['no'],\n",
      "                  'op': '$in',\n",
      "                  'source': 'transcript.in_last_exon',\n",
      "                  'value': ['no']}]}}\n"
     ]
    }
   ],
   "source": [
    "#POST - http://localhost:5001/api/v1/acmg/alleles/ - \n",
    "json_data = {\"allele_ids\":[\"3\",\"4\",\"5\",\"6\",\"7\",\"8\"],\"gp_name\":\"HBOCUTV\",\"gp_version\":\"v01\",\"referenceassessments\":[]}\n",
    "\n",
    "query_url = 'api/v1/acmg/alleles/' \n",
    "response = session.post('{}/{}'.format(url, query_url), json=json_data)\n",
    "response = json.loads(response.content.decode('utf-8'))\n",
    "pprint(response)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
